{% extends "base.html" %}
{% load url from future %}
{% comment %}
{% endcomment %}

{% block content %}
<div id="mainpage">
<h1>Genome Descriptions</h1>
<div class="content">
<p>Below you can browse descriptions of some of the genomes in our database in alphabetical order.  A list of the sources of our genomes can also be found in a plain format <a href="{% url 'home.views.sources' %}">here</a>.</p>
</div>

<hr />

<div class="node">
<h1>Acinetobacter</h1>
<div class="content">
<p>Acinetobacter is a Gram-negative genus of Bacteria belonging to the Gammaproteobacteria. Non-motile, Acinetobacter species are oxidase-negative, and occur in pairs under magnification. They are important soil organisms where they contribute to the mineralisation of, for example, aromatic compounds. Acinetobacter are a key source of infection in debilitated patients in the hospital.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Many of the characteristics of Acinetobacter ecology, taxonomy, physiology and genetics point to the possibility of exploiting its unique features for future applications. Acinetobacter strains are often ubiquitous, exhibit metabolic versatility, are robust and some provide convenient systems for modern molecular genetic manipulation and subsequent product engineering. These characteristics are being exploited in various biotechnological applications including biodegradation and bioremediation, novel lipid and peptide production, enzyme engineering, biosurfactant and biopolymer production and engineering of novel derivatives of these products. It is anticipated that progress in these fields will broaden the range of applications of Acinetobacter for modern biotechnology.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Aedes Aegypti</h1>
<div class="content">
<p>The yellow fever mosquito, Aedes aegypti (=Stegomyia aegypti, =Aedes (Stegomyia) aegypti), is a mosquito that can spread the dengue fever, Chikungunya and yellow fever viruses, and other diseases. The mosquito can be recognized by white markings on legs and a marking of the form of a lyre on the thorax. The mosquito originated from Africa but is now found in the tropics worldwide.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The genome of this species of mosquito was sequenced by a consortium including scientists at the J Craig Venter Institute and the University of Notre Dame, and published in 2007 . The effort in sequencing its DNA was intended to provide new avenues for research into insecticides and possible genetic modification to prevent the spread of virus. This was the second mosquito species to have its genome sequenced in full (the first was Anopheles gambiae). The published data included the 1.38 billion base pairs containing the insect's estimated 15,419 protein encoding genes. The sequence indicates that the species diverged from Drosophila melanogaster (the common fruit fly) about 250 million years ago, and that Anopheles gambiae and this species diverged about 150 million years ago.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Aeropyrum Pernix</h1>
<div class="content">
<p>Aeropyrum pernix is an extremophile species of Archaea that is part of the phylum Crenarchaeota. It is a heat-requiring species and was isolated from sediments in the sea off the coast of Japan.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Its complete genome was sequenced in 1999 and is 1,669 kilobases in size, with 2,694 possible genes detected. All of the genes in the TCA cycle were found except for that of ?-ketoglutarate dehydrogenase. In its place, the genes coding for the two subunits of 2-oxoacid:ferredoxin oxidoreductase were identified.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Agrobacterium Tumefaciens</h1>
<div class="content">
<p>Agrobacterium tumefaciens is the causal agent of Crown Gall disease (the formation of tumours) in over 140 species of dicot. It is a rod shaped, Gram negative soil bacterium (Smith et al., 1907). Symptoms are caused by the insertion of a small segment of DNA (known as the T-DNA, for 'transfer DNA') into the plant cell,[1] which is incorporated at a semi-random location into the plant genome.\n\nAgrobacterium tumefaciens (or A. tumefaciens) is an alphaproteobacterium of the family Rhizobiaceae, which includes the nitrogen fixing legume symbionts. Unlike the nitrogen fixing symbionts, tumor producing Agrobacterium are parasitic and do not benefit the plant. The wide variety of plants affected by Agrobacterium makes it of great concern to the agriculture industry[2].\n\nEconomically, A. tumefaciens is a serious pathogen of grape vines, stone fruits, nut trees, sugar beets, horse radish and rhubarb.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The DNA transmission capabilities of Agrobacterium have been extensively exploited in biotechnology as a means of inserting foreign genes into plants. Marc Van Montagu and Jeff Schell, (University of Ghent and Plant Genetic Systems, Belgium) discovered the gene transfer mechanism between Agrobacterium and plants, which resulted in the development of methods to alter Agrobacterium into an efficient delivery system for genetic engineering in plants.[8] The plasmid T-DNA that is transferred to the plant is an ideal vehicle for genetic engineering.[9] This is done by cloning a desired gene sequence into the T-DNA that will be inserted into the host DNA. This process has been performed using firefly luciferase gene to produce glowing plants. This luminescence has been a useful device in the study of plant chloroplast function and as a reporter gene.[10] It is also possible to transform Arabidopsis by dipping their flowers into a broth of Agrobacterium, the seed produced will be transgenic. Under laboratory conditions the T-DNA has also been transferred to human cells, demonstrating the diversity of insertion application.[11]\n\nThe mechanism by which Agrobacterium inserts materials into the host cell by a type IV secretion system, is very similar to mechanisms used by pathogens to insert materials (usually proteins) into human cells by type III secretion. It also employs a type of signaling conserved in many Gram-negative bacteria called quorum sensing. This makes Agrobacterium an important topic of medical research as well.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Anopheles Gambiae</h1>
<div class="content">
<p>Anopheles gambiae, refers to a complex of morphologically indistinguishable mosquitoes in the genus Anopheles, which contains the most important vectors of malaria in Sub-Saharan Africa, and the most efficient malaria vectors in the world. This species complex consists of:\n\n    * Anopheles arabiensis\n    * Anopheles bwambae\n    * Anopheles merus\n    * Anopheles melas\n    * Anopheles quadriannulatus\n    * Anopheles gambiae sensu stricto\n\nDespite being morphologically indistinguishable, individual species of Anopheles gambiae complex exhibit different behavioural traits. For example, the Anopheles quadriannulatus, is generally considered to be zoophilic, (taking its blood meal from animals) whereas Anopheles gambiae sensu stricto is generally anthropophilic. Identification to the individual species level using the molecular methods of Scott et al. (1993)[3] can have important implications in subsequent control measures.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
A. gambiae s.str. has been discovered to be currently in a state of diverging into two different species  the Mopti (M) and Savannah (S) strains  though as of 2007, the two strains are still considered to be a single species. The A. gambiae s.str. genome has been sequenced [4], though there is controversy over the choice of strain used, which is considered a hybrid of two different strains. Currently, ~90 miRNA have been predicted in the literature (38 miRNA officially listed in miRBase) for A. gambiae s.str. based upon conserved sequences to miRNA found in Drosophila.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Apis Mellifera</h1>
<div class="content">
<p>The European honey bee or western honey bee (Apis mellifera) is a species of honey bee. The genus Apis is Latin for "bee", and mellifera comes from Latin melli- "honey" and ferre "to bear"  hence the scientific name means "honey-bearing bee". The name was coined in 1758 by Carolus Linnaeus who, realizing that the bees do not bear honey, but nectar, tried later to correct it to Apis mellifica ("honey-making bee") in a subsequent publication. However, according to the rules of synonymy in zoological nomenclature, the older name has precedence. Some people who are unaware of this still use the incorrect subsequent spelling. As of October 28, 2006, the Honey Bee Genome Sequencing Consortium fully sequenced and analyzed the genome of Apis mellifera.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The European honey bee is the third insect, after the fruit fly and the mosquito, to have its genome mapped. According to the scientists who analysed its genetic code, the honey bee originated in Africa and spread to Europe in two ancient migrations. They have also discovered that the number of genes in the honey bees related to smell outnumber those for taste, and they have fewer genes for immunity than the fruit fly and the mosquito. The genome sequence revealed several groups of genes, particularly the genes related to circadian rhythms, were closer to vertebrates than other insects. Genes related to enzymes that control other genes were also vertabrate-like.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Aquifex Aeolicus</h1>
<div class="content">
<p>Aquifex aeolicus is a rod-shaped bacterium with a length of 2 to 6 micrometers and a diameter of around half a micrometer. It is one of a handful of species in the Aquificae phylum, an unusual group of thermophilic bacteria that are thought to be some of the oldest species of bacteria.\n\nA. aeolicus grows best in water between 85 to 95 °C, and can be found near underwater volcanoes or hot springs. It requires oxygen to survive (though it can grow in levels of oxygen as low as 7.5 ppm), and its method of respiration produces water as a byproduct. ("Aquifex" means "water-maker.") Members of the species tend to form large cell conglomerations, comprised of up to 100 individual cells. It was discovered around islands north of Sicily.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The genome of A. aeolicus has been successfully mapped. This was made easier by the fact that the length of the genome is only about a third of the length of the genome for E. coli. Comparison of the Aquifex aeolicus genome to other organisms showed that around 16% of its genes originated from the Archaea domain.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Arabidopsis Thaliana</h1>
<div class="content">
<p>Arabidopsis thaliana ; thale cress, mouse-ear cress or Arabidopsis, is a small flowering plant native to Europe, Asia, and northwestern Africa.[1] A spring annual with a relatively short life cycle, Arabidopsis is popular as a model organism in plant biology and genetics. Its genome is one of the smallest plant genomes and was the first plant genome to be sequenced. Arabidopsis is a popular tool for understanding the molecular biology of many plant traits, including flower development and light sensing.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Ongoing research on Arabidopsis thaliana is being performed on the International Space Station by the European Space Agency. The goals are to study the growth and reproduction of plants from seed to seed in microgravity.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Archaeoglobus Fulgidus</h1>
<div class="content">
<p>The genus Archaeoglobus is a hyperthermophilic member of the Euryarchaeota. It is composed of two species, A. fulgidus and A. profundus which were isolated from hydrothermal vents. Archaeoglobus can also be found in high-temperature oil fields where they may contribute to oil field souring. Optimal growth of these organisms occurs at approximately 83°C. Metabolically, Archaeoglobus are sulfate-reducing Archaea, coupling to the reduction of sulfate (to sulfide) to the oxidation of many different organic carbon sources, including complex polymers. Archaeoglobus can also live chemolithoautotrophically by coupling the oxidation of thiosulfate to the reduction of hydrogen gas. Archaeoglobus are the only organisms other than the traditional sulfate-reducing bacteria capable of the reduction of sulfate.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Intriguingly, the completion of the complete A. fulgidus genome sequence (Klenk et al., 1997) revealed the presence of a nearly complete set of genes for methanogenesis. The function of these genes in A. fulgidus remains unknown, although the lack of the enzyme methyl-CoM reductase does not allow for methanogenesis to occur by a mechanism similar to that found in other methanogens.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus Anthracis - Ames Strain</h1>
<div class="content">
<p>Bacillus anthracis is a Gram-positive spore-forming, rod-shaped bacterium, with a width of 1-1.2µm and a length of 3-5µm. It can be grown in an ordinary nutrient medium under aerobic or anaerobic conditions. It bears close genotypical and phenotypical resemblance to Bacillus cereus and Bacillus thuringiensis. All three species share cellular dimensions and morphology. All form oval spores located centrally in a non-swollen sporangium. Bacillus anthracis spores in particular are highly resilient, surviving extremes of temperature, low-nutrient environments, and harsh chemical treatment over decades or centuries.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The Ames strain is one of 89 strains of the anthrax bacterium (Bacillus anthracis). It was isolated from a diseased cow that died in Texas in 1981. Researchers at the time mistakenly believed the strain came from Ames, Iowa and mislabeled the specimen.\n\nThe Ames strain came to wide public attention in association with the 2001 anthrax attacks. Seven letters mailed to media outlets and US Senators on September 18, 2001 and October 9, 2001 contained anthrax bacteria of this particular strain. This strain is a monomorphic disease, mutating slowly if at all.\n\nBecause of its virulence, the Ames strain is used by the United States as a "gold standard" for development of vaccines and testing their effectiveness, starting in the 1980s, after work on weaponizing the Vollum 1B strain ended and all weaponized stocks were destroyed after the end of the U.S. biological warfare program in 1969.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus Cereus</h1>
<div class="content">
<p>Bacillus cereus is an endemic, soil-dwelling, Gram-positive, rod-shaped, beta hemolytic bacterium. Some strains are harmful to humans and cause foodborne illness, while other strains can be beneficial as probiotics for animals. It is the cause of "Fried Rice Syndrome". B. cereus bacteria are facultative aerobes, and like other members of the genus Bacillus can produce protective endospores.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
B. cereus competes with other microorganisms such as Salmonella and Campylobacter in the gut, so its presence reduces the numbers of those microorganisms. In food animals such as chickens, rabbits, and pigs, some harmless strains of B. cereus are used as a probiotic feed additive to reduce Salmonella in the intestines and cecum. This improves the animals' growth as well as food safety for humans who eat their meat.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bdellovibrio Bacteriovorus</h1>
<div class="content">
<p>Bdellovibrio is a genus of Gram-negative, obligate aerobic bacteria.\n\nOne of the more notable characteristics of this genus is that members parasitize other Gram-negative bacteria by entering into their periplasmic space and feeding on the biopolymers, e.g. proteins and nucleic acids, of their hosts. After entering the periplasmic space of its host the Bdellovibrio bacterium forms a structure called a bdelloplast, which consists of both predator and prey. The predator cell can remain dormant at this stage, without affecting the viability of the host. In most cases, though, Bdellovibrio devours its prey and moves on.\n\nBdellovibrio species are found in river water or soil and live an intraperiplasmic existence. To enrich for Bdellovibrio use NB/500 (nutrient broth at 1:500 dilution) and mix with hot soft agar with E. coli at 30 °C for one week.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Bdellovibrio bacteriovorus was first described by Stolp and Starr in 1963. Two other species, Bdellovibrio starrii and Bdellovibrio stolpii, have been moved to a separate genus Bacteriovorax.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bordetella Bronchiseptica</h1>
<div class="content">
<p>Bordetella bronchiseptica is a small, gram-negative, rod-shaped bacterium of the genus Bordetella. It can cause infectious bronchitis, but rarely infects humans. Closely related to B. pertussisthe obligate human pathogen that causes pertussis or whooping coughB. bronchiseptica can persist in the environment for extended periods.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bartonella Henselae</h1>
<div class="content">
<p>Bartonella henselae is a proteobacterium that can cause bacteremia, endocarditis, bacillary angiomatosis, and peliosis hepatis. It is also the causative agent of cat-scratch disease (Bartonellosis) which, as the name suggests, occurs after a cat bite or scratch. The disease is characterized by lymphadenopathy (swelling of the lymph nodes) and fever.\n\nPeliosis hepatis caused by B. henselae can occur alone or develop with cutaneous bacillary angiomatosis or bacteremia. Patients with peliosis hepatitis present with gastrointestinal symptoms, fever, chills, and an enlarged liver and spleen containing blood-filled cavities. This systemic disease is seen in patients infected with HIV and other immunocompromised individuals.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Bartonella henselae is a member of the class of the Bartonella genus, one of the most common type of bacteria in the world.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bartonella Quintana</h1>
<div class="content">
<p>Bartonella quintana, also known as Rochalimaea quintana, is a microorganism that is transmitted by the human body louse.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This microorganism is trench fever's etiologic agent. This bacteria resulted in about 1 million soldiers in Europe during World War I being infected with Trench Fever.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bifidobacterium Longum</h1>
<div class="content">
<p>Bifidobacterium is a genus of Gram-positive, non-motile, often branched anaerobic bacteria. Bifidobacteria are one of the major genera of bacteria that make up the gut flora, the bacteria that reside in the colon. Bifidobacteria aid in digestion, are associated with a lower incidence of allergies and also prevent some forms of tumor growth.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Some bifidobacteria are being used as probiotics.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bordetella Bronchiseptica</h1>
<div class="content">
<p>Bordetella bronchiseptica is a small, gram-negative, rod-shaped bacterium of the genus Bordetella. It can cause infectious bronchitis, but rarely infects humans. Closely related to B. pertussisthe obligate human pathogen that causes pertussis or whooping coughB. bronchiseptica can persist in the environment for extended periods.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
In veterinary medicine, B. bronchiseptica leads to a range of pathologies in different hosts. It is a serious disease of dogs, pigs, and rabbits and has been seen in cats, horses, and seals. There is a PCR test for the pathogen.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bordetella Parapertussis</h1>
<div class="content">
<p>Bordetella parapertussis is a small Gram-negative bacterium of the genus Bordetella which is adapted to colonise the mammalian respiratory tract. Pertussis caused by B. parapertussis manifests with similar symptoms to B. pertussis-derived disease but tends to be generally less severe.\n\nTwo lineages of B. parapertussis have been described. The first infects humans and is responsible for a minority of cases of the disease pertussis (also known as whooping cough). The second, ovine, lineage causes chronic non-progressive pneumonia in sheep. Both lineages are thought to have evolved from a B. bronchiseptica-like ancestor. This disease can be symptomatic or asymptomatic and may predispose hosts to secondary infection.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bordetella Pertussis</h1>
<div class="content">
<p>Pertussis (or Whooping Cough), is an infection of the respiratory system and characterized by a whooping sound when the person breathes in. In the US it killed 5,000 to 10,000 people per year before a vaccine was available. Vaccination has transformed this and between 1985-88 fewer than 100 children died from pertussis. Worldwide in 2000, according to the WHO, around 39 million people were infected annually and about 297,000 died. A graph is available showing the dramatic effect of introducing vaccination in England.[1]\n\nThe infection occurs most with children under the age of one when they are immunized or children with faded immunity, normally around the age 11 through 18. The signs and symptoms are similar to a common cold: runny nose, sneezing, mild cough, and low-grade fever. After a spell, they might make a whooping sound when breathing in or vomit. Adults have milder symptoms, like prolonged coughing without the whoop. The patient becomes most contagious during the catarrhal stage of infection, normally 2 weeks after the coughing begins. It may become airborne when the person coughs, sneezes, or laughs. Pertussis vaccine is part of the DTaP (diphtheria, tetanus, acellular pertussis) immunization. The paroxysmal cough precedes a crowing inspiratory sound characteristic of pertussis. Infants less than 6 months may not have the typical whoop. A coughing spell may last a minute or more, producing cyanosis, apnoea and seizures. A prolonged cough may be irritating and sometimes a disabling cough may go undiagnosed in adults for many months.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Borrelia Burgdorferi</h1>
<div class="content">
<p>Borrelia burgdorferi is a species of Gram negative bacteria of the spirochete class of the genus Borrelia. B. burgdorferi is predominant in North America, but also exists in Europe, and is the agent of Lyme disease.\n\nIt is a zoonotic, vector-borne disease transmitted by ticks and is named after the researcher Willy Burgdorfer who first isolated the bacterium in 1982. B. burgdorferi is one of the few pathogenic bacteria that can survive without iron, having replaced all of its iron-sulfur cluster enzymes with enzymes that use manganese, thus avoiding the problem many pathogenic bacteria face in acquiring iron.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bos Taurus</h1>
<div class="content">
<p>Cattle, colloquially referred to as cows, are domesticated ungulates, a member of the subfamily Bovinae of the family Bovidae.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
In the April 24, 2009 edition of the journal Science it was reported that a team of researchers led by the National Institutes of Health and the U.S. Department of Agriculture have mapped the bovine genome. The scientists found that the cattle has approximately 22,000 genes, and 80 percent of their genes are shared with humans, and they have approximately 1,000 genes they share with dogs and rodents but are not found in humans.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Brucella Melitensis</h1>
<div class="content">
<p>Brucella melitensis is a gram negative coccobacillus bacteria that causes brucellosis, a disease affecting sheep, cattle, and sometimes humans. It is primarily considered to be associated with caprine brucellosis, but is also found in sheep. It causes reproductive disease such as orchitis, epididymitis, mastitis, and abortions. It is zoonotic, unlike Brucella ovis, causing Malta fever or localized brucellosis in humans.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Brucella Suis</h1>
<div class="content">
<p>Swine brucellosis is a contagious disease in pigs caused by the bacteria, Brucella suis. The disease spreads in semen during breeding and by ingestion, inhalation, or eye contact with bacteria in milk, reproductive fluids, placenta, aborted fetuses and urine. The disease primarily occurs in adult pigs which show non-specific infertility, abortion or lack of sexual drive. Boars can show signs of orchitis, lameness, arthritis, abscesses and posterior paralysis. There is no treatment for the disease and no effective vaccine.\n\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The U.S. Army Chemical Corps selected Brucella suis as its first standardized biological agent in 1949. The M33 cluster bomb was designed to be used with B. suis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Buchnera Aphidicola</h1>
<div class="content">
<p>Buchnera aphidicola a member of the Proteobacteria, is the primary endosymbiont of aphids (A. pisum). It is believed that Buchnera was once a free living gram negative ancestor similar to a modern Enterobacteriaceae such as Escherichia coli. Buchnera are 3 µm in diameter and have some of the key characteristics of their Enterobacteriaceae relatives such as a gram-negative cell wall. However, unlike most other gram-negative bacteria, Buchnera lack the genes to produce lipopolysaccharides (LPS) for their outer membrane. The long association with aphids and the limitation of crossover events due to strictly vertical transmission has seen the deletion of genes required for anaerobic respiration, the synthesis of amino sugars, fatty acids, phospholipids, and complex carbohydrates. This has resulted not only in one of the smallest known genomes of any living organism, it is also one of the most genetically stable.\n\nThe symbiotic relationship with aphids began between 160 million and 280 million years ago[1], and has persisted through maternal transmission and co-speciation. Aphids have developed bacteriocyte cells to house Buchnera. It is estimated that a mature aphid may carry 5.6 × 106 Buchnera individuals. Buchnera have lost regulatory factors allowing continuous overproduction of tryptophan and other amino acids. Each bacteriocyte contains multiple vesicles, symbiosomes derived from the plasma membrane.\n\nThey also increase the transmission of plant viruses by producing symbionin, this protein binds to the viral coat and protects it inside the aphid. This makes it more likely that the virion will survive and be able to infect another plant when the aphid next feeds.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Buchnera was first named by Paul Baumann and his graduate student, and the first molecular characterization of a symbiotic bacterium was carried out by Baumann, using Buchnera. The initial studies on Buchnera later led to studies on symbionts of many groups of insects, pursued by numerous investigators, including Paul and Linda Baumann, Nancy Moran, Serap Aksoy, Roy Gross, who together investigated symbionts of aphids, tsetse flies, ants, leafhoppers, mealybugs, whiteflies, psyllids and others.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia Mallei</h1>
<div class="content">
<p>Burkholderia mallei is a gram-negative bipolar aerobic bacterium, a Burkholderia-genus human and animal pathogen causing Glanders; the Latin name of this disease (malleus) gave name to the causative agent species. It is closely related to B. pseudomallei, and by multilocus sequence typing, it is a subspecies of B. pseudomallei,[1] or at the very least, paraphyletic. Unlike closely related Burkholderia pseudomallei and other genus members, the bacterium is nonmotile; its shape is something in between a rod and a coccus measuring some 1.53 ?m in length and 0.51?m in diameter with rounded ends.\n\nThe bacterium is susceptible to numerous disinfectants including benzalkonium chloride, iodine, mercuric chloride, potassium permanganate, 1% sodium hypochlorite and ethanol. The microorganism can also be destroyed by heating or UV. Antibiotics such as streptomycin, amikacin, tetracycline, doxycycline and sulfathiazole have been reported to be effective against the bacteria in vitro.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
B. mallei as well as B. pseudomallei have a history of being on a list of potential biological warfare agents.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia Pseudomallei</h1>
<div class="content">
<p>Burkholderia pseudomallei is a Gram-negative, bipolar, aerobic, motile rod-shaped bacterium. A human and animal pathogen, B. pseudomallei causes melioidosis.\n\nB. pseudomallei measures 25 ?m in length and 0.40.8 ?m in diameter and are capable of self-propulsion using flagellae. The bacteria can grow in a number artficial nutrient environments, especially betaine- and arginine-containing.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
B. pseudomallei infection in humans is called melioidosis. The mortality of melioidosis is 20 to 50% even with treatment.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Caenorhabditis Elegans</h1>
<div class="content">
<p>C. elegans is unsegmented, vermiform, bilaterally symmetrical, with a cuticle integument, four main epidermal cords and a fluid-filled pseudocoelomate cavity. Members of the species have many of the same organ systems as other animals. In the wild, they feed on bacteria that develop on decaying vegetable matter. C. elegans has two sexes: hermaphrodites and males. Individuals are almost all hermaphrodite, with males comprising just 0.05% of the total population on average. The basic anatomy of C. elegans includes a mouth, pharynx, intestine, gonad, and collagenous cuticle. Males have a single-lobed gonad, vas deferens, and a tail specialized for mating. Hermaphrodites have two ovaries, oviducts, spermatheca, and a single uterus.\n\nC. elegans eggs are laid by the hermaphrodite. After hatching, they pass through four larval stages (L1-L4). When crowded or in the absence of food, C. elegans can enter an alternative third larval stage called the dauer state. Dauer larvae are stress-resistant and do not age. Hermaphrodites produce all their sperm in the L4 stage (150 sperm per gonadal arm) and then switch over to producing oocytes. The sperm are stored in the same area of the gonad as the oocytes until the first oocyte pushes the sperm into the spermatheca (a kind of chamber where the oocytes become fertilized by the sperm). The male can inseminate the hermaphrodite, which will use male sperm preferentially (both types of sperm are stored in the spermatheca). When self-inseminated the wild-type worm will lay approximately 300 eggs. When inseminated by a male, the number of progeny can exceed 1,000. At 20 °C, the laboratory strain of C. elegans has an average life span of approximately 23 weeks and a generation time of approximately 4 days. Hermaphrodites can mate with males or self-fertilize.\n\nC. elegans has five pairs of autosomes and one pair of sex chromosomes. Sex in C. elegans is based on an X0 sex-determination system. Hermaphrodite C. elegans have a matched pair of sex chromosomes (XX); the rare males have only one sex chromosome (X0).\n\nIt is interesting to note that C. elegans is one of the few forms of life not known to have a natural virus.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
C. elegans was the first multicellular organism to have its genome completely sequenced. The finished genome sequence was published in 1998, although a number of small gaps were present (the last gap was finished by October 2002). The C. elegans genome sequence is approximately 100 million base pairs long and contains approximately 18,400 genes. The vast majority of these genes encode for proteins but there are likely to be as many as 1,000 RNA genes. Scientific curators continue to appraise the set of known genes, such that new gene predictions continue to be added and incorrect ones modified or removed.\n\nIn 2003, the genome sequence of the related nematode C. briggsae was also determined, allowing researchers to study the comparative genomics of these two organisms. Work is now ongoing to determine the genome sequences of more nematodes from the same genus such as C. remanei, C. japonica and C. brenneri. These newer genome sequences are being determined by using the whole genome shotgun technique which means that the resulting genome sequences are likely to not be as complete or accurate as C. elegans (which was sequenced using the 'hierarchical' or clone-by-clone approach).\n\nThe official version of the C. elegans genome sequence continues to change as and when new evidence reveals errors in the original sequencing (DNA sequencing is not an error-free process). Most changes are minor, adding or removing only a few base pairs (bp) of DNA. E.g. the WS169 release of WormBase (December 2006) lists a net gain of 6 bp to the genome sequence. Occasionally more extensive changes are made, e.g. the WS159 release of May 2006 added over 300 bp to the sequence.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Campylobacter Jejuni</h1>
<div class="content">
<p>Campylobacter jejuni is a species of curved, rod-shaped, non-spore forming, Gram-negative microaerophilic, bacteria commonly found in animal feces. It is one of the most common causes of human gastroenteritis in the world. Food poisoning caused by Campylobacter species can be severely debilitating but is rarely life-threatening. It has been linked with subsequent development of Guillain-Barré syndrome (GBS), which usually develops two to three weeks after the initial illness.\n\nC. jejuni is commonly associated with poultry and naturally colonises the GI tract of many bird species. It has also been isolated from wombat and kangaroo feces, being a cause of bushwalkers' diarrhea. Contaminated drinking water and unpasteurized milk provide an efficient means for distribution. Contaminated food is a major source of isolated infections, with incorrectly prepared meat and poultry normally the source of the bacteria.\n\nInfection with C. jejuni usually results in enteritis, which is characterised by abdominal pain, diarrhea, fever, and malaise. The symptoms usually persist for between 24 hours and a week, but may be longer. Diarrhea can vary in severity from loose stools to bloody stools. No antibiotics are usually given as the disease is self-limiting, however, severe or prolonged cases may require ciprofloxacin, erythromycin or norfloxacin. Fluid and electrolyte replacement may be required for serious cases.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The first full-genome sequence of C. jejuni was performed in 2000 (strain NCTC11168 with a circular chromosome of 1,641,481 base pairs).
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candida Albicans</h1>
<div class="content">
<p>Candida albicans is a diploid fungus (a form of yeast), which is capable of sexual reproduction but not of meiosis, and a causal agent of opportunistic oral and genital infections in humans. Systemic fungal infections (fungemias) have emerged as important causes of morbidity and mortality in immunocompromised patients (e.g., AIDS, cancer chemotherapy, organ or bone marrow transplantation). In addition, hospital-related infections in patients not previously considered at risk (e.g. patients in an intensive care unit) have become a cause of major health concern.\n\nC. albicans is commensal and is among the gut flora, the many organisms which live in the human mouth and gastrointestinal tract. Under normal circumstances, C. albicans lives in 80% of the human population with no harmful effects, although overgrowth results in candidiasis. Candidiasis is often observed in immunocompromised individuals such as HIV-positive patients. Candidiasis also may occur in the blood and in the genital tract. Candidiasis, also known as "thrush", is a common condition which is usually easily cured in people who are not immunocompromised. To infect host tissue, the usual unicellular yeast-like form of C. albicans reacts to environmental cues and switches into an invasive, multicellular filamentous forms.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
One of the most interesting features of the C. albicans genome is the occurrence of numeric and structural chromosomal rearrangements as means of generating genetic diversity, named chromosome length polymorphisms (contraction/expansion of repeats), reciprocal translocations, chromosome deletions and trisomy of individual chromosomes. These karyotypic alterations lead to changes in the phenotype, which is an adaptation strategy of this fungus. These mechanisms will be better understood with the complete analysis of the C. albicans genome.\n\nThe C. albicans genome for strain SC5314 was sequenced at the Stanford DNA Sequencing and Technology Center. The genome of the WO1 strain was sequenced by the Broad Institute of MIT and Harvard.\n\nThe sequencing of the C. albicans genome and subsequently of the genomes of several other medically relevant Candida species has profoundly and irreversibly changed the way Candida species are now investigated and understood. The C. albicans genome sequencing effort was launched in October 1996. Successive releases of the sequencing data and genome assemblies have marked the last 10 years, culminating with the release of the diploid assembly 19 which provided a haploid version of the genome along with data on allelic regions in the genome. A refined assembly 20 with the eight assembled C. albicans chromosomes was released in the summer of 2006. Importantly, the availability of sequencing data prior to the completion of the genome sequence has made it possible to start C. albicans post-genomics early on. In this regard, genome databases have been made available to the research community providing different forms of genome annotation. These have been merged in a community-based annotation hosted by the Candida Genome Database. The availability of the genome sequence has paved the way for the implementation of post-genomic approaches to the study of C. albicans: macroarrays and then microarrays have been developed and used to study the C. albicans transcriptome; proteomics has also been developed and complements transcriptional analyses; furthermore, systematic approaches are becoming available to study the contribution of each C. albicans gene in different contexts. Other Candida genome sequences have been, or are being, determined: C. glabrata, C. dubliniensis, C. parapsilosis, C. guilliermondii, C. lusitaniae, and C. tropicalis. These species will soon enter the post-genomic era as well and provide interesting comparative data. The genome sequences obtained for the different Candida species along with those of non-pathogenic hemiascomycetes provide a wealth of knowledge on the evolutionary processes which have shaped the hemiascomycete group as well as those which may have contributed to the success of different Candida species as pathogens.\n\nThe genome of C. albicans is highly dynamic and this variability has been used advantageously for molecular epidemiological studies of C. albicans and population studies in this species. A remarkable discovery which has arisen from the genome sequence is the presence of a parasexual cycle in C. albicans. This parasexual cycle is under the control of mating-type loci and switching between white and opaque phenotypes. Investigating the role which the mating process plays in the dynamics of the C. albicans population or in other aspects of C. albicans biology and pathogenicity will undoubtedly represent an important focus for future research.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candida Glabrata</h1>
<div class="content">
<p>Candida glabrata is a haploid yeast of the genus Candida, previously known as Torulopsis glabrata. This species of yeast is non-dimorphic and no mating activity has been observed. Until recently, C. glabrata was thought to be a primarily non-pathogenic organism. However, with the ever increasing population of immunocompromised individuals, trends have shown C. glabrata to be a highly opportunistic pathogen of the urogenital tract, and of the bloodstream (Candidemia). It is especially prevalent in HIV positive people, and the elderly.\n\nThere are two widely cited potential virulence factors that contribute to the pathogenicity of C. glabrata. The first is a series of adhesins coded by the EPA (epithelial adhesin) genes. These genes, located in the subtelomeric region, can respond to environmental cues that allow them to be expressed en masse so the organism can adhere to biotic and abiotic surfaces in microbial mats. This is also the suspected mechanism by which C. glabrata forms microbial "biofilms" on urinary catheters, and less commonly in-dwelling IV catheters. It also causes problems with dental devices, such as dentures.\n\nA major phenotype and potential virulence factor that C. glabrata possesses is low-level intrinsic resistance to the azole drugs, which are the most commonly prescribed antifungal (antimycotic) drugs. These drugs, including fluconazole and ketoconazole, are "not effective in 15-20% of cases"* against C. glabrata. While some have said that the organism possesses an "innate" immunity to the drugs, it is more accurate to say that the organism possesses an evolved resistance to the drugs. It is still highly vulnerable to polyene drugs such as amphotericin B and nystatin, along with variable vulnerability to flucytosine and caspofungin. However doctors asked about amphotericin B indicate that this is a drug of last resort, in that this drug can often kill the patient while "curing" the fungal infection.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Canis Familiaris</h1>
<div class="content">
<p>The dog (Canis lupus familiaris, pronounced /?ke?.nis ?lu?p?s f??m?li??ris/) is a domesticated subspecies of the gray wolf, a member of the Canidae family of the order Carnivora. The term is used for both feral and pet varieties.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The domestic dog has been one of the most widely kept working and companion animals in human history.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Caulobacter Crescentus</h1>
<div class="content">
<p>Caulobacter crescentus is a Gram-negative, oligotrophic bacterium widely distributed in fresh water lakes and streams. It plays an important role in the carbon cycle.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Caulobacter is an important model for study of the regulation of the cell cycle and cellular differentiation. Caulobacter daughter cells have two very different forms. One is a mobile "swarmer" cell that has a flagellum for swimming. The other, called the "stalked" cell has a long tubular stalk structure protruding from one pole that has an adhesive holdfast material on its end, with which the stalked cell can adhere to surfaces. Chromosome replication and cell division only occurs in the stalked cells. Swarmer cells differentiate into stalked cells as they mature. Often surviving in nutrient-poor environs, Caulobacter crescentus is a Gram-negative bacterium ubiquitous in fresh water, soil, and sea water. C. crescentus exhibits a dimorphic life cycle that most likely provides an advantage in such competitive environments. The stalk cell can attach to a surface, while the swarmer cell can search for nutrients. The adhesive material of the holdfast has been reported to be one of the strongest natural glues.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Cavia Porcellus</h1>
<div class="content">
<p>The guinea pig (also commonly called the Cavy after its scientific name, Cavia porcellus) is a species of rodent belonging to the family Caviidae and the genus Cavia. Despite their common name, these animals are not pigs, nor do they come from Guinea. They originated in the Andes, and studies based on biochemistry and hybridization suggest they are domesticated descendants of a closely related species of cavy such as Cavia aperea, C. fulgida, or C. tschudii, and therefore do not exist naturally in the wild.The guinea pig plays an important role in the folk culture of many Indigenous South American groups, especially as a food source, but also in folk medicine and in community religious ceremonies. Since the 1960s, efforts have been made to increase consumption of the animal outside South America.\n\nIn Western societies, the guinea pig has enjoyed widespread popularity as a household pet since its introduction by European traders in the 16th century. Their docile nature, their responsiveness to handling and feeding, and the relative ease of caring for them, continue to make the guinea pig a popular pet. Organizations devoted to competitive breeding of guinea pigs have been formed worldwide, and many specialized breeds of guinea pig, with varying coat colors and compositions, are cultivated by breeders.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Biological experimentation on guinea pigs has been carried out since the 17th century. The animals were frequently used as a model organism in the 19th and 20th centuries, giving way to the metaphor "guinea pig" for a test subject, but have since been largely replaced by other rodents such as mice and rats. They are still used in research, primarily as models for human medical conditions such as juvenile diabetes, tuberculosis, scurvy, and pregnancy complications.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlamydia Muridarum</h1>
<div class="content">
<p>Chlamydia muridarum is an intracellular bacterial species that at one time belonged to Chlamydia trachomatis. However, C. trachomatis naturally only infects humans and C. muridarum naturally infects only members of the family Muridae (includes both mice and hamsters).\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Two strains of Chlamydia muridarum, MoPn (originally named Nigg) and SFPD, have been isolated from mice and hamsters. Glycogen production by both strains has been demonstrated, although it is difficult to detect glycogen in MoPn. The chromosome and extrachromosomal plasmid of MoPn have been sequenced.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlamydia Trachomatis</h1>
<div class="content">
<p>Chlamydia trachomatis, an obligate intracellular human pathogen, is one of three bacterial species in the genus Chlamydia. C. trachomatis is Gram-indeterminate (i.e. cannot be stained with the Gram stain); structurally the organism is Gram-negative. Identified in 1907, C. trachomatis was the first chlamydial agent discovered in humans.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
C. trachomatis is a obligate intracellular pathogen (i.e. the bacterium lives within human cells) and can cause numerous disease states in both men and women. Both sexes can display urethritis, proctitis (rectal disease and bleeding), trachoma, and infertility. The bacterium can cause prostatitis and epididymitis in men. In women, cervicitis, pelvic inflammatory disease (PID), ectopic pregnancy, and acute or chronic pelvic pain are frequent complications. C. trachomatis is also an important neonatal pathogen, where it can lead to infections of the eye (trachoma) and pulmonary complications.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlamydophila Caviae</h1>
<div class="content">
<p>Chlamydophila caviae can be recovered from the conjunctiva of Guinea pigs suffering from ocular inflammation and eye discharge. It is also possible to infect the genital tract of Guinea pigs with C. caviae and elicit a disease that is very similar to human Chlamydia trachomatis infection. C. caviae infects primarily the mucosal epithelium and is not invasive.\n\nC. caviae is markedly specific for Guinea pigs, as attempts to infect mice, hamsters, rabbits and gerbils have been unsuccessful, except for one experimentally infected gerbil. The five known C. caviae isolates are indistinguishable, based on ompA gene sequence. C. caviae strain GPIC contains an extrachromosomal plasmid, pCpGP1.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlamydophila Pneumoniae</h1>
<div class="content">
<p>Chlamydophila pneumoniae is a species of Chlamydophila bacteria that infects humans and is a major cause of pneumonia.\n\nUntil recently it was known as "Chlamydia pneumoniae", and that name is used as an alternate in some sources. In some cases, to avoid confusion, both names are given.\n\nC. pneumoniae has a complex life cycle and must infect another cell in order to reproduce and thus is classified as an obligate intracellular pathogen. In addition to its role in pneumonia, there is evidence associating C. pneumoniae with atherosclerosis, Alzheimer's disease and with asthma. The full genome sequence for C. pneumoniae was published in 1999.\n\nC. pneumoniae also infects and causes disease in Koalas, emerald tree boa (Corallus caninus), iguanas, chameleons, frogs, and turtles.\n\nThe first known case of infection with C. pneumoniae was a case of sinusitis in Taiwan.\n\nThis atypical bacterium commonly causes pharyngitis, bronchitis and atypical pneumonia mainly in elderly and debilitated patients but in healthy adults also.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chromobacterium Violaceum</h1>
<div class="content">
<p>Chromobacterium violaceum is a Gram-negative, facultative anaerobic, non-sporing coccobacillus. It is part of the normal flora of water and soil of tropical and sub-tropical regions of the world. It produces a natural antibiotic called violacein. It grows readily on nutriet agar, producing distinctive smooth low convex colonies with a dark violet metallic sheen (due to violacein production).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Its full genome was published in 2003.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ciona Intestinalis</h1>
<div class="content">
<p>Ciona intestinalis or vase tunicate is a Urochordata (sea squirt) whose genome has been sequenced. It has become, over the past decade, a major experimental model for developmental biologists. It is widely distributed in North European waters, but has also spread to other parts of the world as an invasive species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
C. intestinalis is one of the animals whose full genome has been sequenced. It has a relatively small genome (about 160 Mbp) consisting of 14 pairs of chromosomes with about 16,000 genes. The draft genome analysis identified nine Hox genes, which are Ci-Hox1, 2, 3, 4, 5, 6, 10, 12, and 13. Ciona savignyi, the closest relative of Ciona intestinalis, also have the same set of Hox genes. The organization of Hox genes is only known for C. intestinalis among asicdians. The nine Hox genes are located on two chromosomes; Ci-Hox1 to 10 on one chromosome and Ci-Hox12 and 13 on another. The intergenic distances within the Ciona Hox genes are extraordinarily long. Seven Hox genes, Ci-Hox1 to 10, are distributed, spanning approximately half the length of the chromosome. From comparisons to the Hox gene expressions and locations in other species, it is likely that the ascidian genomes are under a dispersing condition
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium acetobutylicum</h1>
<div class="content">
<p>Clostridium acetobutylicum, included in the genus Clostridium, is a commercially valuable bacterium. It is sometimes called the "Weizmann Organism", after Chaim Weizmann, who in 1916 helped discover how C. acetobutylicum culture could be used to produce acetone, butanol, and ethanol from starch using the ABE process (Acetone Butanol Ethanol process) for industrial purposes such as gunpowder and Cordite (using acetone) production. The A.B.E. process was an industry standard until the late 1940s, when low oil costs drove more-efficient processes based on hydrocarbon cracking and petroleum distillation techniques. C. acetobutylicum also produces acetic acid (vinegar), butyric acid (a substance that smells like vomit), carbon dioxide, and hydrogen.\n\nAnaerobic fermentation using C. acetobutylicum recently regained marked interest for use in vehicle biofuel production as a gasoline and diesel fuel replacement. This is because butanol, as produced by a fibrous bed bioreactor utilizing recent biotechnology co-developed by Environmental Energy Inc. and Ohio State University, produces the alcohol butanol as its primary output. The patented process using C. tyrobutyricum produces little acetone or ethanol, instead producing butyric acid and hydrogen, which is then pumped into another fibrous bed bioreactor where C. acetobutylicum converts the butyric acid into butanol, thus optimizing butanol production. The new process, then, obviates the A.B.E. process, making butanol production competitive with other biofuels with regard to both economics and energy production.\n\nPure butanol can be utilized in gasoline-powered cars without any modifications, producing similar mileage performance to gasoline but producing fewer NOx pollutants. If produced from a biomass source, there is no net carbon dioxide production.\n\nUnlike yeast, which can digest sugar only into alcohol and carbon dioxide, C. acetobutylicum and many other Clostridia can digest whey, sugar, starch, lignin, cellulose fiber[dubious  discuss], and other biomass directly into butanol, propionic acid, ether, and glycerin. Apart from the need for temperature control, the A.B.E. synthesis process is relatively simple. The products are formed in layers that are easy to separate.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Biobutanol supporters claim significant advantages over other biofuels used to fuel internal combustion engine vehicles (ICEVs) and other liquid-fueled processes:\n\n    * Butanol has a higher octane fuel value than gasoline with increased low-end torque. A V8 engine has been tested on a 10,000-mile U.S. tour supporting a U.S. Department of Energy grant in 2005. The results of the butanol auto fuel demonstration were presented to the U.S. Department of Energy National Renewable Energy Laboratory's Clean Energy Forum in San Francisco on November 7, 2005.\n    * Butanol does not readily adsorb moisture (it is not hygroscopic), so is less affected by changes in the weather, unlike the combustion of pure ethanol, which requires engine and fuel system modifications.\n\n    * Butanol does not attack materials commonly used in vehicular internal combustion engines.\n    * Biobutanol can also be used in the industrial paint and solvent industry to replace fossil butanol.\n\nJames Liao, a chemical engineer at the University of California, Los Angeles, developed a method to insert genes from Clostridium acetobutylicum which are responsible for production of butanol into the bacterium Escherichia coli.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium perfringens</h1>
<div class="content">
<p>Clostridium perfringens (formerly know as Clostridium welchii) is a Gram-positive, rod-shaped, anaerobic, spore-forming bacterium of the genus Clostridium. C. perfringens is ubiquitous in nature and can be found as a normal component of decaying vegetation, marine sediment, the intestinal tract of humans and other vertebrates, insects, and soil.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
In the United Kingdom and United States they are the third most common cause of food-borne illness, with poorly prepared meat and poultry the main culprits in harboring the bacterium. The Clostridium perfringens enterotoxin (CPE) mediating the disease is often heat-resistant and can be detected in contaminated food and feces.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium tetani</h1>
<div class="content">
<p>Clostridium tetani is a rod-shaped, anaerobic bacterium of the genus Clostridium. Like other Clostridium species, it is Gram-positive, and its appearance on a gram stain resembles tennis rackets or drumsticks. C. tetani is found as spores in soil or as parasites in the gastrointestinal tract of animals. C. tetani produces a potent biological toxin, tetanospasmin, and is the causative agent of tetanus.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Corynebacterium diphtheriae</h1>
<div class="content">
<p>Corynebacterium diphtheriae is a pathogenic bacterium that causes diphtheria. It is also known as the Klebs-Löffler bacillus, because it was discovered in 1884 by German bacteriologists Edwin Klebs (1834  1912) and Friedrich Löffler (1852  1915).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Four subspecies are recognized: C. diphtheriae mitischodis, C. diphtheriae intermedius, C. diphtheriae gravis, and C. diphtheriae belfanti. The four subspecies differ slightly in their colonial morphology and biochemical properties such as the ability to metabolize certain nutrients, but all may be toxigenic (and therefore cause diphtheria) or non-toxigenic.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Coxiella burnetii</h1>
<div class="content">
<p>Coxiella burnetii is a species of intracellular, pathogenic bacteria, and is the causative agent of Q fever. The genus Coxiella is morphologically similar to Rickettsia, but with a variety of genetic and physiological differences. C. burnetii are small Gram negative bacteria with two growth phases, as well as a spore form which lies idle in soil. It can survive standard disinfectants, and is resistant to many other environmental changes like those presented in the phagolysosome.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The United States ended its biological warfare program in 1969. When it did, C. burnetii was one of seven agents it had standardized as biological weapons.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Danio rerio</h1>
<div class="content">
<p>The zebrafish, Danio rerio, is a tropical freshwater fish belonging to the minnow family (Cyprinidae).[1] It is a popular aquarium fish, frequently sold under the trade name zebra danio, and is an important vertebrate model organism in scientific research.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
D. rerio are a common and useful model organism for studies of vertebrate development and gene function. They may supplement higher vertebrate models, such as rats and mice.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Deinococcus radiodurans</h1>
<div class="content">
<p>Deinococcus radiodurans is an extremophilic bacterium, one of the most radioresistant organisms known. It can survive cold, dehydration, vacuum, and acid, and is therefore known as a polyextremophile and has been listed as the world's toughest bacterium in The Guinness Book Of World Records.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
In 2003, U.S. scientists demonstrated that D. radiodurans could be used as a means of information storage that might survive a nuclear catastrophe. They translated the song It's a Small World into a series of DNA segments 150 base pairs long, inserted these into the bacteria, and were able to retrieve them without errors 100 bacterial generations later.\n\nDeinococcus has been Genetically engineered for use in bioremediation to consume and digest solvents and heavy metals, even in a highly radioactive site. For example, the bacterial mercuric reductase gene has been cloned from Escherichia coli into Deinococcus to detoxify the ionic mercury residue frequently found in radioactive waste generated from nuclear weapons manufacture. Those researchers developed a strain of Deinococcus that could detoxify both mercury and toluene in mixed radioactive wastes.\n\nThe Craig Venter Institute has used a system derived from the rapid DNA repair mechanisms of D. radiodurans to assemble synthetic DNA fragments into chromosomes, with the ultimate goal of producing a synthetic organism they call Mycoplasma laboratorium.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Desulfotalea psychrophila</h1>
<div class="content">
<p>Drosophila melanogaster (from the Greek for dark-bellied dew lover) is a two-winged insect that belongs to the Diptera, the order of the flies. The species is commonly known as the common fruit fly or vinegar fly, and is one of the most commonly used model organisms in biology, including studies in genetics, physiology and life history evolution. Flies belonging to the family Tephritidae are also called fruit flies, which can lead to confusion, especially in Australia where the term fruit fly refers to the Tephritidae, an economic pest in fruit production.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Drosophila melanogaster is one of the most studied organisms in biological research, particularly in genetics and developmental biology. There are several reasons:\n\n    * The care and culture requires little equipment and use little space even when using large cultures, and the overall cost is low.\n    * It is small and easy to grow in the laboratory and their morphology is easy to identify once they are anesthetized (usually with ether, carbon dioxide gas, by cooling them, or with products like FlyNap)\n    * It has a short generation time (about 10 days at room temperature) so several generations can be studied within a few weeks.\n    * It has a high fecundity (females can lay more than 800 eggs in a lifetime, i.e. one egg every 30 minutes with sufficient food).\n    * Males and females are readily distinguished and virgin females are easily isolated, facilitating genetic crossing.\n    * The mature larvae show giant chromosomes in the salivary glands called polytene chromosomes"puffs" indicate regions of transcription and hence gene activity.\n    * It has only four pairs of chromosomes: three autosomes, and one sex chromosome.\n    * Males do not show meiotic recombination, facilitating genetic studies.\n    * Recessive lethal "balancer chromosomes" carrying visible genetic markers can be used to keep stocks of lethal alleles in a heterozygous state without recombination due to multiple inversions in the balancer.\n    * Genetic transformation techniques have been available since 1987.\n    * Its complete genome was sequenced and first published in 2000.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Echinops telfairi</h1>
<div class="content">
<p>The Lesser Hedgehog Tenrec (Echinops telfairi) is a species of mammal in the Tenrecidae family. It is endemic to Madagascar. Its natural habitats are subtropical or tropical dry forests, dry savanna, subtropical or tropical dry shrubland, and subtropical or tropical dry lowland grassland.\n\nIt is the only species in the genus Echinops and is named in honour of Charles Telfair.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Enterococcus faecalis</h1>
<div class="content">
<p>Enterococcus faecalis  formerly classified as part of the Group D Streptococcus system  is a Gram-positive commensal bacterium inhabiting the gastrointestinal tracts of humans and other mammals. It is among the main constituents of some probiotic food supplements. A commensal organism like other species in the genus Enterococcus, E. faecalis can cause life-threatening infections in humans, especially in the nosocomial (hospital) environment, where the naturally high levels of antibiotic resistance found in E. faecalis contribute to its pathogenicity.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ashbya gossypii</h1>
<div class="content">
<p>Ashbya gossypii (also known as Eremothecium gossypii[1]) is a filamentous fungus or mold closely related to yeast, but growing exclusively in a filamentous way. It was originally isolated from cotton as a pathogen causing stigmatomycosis by Ashby and Novell in 1926. This disease affects the development of hair cells in cotton bolls and can be transmitted to citrus fruits, which thereupon dry out and collapse (dry rot disease). In the first part of the 20th century, A. gossypii and two other fungi causing stigmatomycosis (Eremothecium coryli, Aureobasidium pullulans) made it virtually impossible to grow cotton in certain regions of the subtropics, causing severe economical losses. Control of the spore transmitting insects - cotton stainer (Dysdercus suturellus) and Antestia - permitted to fully eradicate infections. It was recognized that A. gossypii is a natural overproducer of riboflavin, also known as vitamin B2, which protects its spores against ultraviolet light. This made it an interesting organism for industries, where genetically modified strains are still used to produce this vitamin.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The complete sequencing and annotation of the entire A. gossypii genome, as published in 2004, was initiated when a significant degree of gene synteny was observed in preliminary studies in comparison to the genome of budding yeast, Saccharomyces cerevisiae. This not only helped to improve gene annotation of S. cerevisiae, but also allowed the reconstruction of the evolutionary history of both organisms. A. gossypii and S. cerevisiae originate from a common ancestor which carried about 5000 genes. Divergence of these two close relatives started some 100 million years ago. One branch of evolution involving up to 100 viable genome rearrangements (translocations and inversions), a few million base pair changes, and a limited number of gene deletions, duplications and additions lead to modern A. gossypii with its 4718 protein-coding genes and 9.2 million base pairs (smallest genome of a free-living eukaryote yet characterized) spread over seven chromosomes. The genome of S. cerevisiae underwent a more eventful evolution, which includes a whole-genome duplication.\n\nDespite the long evolutionary history and of the two organisms and fundamentally different ways of growth and development, the complete synteny map of both genomes reveals that 95 % of A. gossypii genes are orthologs of S. cerevisiae genes and 90 % map within blocks of synteny (syntenic homologs).
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli</h1>
<div class="content">
<p>Escherichia coli (commonly E. coli; pronounced /?????r?ki? ?ko?la?/, /i? ~/, and named for its discoverer), is a Gram negative bacterium that is commonly found in the lower intestine of warm-blooded organisms (endotherms). Most E. coli strains are harmless, but some, such as serotype O157:H7, can cause serious food poisoning in humans, and are occasionally responsible for costly product recalls. The harmless strains are part of the normal flora of the gut, and can benefit their hosts by producing vitamin K2, or by preventing the establishment of pathogenic bacteria within the intestine.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
E. coli and related bacteria possess the ability to transfer DNA via bacterial conjugation, transduction or transformation, which allows genetic material to spread horizontally through an existing population. This process led to the spread of the gene encoding shiga toxin from Shigella to E. coli O157:H7, carried by a bacteriophage.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Felix catus</h1>
<div class="content">
<p>The cat (Felis catus), also known as the domestic cat or house cat to distinguish it from other felines and felids, is a small predatory carnivorous species of crepuscular mammal that is valued by humans for its companionship and its ability to hunt vermin, snakes, scorpions, and other unwanted household pests. It has been associated with humans for at least 9,500 years.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
A 2007 study published in the journal Science asserts that all house cats are descended from a group of self-domesticating desert wildcats Felis silvestris lybica circa 10,000 years ago, in the Near East.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Francisella tularensis</h1>
<div class="content">
<p>Francisella tularensis is a pathogenic species of gram-negative bacteria and the causative agent of tularemia or rabbit fever. Due to its ease of spread by aerosol and its high virulence, F. tularensis is classified as a Class A agent by the U.S. government.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Like many other bacteria, F. tularensis undergoes asexual replication. Bacteria will divide into two daughter cells, each of which contains identical genetic information. Genetic variation may be introduced via mutation or horizontal gene transfer.\n\nThe genome of F. tularensis biovar tularensis strain SCHU4 has been sequenced. The studies resulting from the sequencing suggest that a number of gene coding regions in the F. tularensis genome are disrupted by mutations and thus create blocks in a number of metabolic and synthetic pathways that are required for survival. This indicates that F. tularensis has evolved to depend on the host organism for certain nutrients and other processes ordinarily taken care of by these disrupted genes.\n\nThe F. tularensis genome contains unusual transposon-like elements resembling counterparts that normally are found in eukaryotic organisms.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Fusobacterium nucleatum</h1>
<div class="content">
<p>Fusobacterium is a genus of filamentous, anaerobic, Gram-negative bacteria, similar to Bacteroides. Fusobacterium contribute to several human diseases, including periodontal diseases, Lemierre's syndrome, and topical skin ulcers. Although older resources have stated that Fusobacterium is a common occurrence in the human oropharynx, the current consensus is that Fusobacterium should always be treated as a pathogen. \n\nIn contrast to Bacteroides spp., Fusobacteria have a potent LPS.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Gallus gallus</h1>
<div class="content">
<p>The chicken (Gallus gallus domesticus) is a domesticated fowl. Recent evidence suggests that domestication of the chicken was under way in Vietnam over 10,000 years ago. Until this discovery, conventional wisdom held that the chicken was domesticated in India. From India the domesticated fowl made its way to Persia. From the Persianized kingdom of Lydia in western Asia Minor, domestic fowl were imported to Greece perhaps as late as the fifth century BCE. Fowl had been known in Egypt since the 18th Dynasty, the "bird that lays every day" having come to Egypt, according to the annals of Tutmose III, as tribute from from the land between Syria and Shinar, that is Babylonia. Fowl make no appearance in the Old Testament.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
[C]ontrary to what one may hear from the industry, chickens are  complex behaviorally, do quite well in learning, show a rich social organization, and have a diverse repertoire of calls. Anyone who has kept barnyard chickens recognizes their significant differences in personality.  Bernard Rollin, Farm Animal Welfare: Social, Bioethical, and Research Issues, Iowa State University Press: Ames, Iowa, 1995: 118.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Haemophilus ducreyi</h1>
<div class="content">
<p>Haemophilus ducreyi is a fastidious Gram-negative coccobacillus causing the sexually transmitted disease chancroid characterized by painful sores on the genitalia. Another early symptom is dark or light green shears in excrement.\n\nHaemophilus ducreyi is responsible for chancroid, a major cause of genital ulceration in developing countries. It starts as an erythematous papular lesion which breaks down into a painful bleeding ulcer with a necrotic base and ragged edge.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Haemophilus influenzae</h1>
<div class="content">
<p>Haemophilus influenzae, formerly called Pfeiffer's bacillus or Bacillus influenzae, is a non-motile Gram-negative coccobacillus first described in 1892 by Richard Pfeiffer during an influenza pandemic. A member of the Pasteurellaceae family, it is generally aerobic, but can grow as a facultative anaerobe.[1] H. influenzae was mistakenly considered to be the cause of influenza until 1933, when the viral etiology of the flu became apparent. Still, H. influenzae is responsible for a wide range of clinical diseases.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
H. influenzae was the first free-living organism to have its entire genome sequenced. Haemophilus was chosen because one of the project leaders, Nobel laureate Hamilton Smith, had been working on it for decades and was able to provide high-quality DNA libraries. The genome consists of 1,830,140 base pairs of DNA in a single circular chromosome that contains 1740 protein-coding genes, 58 transfer RNA genes tRNA, and 18 other RNA genes.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Helicobacter hepaticus</h1>
<div class="content">
<p>Helicobacter is a genus of Gram-negative bacteria possessing a characteristic helix shape. They were initially considered to be members of the Campylobacter genus, but since 1989 they have been grouped in their own genus.\n\nSome species have been found living in the lining of the upper gastrointestinal tract, as well as the liver of mammals and some birds.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Helicobacter pylori</h1>
<div class="content">
<p>Helicobacter is a genus of Gram-negative bacteria possessing a characteristic helix shape. They were initially considered to be members of the Campylobacter genus, but since 1989 they have been grouped in their own genus.\n\nThe most widely known species of the genus is H. pylori which infects up to 50% of the human population. Some strains of this bacterium are pathogenic to humans as it is strongly associated with peptic ulcers, chronic gastritis, duodenitis, and stomach cancer. It also serves as the type species of the genus.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Homo sapiens</h1>
<div class="content">
<p>Humans are a eukaryotic species. Each diploid cell has two sets of 23 chromosomes, each set received from one parent. There are 22 pairs of autosomes and one pair of sex chromosomes. By present estimates, humans have approximately 20,00025,000 genes. Like other mammals, humans have an XY sex-determination system, so that females have the sex chromosomes XX and males have XY.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Humans are considered a dangerous predator and should be approached cautiously, like any other of the great apes.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Kluyveromyces lactis</h1>
<div class="content">
<p>Kluyveromyces lactis is a Kluyveromyces yeast commonly used for genetic studies and industrial applications. Its name comes from the ability to assimilate lactose and convert it into lactic acid.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Besides E.coli and Aspergillus niger var awamori, K.lactis is one of the main organisms grown in industry in fermenters to produce chymosin on a large scale. Chymosin is used for cheese production.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactobacillus acidophilus</h1>
<div class="content">
<p>Lactobacillus acidophilus (meaning acid-loving milk-bacterium) is a species in the genus Lactobacillus. L. acidophilus is a homo-fermentative species, fermenting sugars into lactic acid, which grows readily at rather low pH values (below pH 5.0) and has an optimum growth temperature of 30 °C (86 °F)[citation needed]. L. acidophilus occurs naturally in the human and animal gastrointestinal tract, mouth, and vagina.[1]. Some strains of L. acidophilus may be considered to have probiotic characteristics [2]. These strains are commercially used in many dairy products, sometimes together with S. salivarius ssp. thermophilus and Lactobacillus delbrueckii ssp. bulgaricus in the production of acidophilus-type yogurt.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
L. acidophilus is part of the normal vaginal flora. The acid produced by L. acidophilus in the vagina may help to control the growth of the fungus Candida albicans, thus helping to prevent vaginal yeast infections. The same beneficial effect has been observed in cases of oral or gastrointestinal Candidiasis infections. Certain spermicides and contraceptive creams can kill L. acidophilus in the vagina, clearing the path to possible yeast infections.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactobacillus plantarum</h1>
<div class="content">
<p>Lactobacillus plantarum is a widespread member of the genus Lactobacillus, commonly found in many fermented food products as well as anaerobic plant matter. It is also present in saliva (from which it was first isolated). It has the ability to liquefy gelatin. L. plantarum has one of the largest genomes known among the lactic acid bacteria and is a very flexible and versatile species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactococcus lactis</h1>
<div class="content">
<p>Lactococcus lactis is a Gram-positive bacteria used extensively in the production of buttermilk and cheese.  L. lactis are cocci that group in pairs and short chains, and depending on growth conditions appears ovoid with typically 0.5 - 1.5 µm in length. L. lactis do not produce spores (non-sporulating) and are not motile (non-motile). They have a homo-fermentative metabolism and have been reported to produce exclusively L(+) lactic acid. However, reported that D(-) lactic acid can be produced when cultured at low pH. The capability to produce lactic acid is one of the reasons why Lactococcus lactis is one of the most important micro-organisms involved in the dairy industry. Generally, it has been considered as an opportunistic pathogen. Even though, the number of clinical cases associated with infections by these microorganisms has increased in the last decade in both humans and animals. L. lactis is a bacterium which has a crucial importance for manufacturing dairy products such as buttermilk and cheeses. When L. lactis ssp. lactis is added to milk, the bacterium uses enzymes to produce energy molecules, called ATP, from lactose.The byproduct of ATP energy production is lactic acid. The lactic acid produced by the bacterium curdles the milk that then separates to form curds, which are used to produce cheese and whey.\n\nOther uses that have been reported for this bacteria include the production of pickled vegetables, beer or wine, some breads and other fermented food-stuffs such as soymilk kefir, buttermilk, .... Nowadays, researchers believe that understanding the physiology and genetic make-up of this bacterium will provide food manufacturers as well as the pharmaceutical industry with invaluable benefits \n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
L. lactis subsp. lactis (formerly Streptococcus lactis) is used in the early stages for the production of many cheeses including Brie, Camembert cheese, cheddar, Colby, Gruyère, Parmesan, and Roquefort.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Legionella pneumophila</h1>
<div class="content">
<p>Legionella pneumophila is a thin, pleomorphic, flagellated Gram-negative bacterium of the genus Legionella.[1][2] L. pneumophila is the primary human pathogen in this group and is the causative agent of legionellosis or Legionnaires' disease.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Leifsonia xyli</h1>
<div class="content">
<p>Leifsonia xyli subsp. xyli is a small, fastidious, Gram-positive, coryneform bacterium that causes ratoon stunting disease, a major worldwide disease of sugarcane.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The bacterium has never been found affecting wild clones of Saccharum officinarum in its centre of diversity, Papua New Guinea, which suggests either it has recently evolved to infect its host, or has been acquired from a related species. It is believed to have evolved from a single pathogenic clone, since no genetic variation was found among different isolates of distinct countries and cultivars
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Leptospira interrogans serovar Copenhageni</h1>
<div class="content">
<p>Leptospira (from the Greek leptos, meaning fine or thin, and the Latin spira, meaning coil) is a genus of spirochaete bacteria, including a small number of pathogenic and saprophytic species. Leptospira was first observed in 1907 in kidney tissue slices of a leptospirosis victim who was described as having died of "yellow fever."</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The genome of pathogenic Leptospira consists of two chromosomes. The size of the genomes of L. interrogans serovars Copenhageni and Lai is approximately 4.6 Mb.[35][36] However, the genome of L. borgpetersenii serovar Hardjo is only 3.9 Mb in size with a large number of pseudogenes, gene fragments, and insertion sequences relative to the genomes of L. interrogans. L. borgpetersenii serovar Hardjo is usually transmitted by direct exposure to infected tissues, whereas L. interrogans is often acquired from water or soil contaminated by the urine of carrier animals harboring Leptospira in their kidneys. The high number of defective genes and insertion sequences in L. borgpetersenii Hardjo together with the poor survival outside of the host and difference in transmission patterns compared to L. interrogans suggest that L. borgpetersenii is undergoing insertion-sequence mediated genomic decay, with ongoing loss of genes necessary for survival outside of the host animal.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Listeria monocytogenes</h1>
<div class="content">
<p>Listeria monocytogenes is the causative agent of Listeriosis. It is one of the most virulent foodborne pathogens with 20 to 30 percent of clinical infections resulting in death . Responsible for approximately 2,500 illnesses and 500 deaths in the United States (U.S.) annually, Listeriosis is the leading cause of death among foodborne bacterial pathogens with fatality rates exceeding even Salmonella and Clostridium botulinum. \n\nL. monocytogenes is a Gram-positive bacterium, in the division Firmicutes, named for Joseph Lister. Motile via flagella at 30°C and below but usually not at 37°C, L. monocytogenes can instead move within eukaryotic cells by explosive polymerization of actin filaments (known as comet tails or actin rockets). Studies suggest that up to 10% of human gastrointestinal tracts may be colonized by L. monocytogenes. Nevertheless, clinical diseases due to L. monocytogenes are more frequently recognized by veterinarians, especially as meningo-encephalitis in ruminants. See: listeriosis in animals. Due to its frequent pathogenicity causing meningitis in newborns (acquired transvaginally), pregnant mothers are often advised not to eat soft cheeses such as Brie, Camembert, feta and queso blanco fresco, which may be contaminated with and permit growth of L. monocytogenes. It is the third most common cause of meningitis in newborns. More recently, L. monocytogenes has been used as the model organism to illustrate the Patho-biotechnology concept.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Loxodonta africana</h1>
<div class="content">
<p>The African Bush Elephant (Loxodonta africana) is the larger of the two species of African elephant. Both it and the African Forest Elephant have usually been classified as a single species, known simply as the African Elephant. Some authorities still consider the currently available evidence insufficient for splitting the African Elephant into two species.[3] It is also known as the Bush Elephant or Savanna Elephant.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The African Bush Elephant is an intelligent animal. Experiments with reasoning and learning show that they are the smartest ungulates together with their Asian cousins. This is mostly due to their large brain.\n\nHerds are made up of related females and their young, directed by the eldest female, called the matriarch. Infrequently, an adult male goes with them, but those usually leave the pack when reaching adolescence to form herds with other elephants of the same age. Later, they spread out, carrying out a lonely life, approaching the female herds only during the mating season. Nevertheless, elephants do not get too far from their families and recognize them when reencountered. Sometimes, several female herds can blend for a period of time, reaching even hundreds of individuals.\n\nThe matriarch is the one who decides the route and shows to each other member of the herd all the water sources she knows, which the rest will memorize in the future. The relations among the members of the herd is very tight; when a female gives birth to a baby the rest go to acknowledge it touching her with the trunk; and when an old elephant dies the rest of the herd will stay by the corpse for a while. The famous elephant graveyards are a myth, but it is true that these animals can recognize a carcass of its species when they find one during their trips, and even if it is a stranger, they form around it and sometimes they even touch its forehead with their trunk.\n\nMating happens when the female becomes receptive, an event that can occur anytime during the year. When she is ready, she starts emitting infrasounds that attract the males, sometimes from kilometers away. The adult males start arriving at the herd during the following days and begin fighting, causing some injuries and even broken tusks. The female shows her acceptance of the victor by rubbing her body against his. They mate, and then both go their own way. After 22 months of gestation (the longest among mammals), the female gives birth to a single 90 cm high calf which weighs more than 100 kg. The baby feeds on the mothers milk until the age of 5, but also eats solid food from as early as 6 months old. Just a few days after birth, the calf can follow the herd by foot.\n\nSome African Bush Elephants will attack and kill rhinoceroses. This behavior, when it occurs, is mostly observed with younger adult male elephants who have come into musth prematurely.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Macaca mulatta</h1>
<div class="content">
<p>The Rhesus Macaque (Macaca mulatta), often called the Rhesus Monkey, is one of the best known species of Old World monkeys.\n\nAdult males measure approximately 53 centimeters on average and weigh an average of 7.7 kilograms. Females are smaller, averaging 47 centimeters in length and 5.3 kilograms in weight. This macaque is brown or grey in color and have pink faces which are typically bereft of fur. Its tail is of medium length and averages between 20.7 and 22.9 centimeters. It typically has a lifespan of about 25 years.\n\nThe species is native to northern India, Bangladesh, Pakistan, Burma, Thailand, Afghanistan, southern China, and some neighboring areas.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
here is also a colony that is well established and thriving in Florida, USA. This population gained a foothold niche in the Silver River State Park as a result of being released by a tour boat operator known locally as "Colonel Tooey" to enhance his "Jungle Cruise" ride some time around the Spring of 1938. A traditional story that the monkeys were released for scenery enhancement in the Tarzan movies that were filmed at that location is false, as the only Tarzan movie filmed in the area, 1939's Tarzan Finds a Son! contains no Rhesus Macaques. In addition, various colonies of rhesus and other monkey species are speculated to be the result of zoos and wildlife parks destroyed in hurricanes, most notably Hurricane Andrew.\n\nThere is a notable colony of Rhesus Monkeys on Morgan Island in Lowcountry South Carolina. They were imported in the 1970s for use in the local labs and are by all accounts thriving
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methanopyrus kandleri</h1>
<div class="content">
<p>Methanopyrus kandleri is a a rod-shaped organism that grows optimally at temperatures near and above the boiling point of water.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
M. kandleri appears to have fewer genes acquired via lateral transfer than other archaea. This feature might reflect the fact that the extreme habitat of this organism doesn't provide the opportunities for lateral gene transfer.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methanosarcina acetivorans</h1>
<div class="content">
<p>Methanosarcina acetivorans is a versatile methane producing microbe which is found in such diverse environments as oil wells, trash dumps, deep sea hydrothermal vents, and oxygen depleted sediments beneath kelp beds. Only M. acetivorans and microbes in the genus Methanosarcina use all three known metabolic pathways for methanogenesis.  Methanosarcinides, including M. acetivorans, are also the only archaea capable of forming multicellular colonies, and even show cellular differentiation. As of 2006[update], the genome of M. acetivorans is the largest of all sequenced archaeal genomes.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methylococcus capsulatus</h1>
<div class="content">
<p>Methylococcus capsulatus is an obligately methanotrophic gram-negative bacterium. In addition to methane, M. capsulatus is able to oxidize some organic hydrogen containing compounds. In addition, M. capsulatus can live in conditions in which there is little oxygen available.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Monodelphis domestica</h1>
<div class="content">
<p>The Gray Short-tailed Opossum (Monodelphis domestica) is a small member of the Didelphidae family of opossums. It was the first marsupial to have its genome sequenced. It is naturally found in arboreal habitats in Bolivia, Brazil and Paraguay. The opossum is used as research model in science, and is also frequently found in the exotic pet trade. It is also known as the Brazilian Opossum, Rainforest Opossum and in a research setting the Laboratory Opossum.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The gray short-tailed opossum possesses several features that make in ideal research model particularly in studies of marsupials, as well as the immunological and developmental research on mammalian systems. It breeds relatively easily in laboratory settings and neonates are exposed and can be readily accessed because, unlike other marsupial species, females opossums lack a pouch: neonates simply cling to the teats. Opossums are born at a stage that is approximately equivalent to 13-15-day old fetal rats or 40-day old human embryos. Like other marsupials, the inadequacies of the neonates immune system function make it an ideal model for both transplant and cancer research, as well as general investigations into immune system development. Its genome was sequenced and a working draft published in May 2007: the decoding work, directed by MIT and Harvard, reveals the opossum to have between 18,000 and 20,000 protein-coding genes.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mus musculus</h1>
<div class="content">
<p>The House Mouse (Mus musculus) is one of the most numerous species of the genus Mus commonly termed a mouse. It is a small mammal and a rodent.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Laboratory mice belong to strains of House Mice and are some of the most important model organisms in biology and medicine; they are by far the most commonly used genetically altered laboratory mammal.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium</h1>
<div class="content">
<p>Mycobacterium is a genus of Actinobacteria, given its own family, the Mycobacteriaceae. The genus includes pathogens known to cause serious diseases in mammals, including tuberculosis and leprosy. The Latin prefix "myco" means both fungus and wax; its use here relates to the "waxy" compounds in the cell wall.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycoplasma</h1>
<div class="content">
<p>Mycoplasma is a genus of bacteria which lack a cell wall. Without a cell wall, they are unaffected by many common antibiotics such as penicillin or other beta-lactam antibiotics that target cell wall synthesis. They can be parasitic or saprotrophic. Several species are pathogenic in humans, including M. pneumoniae, which is an important cause of atypical pneumonia and other respiratory disorders, and M. genitalium, which is believed to be involved in pelvic inflammatory diseases.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nanoarchaeum equitans</h1>
<div class="content">
<p>Nanoarchaeum equitans is a species of tiny microbe, discovered in 2002 in a hydrothermal vent off the coast of Iceland by Karl Stetter. Since it grows in temperatures approaching boiling, it is considered to be a thermophile. Nanoarchaeum appears to be an obligatory symbiont on the archaeon Ignicoccus; it must be in contact with the host organism to survive. Its cells are only 400 nm in diameter, making it the next smallest known living organism, excepting possibly nanobacteria and nanobes, whose status as living organisms are controversial. Its genome is only 490,885 nucleotides long; the smallest non-viral genome ever sequenced next to C. ruddii's in 2006.\n\nGenetically, Nanoarchaeum is peculiar in that its 16S rRNA sequence is undetectable by the most common methods. Initial examination of single-stranded ribosomal RNA indicated that the organism most likely belonged to the Archaea domain. However, its difference from the existing phyla, Euryarchaeota and Crenarchaeota, was as great as the difference between the phyla. Therefore, it was given its own phylum, called Nanoarchaeota. However, another group (see References) compared all of the open reading frames to the other Archaea. They argue that the initial sample, ribosomal RNA only, was biased and Nanoarchaeum actually belongs to the Euryarchaeota phylum.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The sequencing of the Nanoarchaeum genome has revealed a wealth of information about the organism's biology. The genes for several vital metabolic pathways appear to be missing. Nanoarchaeum cannot synthesize most nucleotides, amino acids, lipids, and cofactors. The cell most likely obtains these biomolecules from Ignicoccus. However, unlike many parasitic microbes, Nanoarchaeum has many DNA repair enzymes, as well as everything necessary to carry out DNA replication, transcription, and translation. This may explain why the genome lacks the large stretches of non-coding DNA characteristic of other parasites. The organism's ability to produce its own ATP is also in question. Nanoarchaeum lacks the ability to metabolize hydrogen and sulfur for energy, as many thermophiles do. It does have five subunits of an ATP synthase as well as pathways for oxidative deamination. Whether it obtains energy from biological molecules imported from Ignicoccus, or whether it receives ATP directly is currently unknown. The genome and proteome composition of N. equitans are marked with the signatures of dual adaptation  one to high temperature and the other to obligatory parasitism (or symbiosis).
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Neisseria gonorrhoeae</h1>
<div class="content">
<p>Neisseria gonorrhoeae, also known as Gonococci (plural), or Gonococcus (singular), is a species of Gram-negative kidney bean-shaped diplococci bacteria responsible for the sexually transmitted disease gonorrhoea.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Neisseria are fastidious cocci, requiring nutrient supplementation to grow in laboratory cultures. Specifically, they grow on Chocolate agar with carbon dioxide. These cocci are facultatively intracellular and typically appear in pairs (diplococci).\n\nNeisseria is usually isolated on Thayer-Martin agaran agar plate with three different antibiotics and nutrients which not only facilitate the growth of Neisseria species, but inhibit the growth of Gram-positive organisms and most bacilli and fungi. Further testing to differentiate the species includes testing for oxidase (all Neisseria show a positive reaction) and the carbohydrates maltose, sucrose, and glucose test in which N. gonorrhoeae will only oxidize (that is, utilize) the glucose.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Neisseria meningitidis</h1>
<div class="content">
<p>Neisseria meningitidis is a heterotrophic gram-negative diplococcal bacterium best known for its role in meningitis and other forms of meningococcal disease such as meningococcemia. N. meningitidis is a major cause of morbidity and mortality in childhood in industrialized countries and is responsible for epidemics in Africa and in Asia.\n\nAnton Weichselbaum in 1887 first discovered the disease in patients infected with meningococci.\n\nMeningococci only infect humans and have never been isolated from animals because the bacterium cannot get iron other than from human sources (transferrin and lactoferrin).\n\nIt exists as normal flora in the nasopharynx of up to 40% of adults. It causes the only form of bacterial meningitis known to cause epidemics.\n\nMeningococcus is spread through the exchange of saliva and other respiratory secretions during activities like coughing, kissing, and chewing on toys. Though it initially produces with general symptoms like fatigue, it can rapidly progress from fever, headache and neck stiffness to coma and death. Death occurs in approximately 10% of cases.[4] Those with impaired immunity may be at particular risk of meningococcus (e.g. those with nephrotic syndrome or splenectomy; vaccines are given in cases of removed or non-functioning spleens).\n\nSuspicion of meningitis is a medical emergency and immediate medical assessment is recommended. Current guidance in the United Kingdom is that any doctor who suspects a case of meningococcal meningitis or septicaemia (infection of the blood) should give intravenous antibiotics (benzylpenicillin or Cefotaxime) and admit the ill person to the hospital. This means that laboratory tests may be less likely to confirm the presence of Neisseria meningitidis as the antibiotics will dramatically lower the number of bacteria in the body. The UK guidance is based on the idea that the reduced ability to identify the bacteria is outweighed by reduced chance of death.\n\nSepticaemia caused by Neisseria meningitidis has received much less public attention than meningococcal meningitis even though septicaemia has been linked to infant deaths. [6] Meningococcal septicaemia typically causes a purpuric rash that does not lose its colour when pressed with a glass ("non-blanching") and does not cause the classical symptoms of meningitis. This means the condition may be ignored by those not aware of the significance of the rash. Septicaemia carries an approximate 50% mortality rate over a few hours from initial onset. Many health organizations advise anyone with a non-blanching rash to go to a hospital emergency room as soon as possible.[citation needed] Note that not all cases of a purpura-like rash are due to meningococcal septicaemia; however, other possible causes need prompt investigation as well (e.g. ITP a platelet disorder or Henoch-Schönlein purpura).\n\nWaterhouse-Friderichsen syndrome is a massive, usually bilateral, hemorrhage into the adrenal glands caused by fulminant meningococcemia.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Neisseria meningitidis</h1>
<div class="content">
<p>Neisseria meningitidis is a heterotrophic gram-negative diplococcal bacterium best known for its role in meningitis and other forms of meningococcal disease such as meningococcemia. N. meningitidis is a major cause of morbidity and mortality in childhood in industrialized countries and is responsible for epidemics in Africa and in Asia.\n\nAnton Weichselbaum in 1887 first discovered the disease in patients infected with meningococci.\n\nMeningococci only infect humans and have never been isolated from animals because the bacterium cannot get iron other than from human sources (transferrin and lactoferrin).\n\nIt exists as normal flora in the nasopharynx of up to 40% of adults. It causes the only form of bacterial meningitis known to cause epidemics.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Meningococcus is spread through the exchange of saliva and other respiratory secretions during activities like coughing, kissing, and chewing on toys. Though it initially produces with general symptoms like fatigue, it can rapidly progress from fever, headache and neck stiffness to coma and death. Death occurs in approximately 10% of cases. Those with impaired immunity may be at particular risk of meningococcus (e.g. those with nephrotic syndrome or splenectomy; vaccines are given in cases of removed or non-functioning spleens).\n\nSuspicion of meningitis is a medical emergency and immediate medical assessment is recommended. Current guidance in the United Kingdom is that any doctor who suspects a case of meningococcal meningitis or septicaemia (infection of the blood) should give intravenous antibiotics (benzylpenicillin or Cefotaxime) and admit the ill person to the hospital. This means that laboratory tests may be less likely to confirm the presence of Neisseria meningitidis as the antibiotics will dramatically lower the number of bacteria in the body. The UK guidance is based on the idea that the reduced ability to identify the bacteria is outweighed by reduced chance of death.\n\nSepticaemia caused by Neisseria meningitidis has received much less public attention than meningococcal meningitis even though septicaemia has been linked to infant deaths. Meningococcal septicaemia typically causes a purpuric rash that does not lose its colour when pressed with a glass ("non-blanching") and does not cause the classical symptoms of meningitis. This means the condition may be ignored by those not aware of the significance of the rash. Septicaemia carries an approximate 50% mortality rate over a few hours from initial onset. Many health organizations advise anyone with a non-blanching rash to go to a hospital emergency room as soon as possible.[citation needed] Note that not all cases of a purpura-like rash are due to meningococcal septicaemia; however, other possible causes need prompt investigation as well (e.g. ITP a platelet disorder or Henoch-Schönlein purpura).\n\nWaterhouse-Friderichsen syndrome is a massive, usually bilateral, hemorrhage into the adrenal glands caused by fulminant meningococcemia.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nematostella vectensis</h1>
<div class="content">
<p>The starlet sea anemone (Nematostella vectensis) is a species of sea anemone native to the east coast of the United States, with introduced populations along the coast of southeast England and west coast of the United States. Populations have also been located in Nova Scotia, Canada. The starlet sea anemone has a bulbous basal end and a contracting column (usually less than two cm but no more than six cm) in order to burrow into the mud. At the top of the column is an oral disk containing the mouth surrounded by two rings of tentacles - typically 16 but up to 20. Although the general color is a greyish white, recent food consumption may temporarily affect pigmentation.\n\nN. vectensis is the first cnidarian with a completely sequenced genome.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nitrosococcus oceani</h1>
<div class="content">
<p>Nitrosococcus oceani is is a marine, gram-negative, chemolithoautotrophic, ammonia-oxidizing bacterium.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nitrosomonas europaea</h1>
<div class="content">
<p>Nitrosomonas europaea is a Gram-negative obligate chemolithoautotroph that can derive all its energy and reductant for growth from the oxidation of ammonia to nitrite and lives in several places such as soil, sewage, freshwater, the walls of buildings and on the surface of monuments especially in polluted areas where the air contains high levels of nitrogen compounds.\n\nBecause of the large amounts of ammonia this bacterium needs to consume for energy to divide, cell division can take up to several days. This is perhaps one reason why this microbe is not studied very much. This microbe has been shown to be an ammonia-oxidizing soil bacterium and it is known to have a range of substrates that might be useful in bioremediation. Several studies are still being done with the bacterium, but will take some time due to the slow cell division rate and the high amounts of nitrogen needed to live.\n\nWhile not using photosynthesis for energy is not unique, "burning" ammonia with oxygen is. Both are characteristics of Nitrosomonas europaea. This microbe tolerates a pH of 6.0-9.0, the optimal conditions being slightly basic; has an aerobic metabolism; and prefers a temperature between 20-30 degrees Celsius. Most are mobile with flagella located in the polar regions although some species are nonmobile.\n\nThe reaction catalysed by these bacteria is the first step in the oxidation of ammonia to nitrate. Nitrosomonas europaea are also important in the treatment of industrial and sewage waste in the first step of oxidizing ammonia to nitrate. Evidence suggests that ammonia-oxidizing bacteria contribute significantly to the global production of nitrous oxide (produced by the reduction of nitrite). Nitrosomonas europaea is also capable of degrading a variety of halogenated organic compounds, including trichloroethylene, benzene, and vinyl chloride. The ability of nitrifying organisms to degrade some pollutants may make these organisms attractive for controlled bioremediation in nitrifying soils and waters.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Nitrosomonas europaea is also accountable for nitric acid production which can cause the dissolvence of some stone and other construction materials found on statues and in buildings.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Oceanobacillus iheyensis</h1>
<div class="content">
<p>Oceanobacillus iheyensis HTE831 is an extremely halotolerant and alkaliphilic Bacillus related species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ochotona princeps</h1>
<div class="content">
<p>The American Pika (Ochotona princeps) is a diurnal species of pika that is found in the mountains of western North America, usually in boulder fields at or above tree line. They are flower-gathering animals, which are the smallest of the rabbit group.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The US government is currently considering the addition of the American pika under the US Endangered Species Act, in the IUCN Red List it is still considered a Species of Least Concern.\n\nRecent studies suggest that species populations are declining due to various factors, most notably global Warming. A 2003 study, published in the Journal of Mammology, showed that 9 out of 25 sampled populations of American Pika had disappeared, causing biologists to conclude that the species is reaching extinction.\n\nAs they live in the high and cooler mountain regions, they are very sensitive to high temperatures, and are considered to be one of the best early warning systems for detecting global warming in the western United States. Because their regular habitat's temperature rises, the American Pikas move higher up the mountain. Scientists report that pikas can die within an hour if the outside temperature reaches above 23°C (75°F).
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ornithorhynchus anatinus</h1>
<div class="content">
<p>The Platypus (Ornithorhynchus anatinus) is a semi-aquatic mammal endemic to eastern Australia, including Tasmania. Together with the four species of echidna, it is one of the five extant species of monotremes, the only mammals that lay eggs instead of giving birth to live young. It is the sole living representative of its family (Ornithorhynchidae) and genus (Ornithorhynchus), though a number of related species have been found in the fossil record.\n\nThe bizarre appearance of this egg-laying, venomous, duck-billed, beaver-tailed, otter-footed mammal baffled European naturalists when they first encountered it, with some considering it an elaborate fraud. It is one of the few venomous mammals; the male Platypus has a spur on the hind foot that delivers a venom capable of causing severe pain to humans. The unique features of the Platypus make it an important subject in the study of evolutionary biology and a recognizable and iconic symbol of Australia; it has appeared as a mascot at national events and is featured on the reverse of the Australian 20 cent coin. The platypus is the animal emblem of the state of New South Wales.\n\nUntil the early 20th century it was hunted for its fur, but it is now protected throughout its range. Although captive breeding programs have had only limited success and the Platypus is vulnerable to the effects of pollution, it is not under any immediate threat.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The Platypus is sometimes jokingly referred to as proof that God has a sense of humour (at the beginning of the film Dogma, for example). Its unusual appearance has led to its featuring in many media, particularly in its native Australia.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Oryctolagus cuniculus</h1>
<div class="content">
<p>The European Rabbit (Oryctolagus cuniculus) is a species of rabbit native to south west Europe (Spain and Portugal). It has been widely introduced elsewhere often with devastating effects on local biodiversity. However, its decline in its native range (caused by the diseases myxomatosis and rabbit calicivirus as well as over-hunting and habitat loss) has caused the decline of its highly dependent predators, the Iberian Lynx and the Spanish Imperial Eagle. It is known as an invasive species for the fact it has been introduced to countries on all continents with the exception of Antarctica and Asia, and caused many problems within the environment and ecosystems as well. Australia has the most problems with European Rabbits, since there is lack of predators to these animals there.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Rabbits are famed for their reproductive capabilities. Although certainly not the strongest, fastest, or smartest of the mammals, they have carved out a strong ecological niche through their rate of impregnation, due to the fact that female rabbits ovulate at the time of copulation. One striking example of rapid rabbit reproduction took place in Australia, where the 24 rabbits first introduced in 1859 had multiplied in number to over 600 million over the course of less than a century.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Oryza sativa</h1>
<div class="content">
<p>Oryza sativa (common names include Asian rice) is the plant species known in English as rice. Oryza sativa has the smallest cereal genome consisting of just 430Mb across 12 chromosomes. It is renowned for being easy to genetically modify and is a model organism for cereal biology.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Oryza sativa contains two major subspecies: the sticky, short grained japonica or sinica variety, and the non-sticky, long-grained indica variety. Japonica are usually cultivated in dry fields, in temperate East Asia, upland areas of Southeast Asia and high elevations in South Asia, while indica are mainly lowland rices, grown mostly submerged, throughout tropical Asia. Rice is known to come in a variety of colors, including: white, brown, black, purple, and red.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Oryzias latipes</h1>
<div class="content">
<p>Oryzias latipes, also known as Medaka and Japanese killifish, is a member of genus Oryzias (ricefish), the only genus in the family Oryziinae. This rather small (2-4 cm long) native of Southeast Asia is a common denizen of rice paddies in coastal Asia. It became popular as an aquarium fish because of its hardiness and pleasant coloration: its coloration varies from brown or yellow-gold in the wild to white, creamy yellow, or orange in aquarium-bred individuals. The medaka has been a popular pet since the 17th century in Japan. It is amphidromous, meaning it moves between salt and freshwater at some point in its life. It is found in both ocean and river habitats. The female carries her eggs attached between her anal fins.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Oryzias latipes is a model organism and is extensively used in many areas of biological research. Medakas have short generation time and are reproductively prolific - characteristics that make them easy to rear in the laboratory. They can withstand cold and can be shipped easily. Nearly all aspects of the life cycle of the medaka have been analyzed by researchers including sexual behavior, genetic inheritance of coloration, spawning habits, feeding, pathology, embryological development, ecology, etc.\n\nTransgenic medaka are relatively easy to produce. The fish has been genetically modified to secrete various human hormones, express promoter sequences from other fish, and make antimicrobial proteins and a protein that makes the medaka glow fluorescent green. There are also many mutations that show up in medaka at random, for example, a mutant strain that lacks scales, and one with extra-long fins.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Otolemur garnettii</h1>
<div class="content">
<p>The Northern Greater Galago (Otolemur garnettii), also known as Garnett's Greater Galago, is a prosimian primate endemic to Africa.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
A low-coverage genomic sequence of the Northern Greater Galago, was completed in 2006. As a 'primitive' primate, the sequence is particularly useful in bridging the sequences of higher primates (macaque, chimp, human) to close non-primates such as rodents. The current 2x coverage is not sufficient to create a full genome assembly, but will provide comparative data across most of the human assembly.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pan troglodytes</h1>
<div class="content">
<p>The Common Chimpanzee (Pan troglodytes), also known as the Robust Chimpanzee, is a great ape. The name troglodytes, Greek for 'cave-dweller', was coined by Johann Friedrich Blumenbach in his Handbuch der Naturgeschichte (Handbook of Natural History) published in 1779. Colloquially, it is often called the chimpanzee (or simply 'chimp'), though technically this term refers to both species in the genus Pan: the Common Chimpanzee and the closely-related Bonobo, or Pygmy Chimpanzee.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Recent DNA evidence suggests the Bonobo and Common Chimpanzee species separated from each other less than one million years ago (similar in relation between Homo sapiens and Neanderthals). The chimpanzee line split from the last common ancestor of the human line approximately six million years ago. Because no species other than Homo sapiens has survived from the human line of that branching, both chimpanzee species are the closest living relatives of humans. The chimpanzee's genus, Pan, diverged from the Gorilla's genus about 7 million years ago.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Parachlamydia sp UWE25</h1>
<div class="content">
<p>Bacteria belonging to the Chlamydiales group are obligate intracellular parasites of eukaryotic cells. They are found within vertebrates, invertebrate cells, and amoebae hosts. In 1999 this group was taxonomically reclassified and it now includes a broader range of Gram-negative bacteria based on their similarities with regards to their nucleotide sequences (predominantly ribosomal RNA) and their two-stage developmental cycle of replication (Everett et al., 1999). The Chlamydiales group is now divided into four families, Chlamydiaceae, Simkaniaceae, Parachlamydiaceae and Waddliaceae.\n\nAn endosymbiont of free-living amoebae (Acanthamoeba sp. endosymbiont UWE25) and implicated as a potential human pathogen this strain is an environmental isolate. Its genome size is approximately twice as large as other pathogenic species belonging to the Chlamydiaceae family.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pasteurella multocida</h1>
<div class="content">
<p>Pasteurella multocida is a Gram-negative, non-motile coccobacillus that is penicillin-sensitive and belongs to the Pasteurellaceae family. It can cause a zoonotic infection in humans, which typically is a result of bites or scratches from domestic pets. Many mammals and fowl harbor it as part of their normal respiratory microbiota, displaying asymptomatic colonization.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
A bacteriophage encodes the toxin responsible for most P. multocida virulence factors. This toxin activates Rho GTPases, which bind and hydrolyze GTP, and are important in actin stress fiber formation. Formation of stress fibers may aid in the endocytosis of P. multocida. The host cell cycle is also modulated by the toxin, which can act as an intracellular mitogen.\nP. multocida will grow at 37 degrees Celsius on blood or chocolate agar, but will not grow on MacConkey agar. Colony growth is accompanied by a characteristic "mousy" odor due to metabolic products.\nBeing a facultative anaerobe, it is oxidase- and catalase-positive, and can also ferment a large number carbohydrates in anaerobic conditions.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Photobacterium profundum SS9</h1>
<div class="content">
<p>Photobacterium profundum is a deep sea Gammaproteobacterium, belonging to the family Vibrionaceae and genus Photobacterium. Like other members of this genus, P. profundum is a marine organism and has two circular chromosomes. P. profundum is a gram-negative rod with the ability for growth at temperatures from 0 °C to 25 °C and pressures from 0.1 MPa to 70 MPa depending on the strain. It has a requirement for salt, is able to metabolise a wide range of simple and complex carbohydrates and has two flagella systems. Cells are rod shape, 2-4?m long and 0.8-1.0?m wide, with a single unsheathed flagella. This bacterium was originally isolated in 1986 from the Sulu Sea and there are currently 4 cultured wild-type strains of P. profundum, (strains SS9, 3TCK, DJS4 and 1230).\n\nStrain SS9 was isolated at a depth of 2500 m from the Sulu Trough. The genome contains 15 rRNA operons, the majority of which are on chromosome 1. There are 2 ATP synthase operons, one on each chromosome. Chromosome 1 contains more conserved genes, while chromosome 2 contains a large proportion of unique genes. High-pressure adaption has been studied by microarray analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
In 2005 Vezzi et al. published the genome sequence for P. profundum strain SS9. The genome of P. profundum consists of a 4.1-Mbp circular chromosome, a 2.2-Mbp minor circular chromosome, as well as a 80-kbp circular plasmid. Strain SS9 has 14 ribosomal RNA (rRNA) genes on chromosome 1, and 1 on chromosome 2; this is the largest number of rRNA genes found in any bacterium. Chromosome 1 consists largely of genes which are essential for growth where as chromosome 2 appears to be a large plasmid, which, on an evolutionary time scale, has be gained several transposable elements. With in the genome of P. profundum there is a large number of open reading frames (ORF) which are unique to SS9 and not found in other members of the Vibrionaceae family.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Photorhabdus luminescens</h1>
<div class="content">
<p>Photorhabdus luminescens (formally Xenorhabdus luminescens) is a Gammaproteobacteria, belonging to the family Enterobacteriaceae, and is a symbiotic pathogen of insects.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
It lives in the gut of an entomopathogenic nematode of the family Heterorhabditidae. When the nematode infects an insect, P. luminescens is released into the blood stream and rapidly kills the insect host (within 48 hours) by producing toxins such as TcA. It also secretes enzymes which break down the body of the infected insect and bioconvert it into nutrients which can be utilised by both nematode and bacteria. In this way, both organisms gain enough nutrients to replicate (or reproduce in the case of the nematode) several times. The bacteria enter the nematode progeny as they develop. P. luminescens also secretes antibiotics to protect against competition from other bacteria.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Picrophilus torridus DSM 9790</h1>
<div class="content">
<p>Picrophilus torridus is a thermoacidophile, and can grow at temperatures around 60C and in environments where the pH drops to 0. The genus name, Picrophilus, means acid lover while the species name, torridus, means dried and burnt. The name reflects the area where the organism was isolated, which was a hot dry soil in Kawayu, in northern Japan, that was solfataric (volcanic area that gives off sulfur). The bacterium has a very impermeable membrane to protons, thus ensuring control of the pH, which internally is 4.6. The genome also encodes a large number of proton-driven secondary transporters which represents another adaptation to the acid environment. This organism expresses a surface layer \n(S-layer) that consists of a semicrystalline array of proteins outside the cell membrane.\n\nThe genome sequence of this strain will allow investigation of the evolution of organisms sharing the extreme growth conditions of a unique niche.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Plasmodium falciparum</h1>
<div class="content">
<p>Plasmodium falciparum is a protozoan parasite, one of the species of Plasmodium that cause malaria in humans. It is transmitted by the female Anopheles mosquito. P. falciparum is the most dangerous of these infections as P. falciparum (or malignant) malaria has the highest rates of complications and mortality. In addition it accounts for 80% of all human malarial infections and 90% of the deaths. It is more prevalent in sub-Saharan Africa than in other regions of the world.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The presence of the parasite in human populations caused selection in the human genome in a multitude of ways, as humans have been forced to develop resistance to the disease. Beet, a doctor working in Southern Rhodesia (now Zimbabwe) in 1948, first suggested that sickle-cell disease could offer some protection to malaria. This suggestion was reiterated by J. B. S. Haldane in 1949 who suggested that thalassaemia could provide similar protection. This hypothesis has since been confirmed and has been extended to hemoglobin C and hemoglobin E, abnormalities in ankyrin and spectrin (ovalocytosis , elliptocytosis), in glucose-6-phosphate dehydrogenase deficiency and pyruvate kinase deficiency, loss of the Gerbich antigen (glycophorin C) and the Duffy antigen on the erythrocytes, thalassemias and variations in the major histocompatibility complex classes 1 and 2 and CD32 and CD36.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Porphyromonas gingivalis  W83</h1>
<div class="content">
<p>Porphyromonas gingivalis is associated with severe and chronic periodontal (tissues surrounding and supporting the tooth) diseases. Progression of the disease is caused by colonization by this organism in an anaerobic environment in host tissues and severe progression results in loss of the tissues supporting the tooth and eventually loss of the tooth itself. The black pigmentation characteristic of this bacterium comes from iron acquisition that does not use the typical siderophore system of other bacteria but accumulates hemin. \n\nPorphyromonas gingivalis strain W83. This strain (also known as HG66) is virulent in a mouse model and has been extensively studied.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Prochlorococcus marinus CCMP1375</h1>
<div class="content">
<p>Prochlorococcus marinus is  non-motile bacterium is a free-living marine organism that is one of the most abundant, as well as the smallest, on earth, and contributes heavily to carbon cycling in the marine environment. This cyanobacterium grows in areas of nitrogen and phosphorus limitation and is unique in that it utilizes divinyl chlorophyll a/b proteins as light-harvesting systems instead of phycobiliproteins. These pigments allow harvesting of light energy from blue wavelengths at low light intensity. There are two ecotypes, one that is low light adapted and one that is high light adapted.\n\nThis very low light-adapted strain is also known as strain SS120.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Propionibacterium acnes KPA171202</h1>
<div class="content">
<p>Propionibacterium acnes is a bacterium that is the most common gram-positive, non-spore forming, anaerobic rod encountered in clinical specimens. The causative agent of acne, it typically grows as an obligate anaerobe. Some strains are aerotolerant, but still show better growth as an anaerobe. It has the ability to produce propionic acid, as its name suggests. It also has the ability to produce catalase along with indole, nitrate, or both indole and nitrate. Propionibacterium resembles Corynebacterium in morphology and arrangement, but is non-toxigenic.  Propionibacterium acnes strain KPA171202  is isolated from human skin, this strain is used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudoalteromonas atlantica T6c</h1>
<div class="content">
<p>Pseudoalteromonas atlantica is a common marine bacterium that exists both in the water column, and in biofilms attached to surfaces. This organism produces a well characterized, commercially important agarase. Pseudoalteromonas atlantica has been isolated from lesions on crabs with shell disease. Shell disease is characterized by progressive degradation of the shell, often leading to an infection of the hemolymph (blood) and may be caused by Pseudoalteromonas, Alteromonas, Vibrio, or other marine organisms. Propionibacterium acnes strain KPA171202 is isolated from human skin, this strain is used for comparative genomics. \n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
In addition to producing extracellular enzymes which attack the shell, Pseudoalteromonas atlantica produces a the lipopolysaccharide which has been shown to be a potential virulence factor in shell disease.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudoalteromonas haloplanktis TAC125</h1>
<div class="content">
<p>This organism is adapted to growth at low temperatures and reactive oxygen species by a number of putative dioxygenases and fatty acid desaturases amongst other proteins. The organism can grow optimally in salt concentrations of 1.5 to 3.5% NaCl. The genome consists of 2 chromosomes, one of which may replicate unidirectionally. Many of the proteins encoded by the genome are similar to Shewanella oneidensis proteins. Some interesting features of this genome include the lack of the nucleoid-associated gene hns, a lack of genes involved in molybdopterin metabolism, a lack of the cAMP-CAP complex, and a lack of the PEP-dependent PTS system.\n\nThe Pseudoalteromonas haloplanktis strain TAC125 was isolated from a sample of coastal sea water near a French Antarctic station (66O40' S; 140O 01' E near Dumont d'Urville).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas aeruginosa</h1>
<div class="content">
<p>Pseudomonas aeruginosa is a common bacterium which can cause disease in animals and humans. It is found in soil, water, skin flora and most man-made environments throughout the world. It thrives not only in normal atmospheres, but also with little oxygen, and has thus colonised many natural and artificial environments. It uses a wide range of organic material for food; in animals, the versatility enables the organism to infect damaged tissues or people with reduced immunity. The symptoms of such infections are generalised inflammation and sepsis. If such colonisations occur in critical body organs such as the lungs, the urinary tract, and kidneys, the results can be fatal. Because it thrives on moist surfaces, this bacterium is also found on and in medical equipment including catheters, causing cross infections in hospitals and clinics. It is also implicated in hot-tub rash, a form of dermatitis associated with poor hygiene and low maintenance of hot tubs.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The G+C-rich Pseudomonas aeruginosa chromosome consists of a conserved core and a variable accessory part. The core genomes of P. aeruginosa strains are largely collinear, exhibit a low rate of sequence polymorphism, and contain few loci of high sequence diversity, notably the pyoverdine locus, the flagellar regulon, pilA, and the O-antigen biosynthesis locus. Variable segments are scattered throughout the genome, of which about one-third are immediately adjacent to tRNA or tmRNA genes. The three known hot spots of genomic diversity are caused by the integration of genomic islands of the pKLC102 / PAGI-2 family into tRNALys or tRNAGly genes. The individual islands differ in their repertoire of metabolic genes, but share a set of syntenic genes which confer their horizontal spread to other clones and species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas aeruginosa UCBPP-PA14</h1>
<div class="content">
<p>This strain of Pseudomonas aeruginosa is a human clinical isolate from a human burn patient. It is infectious in mice, Caenorhabditis elegans, Drosophila melanogaster, and Arabidopsis thaliana. It is an excellent model for the study of pathogenesis and biofilm formation.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas fluorescens strain Pf-5</h1>
<div class="content">
<p>This strain of Pseudomonas Flourescens can suppress the diseases caused by Rhizoctonia solani and Pythium ultimum which affect cotton plants. The production of a number of antibiotics (pyrrolnitrin, pyoluteorin, and 2,4-diacetylphloroglucinol) as well as the production of siderophores (which may affect the ability of competing organisms to obtain environmental iron) by this strain can inhibit phytopathogen growth such as the above-mentioned fungi. The genome of this organism contains a number of genes, estimated at 5.7 % of the chromosome, that encode proteins that are involved in secondary metabolism. A large number of repeat elements (REP) are also found in the genome in greater numbers than in related Pseudomonas spp.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Pseudomonas aeruginosa is an opportunistic human pathogen. While it rarely infects healthy individuals, immunocompromised patients, like burn victims, AIDS-, cancer- or cystic fibrosis-patients are at increased risk for infection with this environmentally versatile bacteria.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas putida strain KT2440</h1>
<div class="content">
<p>This strain was derived from a toluene-degrading isolate, Pseudomonas arvilla strain mt-2 (renamed Pseudomonas putida mt-2), by loss of its plasmid. It is the best characterized saprophytic bacteria, and the choice for cloning and gene expression in Gram-negative bacteria and can survive in the general environment (outside the laboratory). It is used in the development of biopesticides and is the first Gram-negative soil bacterium to be certified as a safety strain by the Recombinant DNA Advisory Committee.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Pseudomonas putida share a high degree of homology to Pseudomonas aeruginosa strains. However, key virulence factors like certain exotoxin genes and type III secretion systems are absent, which makes these bacteria non-pathogenic in nature. As they are metabolically versatile, and well characterized, it makes them great candidates for biocatalysis, bioremediation and other agricultural applications. Certain strains have been used in the production of bioplastics.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas syringae tomato DC3000</h1>
<div class="content">
<p>Pseudomonas syringae tomato DC3000 is mainly characterized as causing bacterial speck disease on tomato plants, which has a large economic impact. This organism is mainly endophytic and is a poor colonizes of plant surfaces but can multiply within the host. It is the model organism for the study of plant-pathogen interactions.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Pseudomonas syringae includes many plant pathogens of important crops, which makes it a model organism in plant pathology. Its natural environment is on the surface of plant leaves and it can withstand various stressful conditions, like rain, wind, UV radiation and drought.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Psychrobacter arcticum 273-4</h1>
<div class="content">
<p>This strain is part of an analysis of the growth of organisms at and their adaption to low temperature, a study that is being conducted by Michigan State University as a member of NASA's Astrobiology Institute. Insight into how these organisms grow and adapt to life in low temperature environments could aid in understanding transport of microbes through space, either as a contaminant on human spacecraft, or as a passenger on an asteroid or comet.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pyrobaculum aerophilum</h1>
<div class="content">
<p>This organism is a facultatively anaerobic, nitrate-reducing hyperthermophilic archaeon. Unlike other members of the genus Pyrobaculum, P. aerophilum is able to grow both aerobically and anaerobically by dissimilatory nitrate reduction. Cells are rod-shaped 0.5-0.6 x 1.5-8 µm and may form aggregates arranged in a V-shape.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pyrococcus abyssi</h1>
<div class="content">
<p>This organism is an obligate anaerobic, hyperthermophilic archaeon. Pyrococcus abyssi is highly motile via tufts of polar flagella. Growth occurs at a temperature of 67-103°C with optimal growth at 103°C at 200 atmospheres of pressure. Increased hydrostatic pressure has been found to extend the upper growth temperature and stimulate growth.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pyrococcus furiosus</h1>
<div class="content">
<p>This organism is a strictly anaerobic, hyperthermophilic archaeon. Cells are slightly irregular cocci, 0.8-2.5µm in diameter. Growth occurs at temperatures between 67 and 103°C, with optimal growth at 100°C and a pH of 7.0. This organism is highly motile due to a bundle of flagella.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pyrococcus horikoshii</h1>
<div class="content">
<p>This organism is an obligate anaerobic, hyperthermophilic archaeon. Growth occurs at 80-102°C with optimal growth at 98°C and a pH of 7.0. Presence of elemental sulfur has been found to greatly stimulate growth. Motility is via tufts of polar flagella.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ralstonia solanacearum</h1>
<div class="content">
<p>Ralstonia solanacearum is a Gram-negative, plant pathogenic, soil bacterium. It colonises the xylem, causing a bacterial wilt in a very wide range of potential host plants. It is known as Granville wilt when it occurs in tobacco.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Ralstonia solanacearum is a high profile alien plant pathogen of A2 Quarantine status affecting a very wide range of crops. This means that it is present in parts of Europe but is under statutory control. Worldwide, the most important crops affected are: Potatoes Tomatoes, Tobacco, Banana and Pelargonium. In the UK and the rest of the EU the most important crops affected are Potato and Tomato. It would cause serious economic damage were it to become more established than it currently is. Losses are due to actual yield reduction and also due to statutary measures taken to eliminate the disease. Control within the EU is governed by an EC Control. The bacterium is found worldwide.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rattus norvegicus</h1>
<div class="content">
<p>The brown rat, common rat, Hanover rat, Norway rat, Norwegian rat, or wharf rat (Rattus norvegicus) is one of the best known and most common rats. One of the largest muroids, it is a brown or grey rodent with a body up to 25 cm (10 in) long, and a similar tail length; the male weighs on average 350 g (12 oz) and the female 250 g (9 oz). Thought to have originated in northern China, this rodent has now spread to all continents, except Antarctica, and is the dominant rat in Europe and much of North Americamaking it the most successful mammal on the planet after humans.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Brown rats carry some diseases, including Weil's disease, rat bite fever, cryptosporidiosis, Viral hemorrhagic fever (VHF), Q fever and hantavirus pulmonary syndrome. In the United Kingdom, brown rats are an important reservoir for Coxiella burnetii, the bacteria that causes Q fever, with seroprevalence for the bacteria found to be as high as 53% in some wild populations.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhodopseudomonas palustris CGA009</h1>
<div class="content">
<p>Rhodopseudomonas palustris is an organism that has a diverse metabolism and is capable of growth using light, inorganic, or organic compounds as energy sources and carbon dioxide or organic compounds as carbon sources. This bacterium, commonly found in soil and water environments, is also capable of degrading a wide range of toxic organic compounds, and may be of use in bioremediation of polluted sites. Rhodopseudomonas palustris strain CGA009 contains genes for bacteriochlorophyll and carotenoid synthesis as well as genes for the light reaction center that allow photophosphorylation in addition to carbon (RubisCO) and nitrogen fixation (nitrogenase) genes. It also has genes for the oxidation of inorganic compounds such as thiosulfate and hydrogen and nitrogenases for nitrogen fixation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rickettsia conorii</h1>
<div class="content">
<p>Rickettsia conorii is a species of the genus Rickettsia. It is the causative agent of Boutonneuse fever, a member of the spotted fever group. Subspecies have been identified.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rickettsia prowazekii</h1>
<div class="content">
<p>Rickettsia prowazekii is a species of gram negative, bacillus, obligate intracellular parasitic, aerobic bacteria that is the etiologic agent of epidemic typhus, transmitted in the feces of lice and fleas. In North America, the main reservoir for R. prowazekii is the flying squirrel. R. prowazekii is often surrounded by a protein microcapsular layer and slime layer; the natural life cycle of the bacterium generally involves a vertebrate and an invertebrate host, usually an arthropod, typically a louse. A form of R. prowazekii that exists in the feces of arthropods remains stably infective for months.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The genome of R. prowazekii provides great insight, especially to the origin of modern-day mitochondria. The R. prowazekii is not only the closest known relative to mitochondria through genome sequencing, as it also has a highly derived genome like that of mitochondria. Since the two are so similar, they likely have a very similar evolutionary history.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rickettsia typhi wilmington</h1>
<div class="content">
<p>Rickettsia typhi strain Wilmington is a commonly used strain. Rickettsia typhi is an organism that causes murine typhus and is an obligate intracellular pathogen that infects both the flea vector and hosts such as human, rat, and mouse. It is in the typhus group along with R. prowazekii, and genomic comparisons demonstrate colinearity and similarity to the genome of that organism except for two independent inversions near the origin and terminus. \n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Saccharomyces bayanus</h1>
<div class="content">
<p>Saccharomyces bayanus is a yeast of the genus Saccharomyces, and is used in winemaking and cider fermentation. It is closely related to Saccharomyces cerevisiae. Both Saccharomyces bayanus and Saccharomyces pastorianus contain diverse strains, with different genetic and metabolic characteristics, that may have a hybrid origin. In their current classification, both S. bayanus and S. pastorianus may be more correctly regarded as species complexes.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Saccharomyces castellii</h1>
<div class="content">
<p>Saccharomyces castellii, also known as Naumovia castellii, is a budding yeast that is a member of the Saccharomyces sensu lato group. It is more distantly related to Saccharomyces cerevisiae than are Saccharomyces sensu stricto species. The Saccharomyces castellii nuclear genome is distributed among 9 chromosomes. The circular mitochondrial genome is composed of 25,753 bp.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Saccharomyces cerevisiae</h1>
<div class="content">
<p>Saccharomyces cerevisiae is a species of budding yeast. It is perhaps the most useful yeast owing to its use since ancient times in baking and brewing. It is one of the most intensively studied eukaryotic model organisms in molecular and cell biology, much like Escherichia coli as the model prokaryote. It is the microorganism behind the most common type of fermentation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
S. cerevisiae can be transformed allowing for either the addition of new genes or deletion through homologous recombination. Furthermore, The ability to grow S. cerevisiae as a haploid simplifies the creation of gene knockouts strains. As a eukaryote, S. cerevisiae shares the complex internal cell structure of plants and animals without the high percentage of non-coding DNA that can confound research in higher eukaryotes.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Saccharomyces kluyveri</h1>
<div class="content">
<p>Saccharomyces kluyveri is a plant pathogen. Saccharomyces kluyveri is a budding yeast related to Saccharomyces cerevisiae. S. kluyveri resorts to fermentation (degrading sugars in the absence of oxygen) only when oxygen is limiting. This means that S. kluyveri makes a more efficient use of glucose for energy production.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
S. kluyveri's habitat has not been as extensively characterized as that of yeast species such as S. cerevisiae or Candida albicans; only about 30 isolates have been described. However, the species appears to be wide spread in the environment. It has been isolated from Drosophila species in North America, soil in Europe, and various tree species in India and North America. Little data has been published on S. kluyveri's habitat since it was first described in 1956, but there is a report of S. kluyveri being isolated from the tissues of an HIV infected patient.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Saccharomyces kudriavzevii</h1>
<div class="content">
<p>Saccharomyces kudriavzevii, a budding yeast, is a member of the Saccharomyces sensu stricto group. It is closely related to Saccharomyces cerevisiae; these two yeast species are separated by an estimated 10-20 million years of evolution. The Saccharomyces kudriavzevii genome shows a high conservation of synteny with the S. cerevisiae genome.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Saccharomyces mikatae</h1>
<div class="content">
<p>Saccharomyces mikatae, a budding yeast, is a member of the Saccharomyces sensu stricto group. It is closely related to Saccharomyces cerevisiae; these two yeast species are separated by an estimated 10-20 million years of evolution. The Saccharomyces mikatae genome shows a high conservation of synteny with the Saccharomyces cerevisiae genome, with only four reciprocal translocations and thirteen inversions outside of the telomeric regions.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Saccharomyces paradoxus</h1>
<div class="content">
<p>Saccharomyces paradoxus, a budding yeast used in winemaking, is a member of the Saccharomyces sensu stricto group. It is closely related to Saccharomyces cerevisiae; these two yeast species are separated by an estimated 5-10 million years of evolution. The Saccharomyces paradoxus genome shows a high conservation of synteny with the Saccharomyces cerevisiae genome, with no reciprocal translocations, four inversions, and three segmental duplications outside of the telomeric regions.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Saccharophagus degradans 2-40</h1>
<div class="content">
<p>Saccharophagus degradans is capable of degrading insoluble complex carbohydrates through the collective action of enzyme complexes found on its cell surfaces, utilizing the degradation products as a carbon source. The compounds degraded include chitin, which constitute insect exoskeleton, plant cell wall components such as cellulose, and algal cell wall carbohydrates. This organism may be useful in bioremediation. The Saccharophagus degradans 2-40 strain is a marine gamma-proteobacterium that was isolated from decaying Spartina alterniflora, a salt marsh cord grass, in the Chesapeake Bay, USA. Saccharophagus degradans 2-40 has been used to produce ethanol from plant material and may be useful for the production bioethanol.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica  Paratyphi ATCC 9150</h1>
<div class="content">
<p>Salmonella is a group of Enterobactericiae that have pathogenic characteristics and are one of the most common causes of enteric infections (food poisoning) worldwide. There are now two Salmonella species, Salmonella bongori and Salmonella enterica. Certain serovars of Salmonella enterica are responsible for more serious diseases such as Typhoid fever. Salmonella enterica subsp. enterica serovar Paratyphi A str. ATCC 9150 has a narrow host range and causes a typhoid-like (paratyphoid fever) illness in humans. It is especially prevalent in southern and eastern Asia, and has been associated with some particularly virulent outbreaks. A number of isolates are increasingly antibiotic resistant. This strain is considered wild-type.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella typhi</h1>
<div class="content">
<p>Salmonella is a group of Enterobactericiae that have pathogenic characteristics and are one of the most common causes of enteric infections (food poisoning) worldwide. There are now two Salmonella species, Salmonella bongori and Salmonella enterica. Certain serovars of Salmonella enterica are responsible for more serious diseases such as Typhoid fever. Salmonella enterica Serovar Typhi (historically elevated to species status as S. typhi) is the disease agent in typhoid fever.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella typhimurium LT2</h1>
<div class="content">
<p>Salmonella is a group of Enterobactericiae that have pathogenic characteristics and are one of the most common causes of enteric infections (food poisoning) worldwide. There are now two Salmonella species, Salmonella bongori and Salmonella enterica. Certain serovars of Salmonella enterica are responsible for more serious diseases such as Typhoid fever. Salmonella enterica subsp. enterica serovar typhimurium strain LT2 was first isolated in the 1940s by Lilleengen, and typed based on phage sensitivity. It became the principle laboratory strain and many important biochemical and genetic experiments were performed using this strain, including classical phage transduction studies. LT2 contains a plasmid, pSLT, that encodes additional virulence functions. Numerous prophages, some of which contain virulence genes, are inserted into the chromosome of LT2 as compared to related Salmonella.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Schizosaccharomyces pombe</h1>
<div class="content">
<p>Schizosaccharomyces pombe, also called "fission yeast", is a species of yeast. It is used as a model organism in molecular and cell biology. It is a unicellular eukaryote, whose cells are rod-shaped. Cells typically measure 3 to 4 micrometres in diameter and 7 to 14 micrometres in length. Its genome, which is approximately 14.1 million base pairs, is estimated to contain 4,970 genes, possibly the fewest in any eukaryote.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The sequence of the S. pombe genome was published in 2002, by a consortium led by the Sanger Institute, becoming the sixth model eukaryotic organism whose genome has been fully sequenced. This has fully unlocked the power of this organism, with many genes homologous to human disease genes being identified.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Shewanella oneidensis</h1>
<div class="content">
<p>Shewanella oneidensis is a facultative anaerobe that is capable of using a wide variety of terminal electron acceptors during anaerobic respiration which may make it valuable for bioremediation. Since the bacteria can reduce chromium and uranium from the liquid phase to form insoluble compounds, they may be used to eliminate these two environmental pollutants from water.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Shigella boydii Sb227</h1>
<div class="content">
<p>Shigella boydii is an uncommon species except in India, where it was first isolated. The 18 known serotypes are antigenically distinct, expressing a diverse range of toxins in addition to a Shigella-specific toxin. Progression to clinical dysentery occurs in most patients infected with this organism. The Shigella boydii Sb 227 strain is an isolate from an epidemic that took place in China in the 1950s.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Shigella dysenteriae</h1>
<div class="content">
<p>Shigella dysenteriae is a species of the rod-shaped bacterial genus Shigella. Shigella can cause shigellosis (bacillary dysentery). Shigellae are Gram-negative, non-spore-forming, facultatively anaerobic, non-motile bacteria. S. dysenteriae, spread by contaminated water and food, causes the most severe dysentery because of its potent and deadly Shiga toxin, but other species may also be dysentery agents.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Shigella flexneri 2a</h1>
<div class="content">
<p>Shigella flexneri along with Shigella sonnei, is the major cause of shigellosis in industrialized countries and is responsible for endemic infections. Shigella flexneri can cause diarrhea in humans. There are several different serogroups of Shigella; S. flexneri belongs to group B. S. flexneri infections can usually be treated with antibiotics although some strains have become resistant.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Shigella sonnei Ss046</h1>
<div class="content">
<p>Shigella sonnei is an organism that is the leading cause of dysentery in industrialized countries. The disease is usually less severe than other types of Shigella, causing mild diarrhea and dehydration. Shigella sonnei strain Ss046 is an isolate from an epidemic in China in the 1950s.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Silicibacter pomeroyi DSS-3</h1>
<div class="content">
<p>Silicibacter pomeroyi, is capable of degrading the organic sulfur compound DMSP (dimethylsulfoniopropionate) and can metabolize a number of sulfur compounds. DMSP is synthesized by marine algae and the degradation product dimethylsulfide contributes to the global sulfur cycle. Silicibacter pomeroyi strain DSS-3 is a marine bacterium and member of the Roseobacter clade. It was isolated off of the coast of Georgia in 1998. DSS-3 is the type strain for this species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Sinorhizobium meliloti</h1>
<div class="content">
<p>Sinorhizobium meliloti is a nitrogen-fixing bacterium (rhizobium). It forms a symbiotic relationship with legumes from the genera Medicago, Melilotus and Trigonella, including model legume Medicago truncatula. The S. meliloti genome contains three replicons: a 3.65 megabase chromosome and two megaplasmids, pSymA (1.35 megabases) and pSymB (1.68 megabases), that have all been fully sequenced.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Sodalis glossinidius morsitans</h1>
<div class="content">
<p>Sodalis glossinidius is an organism known exclusively in endosymbiosis with tsetse flies. It is maternally transmitted, and is one of the few bacterial endosymbionts of insects that can be cultured successfully in vitro. Genome data reveals a high proportion of pseudogenes in this species, many of which were, in their functional state, involved in defense or transport of carbohydrates and inorganic ions. This suggests a degenerative adaptation to the immunity and restricted nutritional status of the host. This suggests that this species represtents an evolutionary intermediate transitioning between a free-living to a mutualistic lifestyle. Sodalis glossinidius 'morsitans' is a strain used in comparative genome analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Sorex araneus</h1>
<div class="content">
<p>The Common Shrew or Eurasian Shrew (Sorex araneus) is the most common shrew, and one of the most common mammals, throughout Northern Europe, including Great Britain, but excluding Ireland. The average shrew measures 55-82 mm in length and usually weighs 5-12 grams. Shrews are recognized by their velvety dark brown fur and pale undersides. Juvenile shrews have lighter fur until their first moulting, when darker winter fur grows in. The Common Shrew has small eyes, a pointed, mobile snout, and red-tipped teeth. The Common Shrew has a life span of approximately 23 months.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The chromosome number (karyotype) of Sorex araneus varies from one individual to another. This is an example of chromosomal polymorphism; the variability occurs as a result of chromosome fusions or disassociations.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Staphylococcus aureus COL</h1>
<div class="content">
<p>Staphylococcus aureus is an organism that is a major cause of nosocomial (hospital-acquired) and community-acquired infections. Since its discovery as an opportunistic pathogen, S. aureus continues to be a major cause of mortality and is responsible for a variety of infections including, boils, furuncles, styes, impetigo and other superficial skin infections in humans. Also known to cause more serious infections particularly in the chronically ill or immunocompromised. These include pneumonia, deep abscesses, osteomyelitis, endocarditis, phlebitis, mastitis and meningitis. The ability to cause invasive disease is associated with persistance in the nasal cavity of a host. Staphylococcus aureus COL is a strain that is a methicillin-resistant (MRSA) strain and is also resistant to several other antibiotics including penicillin and tetracycline.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Staphylococcus epidermidis ATCC 12228</h1>
<div class="content">
<p>Staphylococcus epidermidis is an organism that is the most prevalent and persistent Staphylococcus species on human skin. In recent years, S. epidermidis has emerged as a common cause of nosocomial (hospital-acquired) infections, including catheter-associated infections and septicemia, particularly in immunocompromised patients. Similar to S. aureus , Staphylococcus epidermidis is highly resistant to many antibiotics including penicillins and cephalosporins. Staphylococcus epidermidis ATCC 12228 is a strain that is a non-biofilm-forming, non-infection associated strain used for detection of residual antibiotics in food products.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Streptococcus agalactiae 2603</h1>
<div class="content">
<p>Streptococcus agalactiae is the leading cause of meningitis and sepsis in newborns. Additionally this organism is the cause of serious infections in immunocompromised adults. Clinical manifestations of infection include urinary tract and soft tissue infections as well as life-threatening sepsis and meningitis. Additionally S. agalactiae is able to asymptomatically colonize human skin and mucous membranes. This organism is also implicated in cases of mastitis in cows. Streptococcus agalactiae strain 2603 is an isolate and is the most common type among GBS-infected (group B streptococcal) non-pregnant adults.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Streptococcus mutans</h1>
<div class="content">
<p>Streptococcus mutans is a Gram-positive, facultatively anaerobic bacterium commonly found in the human oral cavity and is a significant contributor to tooth decay.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Along with S. sobrinus, S. mutans plays a major role in tooth decay, metabolizing sucrose to lactic acid. The acidic environment created in the mouth by this process is what causes the highly mineralized tooth enamel to be vulnerable to decay. S. mutans is one of a few specialized organisms equipped with receptors that help for better adhesion to the surface of teeth.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Streptococcus pneumoniae R6</h1>
<div class="content">
<p>Streptococcus pneumoniae is an organism that is a member of the normal human nasopharyngeal flora. There are more than 85 different antigenic types of S. pneumoniae, which are classified based on their capsule antigens. This organism is also the most common cause of sinusitis, acute bacterial otitis media, and conjunctivitis beyond early childhood. S. pneumoniae is also known to cause pneumonia, meningitis, and bacteremia. Streptococcus pneumoniae R6 is a strain that is an avirulent descendent of a type 2 capsule S clinical isolate used by Avery, MacLeod, and McCarty to demonstrate the genetic function of DNA.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Streptococcus pyogenes MGAS10394</h1>
<div class="content">
<p>Streptococcus pyogenes is an organism that is a member of the normal human nasopharyngeal flora. S. pyogenes is a group A streptococcus and is the leading cause of uncomplicated bacterial pharyngitis and tonsillitis. Infections can also result in sinusitis, otitis, mastoiditis, pneumonia with empyema, joint or bone infections, necrotizing fasciitis or myositis, and, more infrequently, in meningitis or endocarditis. S. pyogenes infections of the skin can be superficial (impetigo) or deep (cellulitis). This organism is commonly referred to by the lay press as "flesh eating bacteria." Streptococcus pyogenes MGAS10394 is a strain that is a serotype M6 isolate cultured from a child with pharyngitis in private elementary school in Pennsilvania. The strain contains the mefA gene encoding macrolide resistance and shown to be resistant to erythromycin, and has a chromosomal field-inversion gel-electrophoresis pattern characteristic of the serotype M6 clone causing disease on the territory of USA. It has been deposited at the American Type Culture Collection (accession no. BAA-946).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Streptococcus thermophilus CNRZ1066</h1>
<div class="content">
<p>Streptococcus thermophilus is a lactic acid bacteria often used as a starter culture for fermented dairy products including yogurt and cheese. At the same time this dairy species is phylogenetically close to pathogenic streptococci, evoking the possibility that it has a virulence potential. Streptococcus thermophilus strain CNRZ1066 was isolated from yogurt in France.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Streptomyces avermitilis</h1>
<div class="content">
<p>Streptomyces are bacteria that are widely distributed in nature, especially in the soil. The characteristic earthy smell of freshly plowed soil is actually attributed to the aromatic terpenoid geosmin produced by species of Streptomyces. Streptomycetes are crucial in the soil environment because their diverse metabolism allows them to degrade the insoluble remains of other organisms, including recalcitrant compounds such as lignocelluloses and chitin. Streptomycetes produce both substrate and aerial mycelium. Streptomyces avermitilis is a well known producer of the anti-parasitic agent avermectin which is widely used to rid livestock of worm and insect infestations and to protect large numbers of people from river blindness in sub-Saharan Africa.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
There are currently 364 known species of this genus, many of which are the most important industrial producers of antibiotics and other secondary metabolites of antibacterial, antifungal, antiviral, and antitumor nature, as well as immunosuppressants, antihypercholesterolemics, etc.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Streptomyces coelicolor</h1>
<div class="content">
<p>Streptomyces are bacteria that are widely distributed in nature, especially in the soil. The characteristic earthy smell of freshly plowed soil is actually attributed to the aromatic terpenoid geosmin produced by species of Streptomyces. Streptomycetes are crucial in the soil environment because their diverse metabolism allows them to degrade the insoluble remains of other organisms, including recalcitrant compounds such as lignocelluloses and chitin. Streptomycetes produce both substrate and aerial mycelium. Streptomyces coelicolor is a soil-dwelling filamentous organism responsible for producing more than half of the known natural antibiotics. It is a well-studied species of Streptomyces and genetically is the best known representative.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Strongylocentrotus purpuratus</h1>
<div class="content">
<p>Strongylocentrotus purpuratus, or the purple sea urchin, is one of the sharp-spined sea urchin species. The spines are used as a means of defense against would-be predators. This urchin is deep purple in color, and grows to a diameter of about 4 inches. They are found along the Pacific coast of the United States and Canada where they cling to depressions in the rocks during low tide.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
A study of the purple sea urchin by Caltech's Kerckhoff Marine Biological Laboratory (KML) revealed that urchins have developed light sensors - a primitive form of vision. Scientists found genes associated with vision activated in the urchins tube feet, puny projections on the animals shell that help urchins move and feed.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Sulfolobus solfataricus</h1>
<div class="content">
<p>Sulfolobus solfataricus is an obligate aerobic, hyperthermophilic, acidophilic, sulfur-metabolizing archeon. Cells are highly irregular cocci 0.8-2.0 µm in diameter. Growth occurs between 50 and 87°C with optimal growth at 85°C and a pH of 2-4.5.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Sulfolobus tokodaii</h1>
<div class="content">
<p>Sulfolobus tokodaii is an obligate aerobic, hyperthermophilic, acidophilic, sulfur-metabolizing archaeon. Cells are highly irregular cocci 0.8-2.0 µm in diameter. Growth occurs optimally at 80°C and pH 2.5-3.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Symbiobacterium thermophilum IAM 14863</h1>
<div class="content">
<p>Symbiobacterium thermophilum was discovered by screening for thermostable tryptanophases in Japanese compost. Cultured growth of this organism requires the presence of another bacterial species, such as a Bacillus or Escherichia coli, which provides diffusable metabolites required for its growth. Pure cultures can be obtained by growing Symbiobacterium thermophilum in a bioreactor, separated from its symbiotic counterpart by a dialysis membrane. This species is widely distributed in soil, animal intestines, feces, and feeds. Because of its symbiotic nature, it cannot be cultured with conventional methods. Symbiobacterium thermophilum strain IAM 14863 is used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Synechococcus elongatus PCC 6301</h1>
<div class="content">
<p>Synechococcus elongatus is an obligate photoautotroph. It has been studied extensively by an international research community with respect to acquisition of organic carbon, transport and regulation of nitrogen compounds, adaptation to nutrient stresses, and reponse to light intensity. Synechococcus elongatus PCC 6301 is a freshwater organism woth the type strain for the Cyanobacteria originally described as Anacystis nidulans. The genes controlling circadian rythms have been studied. The genome of this strain and its close relative Synechococcus elongatus PCC 7942 are being evaluated for further understanding of the circadian clock mechanism.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Takifugu rubripes</h1>
<div class="content">
<p>a genus of pufferfish, often better known by the Japanese name fugu. There are 25 species belonging to the genus Takifugu, which can be found worldwide from about 45° latitude north to 45° latitude south, mostly in salt water, but sometimes also in fresh water or brackish water. Their diet consists mostly of algae, mollusks, invertebrates and sometimes crustaceans. The fish defend themselves by inflating their bodies to several times normal size and by poisoning their predators. These defenses allow the fish to actively explore their environment without much fear of being attacked.\n\nThe fish is highly toxic, but despite this  or perhaps because of it  is considered a delicacy in Japan. The fish contains lethal amounts of the poison tetrodotoxin in the internal organs, especially the liver and the ovaries, but also in the skin and the testes. Therefore, only specially licensed chefs can prepare and sell fugu to the public, and the consumption of the liver and ovaries is forbidden. But because small amounts of the poison give a special desired sensation on the tongue, these parts are considered the most delicious by some gourmets. Every year a number of people die because they underestimate the amount of poison in the consumed fish parts.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Apparently due to some unknown selection pressure, intronic and extragenic sequences have been drastically reduced within this family. As a result, they have the smallest-known genomes yet found amongst the vertebrate animals, while containing a genetic repertoire very similar to other fishes and thus comparable to vertebrates generally. Since these genomes are relatively compact it is relatively fast and inexpensive to compile their complete sequences, as has been done for two species (Takifugu rubripes and Tetraodon nigroviridis).
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Tetraodon nigroviridis</h1>
<div class="content">
<p>Tetraodon nigroviridis is one of the pufferfish known as the green spotted puffer (the other being Tetraodon fluviatilis). It is found across South and Southeast Asia in coastal freshwater and brackish water habitats. Tetraodon nigroviridis reaches a maximum length of about 15 cm (5.9 in). In February 2009 it was successfully bred in captivity at University of Florida using a new variation of the ovarian lavage technique.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Tetraodon nigroviridis has the smallest known vertebrate genome, roughly 340 million base pairs, and has thus been selected as a model organism for genetics. In 2004 a draft of its genome sequence was published
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Thermoanaerobacter tengcongensis</h1>
<div class="content">
<p>Thermoanaerobacter tengcongensis is an anaerobic thermophile which is able to grow at temperatures between 50 and 80 0C. Unlike other Thermoanaerobacter species T. tengcongensis does not produce spores, is gram-negative and is not motile.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Thermococcus kodakaraensis strain KOD1</h1>
<div class="content">
<p>Thermococcus kodakaraensis is a hyperthermophilic archeon. Proteins from this organism have been extensively studied to find thermostable enzymes for industrial and biotechnological applications.\n\nThis organism was originally identified as Pyrococcus sp. strain KOD1. It was isolated from a solfatara on Kodakara Island, Japan. A gene disruption system has been developed for this organism. A thermostabile DNA polymerase is commercially available that was originally isolated from this organism and research is continuing to develop commercial applications for other heat-stable enzymes from this organism</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Thermoplasma  acidophilum</h1>
<div class="content">
<p>T. acidophilum is a thermoacidophilic archaeon that was isolated from a coal refuse pile. T. acidophilum grows optimally at 56 ?, pH 1.8 and size of a cell is about 1 ?m. T. acidophilum lacks a cell wall and the cell membrane is exposed directly outside. T. acidophilum shows various cell shapes depending upon growth conditions and stages. Though there is no cell wall, it shows various forms.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Thermoplasma volcanium.</h1>
<div class="content">
<p>This organism is a thermophilic, acidophilic, mycoplasma that was isolated from acidic hydrothermal vents on the shore of Aeolian Island of Vulcano, Italy. Growth of this facultative aerobe occurs at temperatures of 33-67°C with optimal at 60°C and pH of 1.0-4.0 (optimal 2.0). This strain is the type strain for this species, and was isolated from a solfataric</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Thermosynechococcus elongatus</h1>
<div class="content">
<p>Thermosynechococcus elongatus is a thermophilic cyanobacterium. This organism is readily transformable making it useful for molecular analyses.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Thermotoga maritima</h1>
<div class="content">
<p>Thermotoga maritima is a hyperthermophile with an optimum growth temperature of 80 0C. This organism, a member of the Thermotogales, has the characteristic sheath-like envelope which extends beyond the cell wall. These organisms are heterotrophs are able to use a wide range of carbohydrates.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Thermus thermophilus strain HB27</h1>
<div class="content">
<p>This organism was originally isolated from a natural thermal environment in Japan. It is naturally competent for DNA uptake, and is a useful organism for biotechnological applications. A host of thermostable enzymes have been purified and crystallized from this organism, and many others may be of use in industrial applications. These include proteases, enzymes for the conversion of carbohydrates such as starch and xylose, phosphatases, catalases, DNA and RNA processing enzymes such as DNA polymerase and ribonucleases, as well as many others. The organism has a megaplasmid that encodes some functions in carotenoid synthesis, which are pigmented molecules that have effects on membrane fluidity. The chromosome encodes genes involved in type IV pilus biogenesis, which may have a role in DNA transformation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Thiomicrospira crunogena XCL-2</h1>
<div class="content">
<p>This organism is a colorless sulfur-oxidizing bacterium isolated from a deep-sea hydrothermal vent.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Treponema denticola ATCC 35405</h1>
<div class="content">
<p>Treponema denticola is a member of the spirochaetales and is a motile, slender, helically shaped flexible organism. Motility is by rapid rotation around the long axis, flexation of the cell and locomotion along a helical path. The most distinctive property is the presence of periplasmic flagella wound around the helical protoplasmic cylinder and encased in an outer sheath. Obligately anaerobic growth occurs at 30-42°C, with a pH of 6.5-8.0. The bacterium is commonly found in the oral cavity of humans and often associated with periodontal disease, an inflammation of the gum tissue that frequently precedes bone resorbtion and subsequent tooth loss. Periodontal disease has now become a major concern in dentistry and it is estimated that 80% of adults in the USA have had periodontal disease at some point in their lives.\n\nTreponema denticola ATCC 35405 was isolated from a human periodontal pocket and is the type strain of this organism.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Treponema pallidum</h1>
<div class="content">
<p>Treponema pallidum is a spirochaete bacterium. It is not seen on a Gram stained smear because the organism is too thin. Habitat: Human genital tract. Transmission by sexual contact and from mother to fetus across placenta.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Tribolium castaneum</h1>
<div class="content">
<p>The red flour beetle (Tribolium castaneum) is considered a common insect pest, since it feeds on stored grains in mills, warehouses and pantries. This beetle is noted for its importance as a genetic model for research, both medically and economically motivated. T. castaneum was the first animal in which prostaglandin synthetase inhibitors were identified, a secretion product of their stink glands. Much of the study to this point has focused on the red flour beetle's resistance mechanisms and how these can help in the improvement of insecticides and agents used to control pests and disease vectors.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Tropheryma whipplei TW08/27</h1>
<div class="content">
<p>Tropheryma whipplei is the causative agent of Whipple's disease. First described by US pathologist and Nobel laureate George H. Whipple in 1907, this potentially lethal and persistent, but rare disease, manifests itself as a multisystem disorder, involving the intestinal tract (diarrhea, abdominal pain) and various other organs (heart, brain, nervious system, joints). Very little is known about the biology of this bacterium. Whipple did not associate the disease with any bacterial agent, and it was only in 1961 when bacteria were detected in electron microphotographs of affected tissues that the connection was made. In 1990, DNA typing revealed the identity of this new genus and species of actinomycetes. The bacterium cannot be cultured apart from its eukaryotic host, and it grows very slowly (reported duplication time 4-17 days).\n\nThis strain was isolated from the cerebrospinal fluid a woman who had suffered severe weight loss in Germany. The patient had been treated with antibiotics followed by a year of no treatment prior to the sample being taken.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ureaplasma urealyticum</h1>
<div class="content">
<p>Ureaplasma urealyticum is a bacterium belonging to the family Mycoplasmataceae. U. urealyticum is part of the normal genital flora of both men and women. It is found in about 70% of sexually active humans. It had also been described to be associated with a number of diseases in humans, including non-specific urethritis (NSU), infertility, chorioamnionitis, stillbirth, premature birth, and, in the perinatal period, pneumonia, bronchopulmonary displasia[ and meningitis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
There are six recognised Ureaplasma species, They have a GC content of 27-30%, and a genome size ranging between 0.76-1.17 Mbp, and cholesterol is required for growth. A defining characteristic of the genus is that they perform urea hydrolysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Vibrio cholerae</h1>
<div class="content">
<p>Vibrio cholerae (also Kommabacillus) is a motile gram negative curved-rod shaped bacterium with a polar flagellum that causes cholera in humans. V. cholerae and other species of the genus Vibrio belong to the gamma subdivision of the Proteobacteria.  V. cholerae colonizes the gastrointestinal tract, where it adheres to villi. It produces a protease called mucinase and an amylase referred to as chitinase. Chitinase is responsible for the ability of Vibrio cholerae to enter copepods. Mucinase is a non-specific protease that assists entry into the human gastro-intestinal tract.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Vibrio fischeri ES114</h1>
<div class="content">
<p>Vibrio fischeri is found in marine environments and was originally named by Bernard Fischer during a sea voyage in the 1800s. It is a symbiont in fish and squids and is responsible for light generation in those organisms, which use it as a defense mechanism to avoid predators. Vibrio fischeri strain ES114 was isolated from the light organs of the squid Euprymna scolopes. Interestingly, the genome encodes a cholera toxin phage-like element on chromosome II. There are also a number of pilus synthesis genes present in the genome, which may function during symbiosis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Vibrio parahaemolyticus</h1>
<div class="content">
<p>Vibrio parahaemolyticus is a curved, rod-shaped, Gram-negative bacterium found in brackish saltwater, which, when ingested, causes gastrointestinal illness in humans. V. parahaemolyticus is oxidase positive, facultatively aerobic, and does not form spores. Like other members of the genus Vibrio, this species is motile, with a single, polar flagellum.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Vibrio vulnificus CMCP6</h1>
<div class="content">
<p>Vibrio vulnificus is an opportunistic pathogen that can attack immunocompromised patients and cause gastroenteritis (inflammation of mucous membranes of stomach and intestine), wound infections, and primary septicemia (spread of the organism through the blood). This organism is the major cause of death from eating raw oysters, especially in people with liver damage. It only affects humans and other primates. Vibrio vulnificus strain CMCP6 is to be used for comparative studies.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Wigglesworthia brevipalpis</h1>
<div class="content">
<p>Wigglesworthia brevipalpis is a Gram-negative bacterium in the family Enterobacteriaceae, related to E. coli, which lives in the gut of the tsetse fly. The bacterium bears the name of the British entomologist who first described it, Sir Vincent Brian Wigglesworth. Wigglesworthia has symbiotically co-evolved with the tsetse fly for millions of years, and is a text book example of a bacterial endosymbiont. Because of this relationship, Wigglesworthia has lost a large part of its genome and has one of the smallest known genomes of any living organism. Wigglesworthia also synthesizes key vitamins which the tsetse fly does not get from its diet of blood. Without the vitamins Wigglesworthia produces, the tsetse fly cannot reproduce.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Wolinella succinogenes</h1>
<div class="content">
<p>Wolinella succinogenes is most commonly isolated from the bovine rumen and is considered nonpathogenic, however, this organism has also been isolated from the human gingiva and dental root canal infections.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Xanthomonas campestris B100</h1>
<div class="content">
<p>Xanthomonas campestris is a major cause of black rot in crucifers, a disease that results in massive tissue degeneration. It also produces an extracellular polysaccharide known as xanthan, which is harvested commercially as a food stabilizing agent for use in industry. Xanthomonas campestris strain B100 is to be sequenced for comparative genome analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Xanthomonas citri</h1>
<div class="content">
<p>Xanthomonas citri is a species of proteobacteria. This species causes citrus canker, a localized infection which affects citrus plants worldwide. Virulence may increase if the insect Phyllocnistis citrella, a citrus leaf miner, is present.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Xanthomonas oryzae KACC10331</h1>
<div class="content">
<p>Xanthomonas oryzae contains two pathovars which cause enconomically significant diseases in rice. Xanthomonas oryzae strain KACC10331 is a plant pathogen that affects rice plants by causing leaf blight, a major problem in Asian countries where rice production occurs on an industrial scale. Losses of up to 50% of the total crop can occur in extreme cases. This pathogen enters the xylem and spreads throughout the vascular tissue of the plant, which results in wilting of the plant, or to leaf blight if the infection occurs later in development. This organism shares common virulence determinants with Xanthomonas campestris including an hrp (hypersensitive response) system that is a type III secretion system for delivery of effector molecules to the host cell cytoplasm, and avr (avirulence) genes. Other pathogenicity determinants include extracellular polysaccharide, adhesins, a type II secretion system and the extracellular enzymes secreted by it, as well as some toxins.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This organism encodes 245 proteins that are not found in other Xanthomonas pathogens.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Xenopus tropicalis</h1>
<div class="content">
<p>The Western clawed frog (Xenopus tropicalis) is a species of frog in the Pipidae family. It is the only species in the Xenopus genus to have a diploid genome. Its genome has been sequenced, making it a significant model organism for genetics that complements the related species Xenopus laevis (the African clawed frog), a widely used vertebrate model for developmental biology.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Xylella fastidiosa</h1>
<div class="content">
<p>Xylella fastidiosa is a Gammaproteobacteria that is an important plant pathogen that causes several plant diseases, including phoney peach disease in the southern United States, oleander leaf scorch and Pierce's disease (named after Newton Barris Pierce (b. 1856, d. 1916; became 60 years old), California's first professional plant pathologist) in California and Texas, and citrus X disease in Brazil.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Yersinia pestis CO92</h1>
<div class="content">
<p>Yersinia pestis was originally described by Institut Pasteur scientist Andre Yersin in 1894 in Hong Kong. It is the causative agent of plague (bubonic and pulmonary) a devastating disease which has killed millions worldwide. The organism can be transmitted from rats to humans through the bite of an infected flea or from human-to-human through the air during widespread infection. Yersinia pestis is an extremely pathogenic organism that requires very few numbers in order to cause disease, and is often lethal if left untreated. The organism is enteroinvasive, and can survive and propagate in macrophages prior to spreading systemically throughout the host. Yersinia pestis strain CO92 is a recent Orientalis subtype isolated in the USA from a patient who died of pneumonic plague after acquiring the disease from an infected cat. This strain is 50 Kbp larger than the Yersinia pestis KIM10+ genome due to numerous insertions.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Yersinia pseudotuberculosis IP 32953</h1>
<div class="content">
<p>Yersinia pseudotuberculosis was first isolated in 1883 by Malassez and Vignal and is termed pseudotuberculosis since it causes lesions in the lung that are similar to those observed during tuberculosis infection. It is ubiquitous in the environment and is a food and waterborne pathogen that affects animals as well as humans by causing gastroenteritis like Yersinia enterocolitica. Yersinia pseudotuberculosis strain IP 32953 is a fully virulent serotype I strain isolated from a human patient.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Zymomonas mobilis ZM4</h1>
<div class="content">
<p>Zymomonas mobilis has a natural habitat of sugar-rich plant saps where the bacterium ferments sugar to ethanol. The high conversion of sugars to ethanol makes this organism useful in industrial production systems, particularly in production of bioethanol for fuel. A recombinant strain of this bacterium is utilized for the conversion of sugars, particularly xylose, which is not utilized by another common sugar-fermenting organism such as yeast, to ethanol. Since xylose is a common breakdown product of cellulose or a waste component of the agricultural industry, it is an attractive source for ethanol production. Zymomonas mobilis was chosen for this process as it is ethanol-tolerant (up to 120 grams of ethanol per litre) and productive (5-10% more ethanol than Saccharomyces). This bacterium ferments using the Enter-Doudoroff pathway, with the result that less carbon is used in cellular biomass production and more ends up as ethanol, another factor that favors this organism for ethanol production. Zymomonas mobilis strain ZM4 was isolated from fermenting sugarcane juice and is wild-type and unable to utilize xylose.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Aspergillus fumigatus</h1>
<div class="content">
<p>Aspergillus fumigatus is a fungus of the genus Aspergillus, and is one of the most common Aspergillus species to cause disease in immuno-compromised individuals.\n\nA. fumigatus is a saprotrophic fungus that is widespread in nature, typically found in soil and decaying organic matter such as compost heaps, where it plays an essential role in carbon and nitrogen recycling. Colonies of the fungus produce thousands of minute grey-green conidia (2-3 ?m) from conidiophores that readily become airborne. Until recently A. fumigatus was only thought to reproduce asexually, as neither mating nor meiosis had ever been observed in the fungus. However, in 2008 it was shown for the first time that A. fumigatus possesses a fully functional sexual reproductive cycle, 145 years after its original description by Fresenius.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
A. fumigatus has a stable haploid genome. The genome sequences of three Aspergillus speciesAspergillus fumigatus, Aspergillus nidulans, and Aspergillus oryzaewere published in the journal Nature in December 2005
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acyrthosiphon pisum</h1>
<div class="content">
<p>Acyrthosiphon pisum, also known as the pea aphid, is an insect responsible for hundreds of millions of dollars of crop damage every year. A. pisum is the primary aphid species used in laboratory and genetic studies. Aphids are intensively studied as a model for understanding bacterial endosymbiosis, phenotypic plasticity, clonal vs. sexual reproduction, and the development of resistance to pesticides. Studying the genome of A. pisum will provide the tools to identify targets for biological control and a resource for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acaryochloris marina MBIC11017</h1>
<div class="content">
<p>Acaryochloris marina MBIC11017 was isolated from algae from the coast of the Palau Islands in the western Pacific. This organism will be used for comparative analysis with other photosynthetic microorganisms.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acholeplasma laidlawii PG-8A</h1>
<div class="content">
<p>This is the first genome project for Acholeplasmataceae family. \nAcholeplasma laidlawii is found in soil, compost, wastewaters, cell cultures as well as in human tissues and in many animal species (birds, bovine, goat, equine, ovine, porcine, feline, rodent, primates).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acidiphilium cryptum JF-5</h1>
<div class="content">
<p>Acidiphilium cryptum strain JF-5 was isolated under iron-reducing conditions from coal mine lake sediment and is able to grow by coupling the reduction of iron to the oxidation of glucose. This organism will be used for comparative genomics.\n\nAcidiphilium cryptum was first isolated from a supposedly pure culture of Thiobacillus ferrooxidans. This organism, a heterotrophic acidophile, has been isolated from acid mine drainage, hydrothermal vent water, and other acidic mineral environments. Acidiphilium species contain bacteriochlorophyll-a, and the genes for the L and M proteins of the photosynthetic reaction center complex have been amplified, indicating that members of this genus are capable of photosynthesis.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acidithiobacillus ferrooxidans ATCC 23270</h1>
<div class="content">
<p>This organism was isolated from bituminous coal mine effluent. The genome sequence of this organism will provide information on the production and regulation of proteins involved in iron-oxidation and will be used for comparative analysis with other iron-oxidizing bacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acidithiobacillus ferrooxidans ATCC 53993</h1>
<div class="content">
<p>Acidithiobacillus ferrooxidans, formerly Thiobacillus ferrooxidans, is an acidophilic iron-oxidizing bacterium that is able to grow at a pH range of 1.3 to 4.0. This organism is often identified in acid mine drainage and as a member of microbial consortia involved in bioleaching. Bioleaching is an increasingly important method of recovering important metals, such as copper, nickel, and uranium, from low-grade ore. Acidithiobacillus ferrooxidans has been identified as the dominant microorganism in percolation and continuous flow mineral leaching bioreactors. \n\nAcidithiobacillus ferrooxidans ATCC 53993 was derived from the type strain DSM 2705. The genome sequence of this organism will provide information on the production and regulation of proteins involved in iron-oxidation and will be used for comparative analysis with other iron-oxidizing bacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acidothermus cellulolyticus strain 11B</h1>
<div class="content">
<p>Acidothermus cellulolyticus is a thermotolerant cellulolytic actinomycete that was originally isolated from an acidic hot spring in Yellowstone National Park. A number of thermostable endoglucanases are produced by this organism, which are useful for degrading cellulose in the production of ethanol as a biofuel. An endoglucanase has been patented and is being studied.\n\nThis strain (11B; ATCC 43068) is the type strain for the species. The genome sequence from this organism will provide information on the regulation and production of potentially useful enzymes.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acidovorax sp. JS42</h1>
<div class="content">
<p>Acidovorax sp. JS42, formerly Pseudomonas sp. JS42, was isolated from nitrobenzene-contaminated sediment and is capable of using 2-nitrotolulene as a sole carbon and energy source. 2-nitrotolulene, a nitroaromatic compound, is used in the manufacture of dyes, pigments and explosives. Nitroaromatic compounds, which contain an aromatic ring with one or more nitro groups attached, are a significant contaminant in industrial soils. Acidovorax sp. JS42 degrades 2-nitrotolulene by first removing the nitro moiety producing 3-methylcatechol. The enzyme involved in this process, 2-nitrotolulene dioxygenase, has been purified and characterized. The genome sequence of this organism will provide information on the production and regulation of this enzyme.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acinetobacter baumannii AB0057</h1>
<div class="content">
<p>Acinetobacter baumannii AB0057 is a multidrug-resistant isolate from a bloodstream infection of a patient at Walter Reed Army Medical Center.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acinetobacter baumannii AB307-0294</h1>
<div class="content">
<p>Acinetobacter baumannii AB307-0294 was isolated from the blood of a hospitalized patient in Buffalo, NY. This is a multi-drug resistant strain and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acinetobacter baumannii strain ACICU</h1>
<div class="content">
<p>Acinetobacter baumannii strain ACICU (also called H34)was isolated from an outbreak in an intensive care unit in Rome. This strain will be used for comparative analysis with other Acinetobacter baumannii strains.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acinetobacter baumannii ATCC 17978</h1>
<div class="content">
<p>Acinetobacter baumannii ATCC 17978 was isolated from a case of meningitis and will be used for comparative analysis with other Acinetobacter species.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acinetobacter baumannii AYE</h1>
<div class="content">
<p>This strain is responsible for community-acquired infections and is highly resistant to antibiotics. This strain will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Acinetobacter baumannii SDF</h1>
<div class="content">
<p>This strain is responsible for community-acquired infections, and is sensitive to antibiotics. It was isolated from body lice collected from homeless people living in France. Since the louse's internal body is usually sterile, the presence of this bacteria can only be due to cryptic bacteremic episodes. This strain is susceptible to antibiotics while strain AYE is highly resistant.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Actinobacillus pleuropneumoniae L20</h1>
<div class="content">
<p>Actinobacillus pleuropneumoniae  is an obligate parasite of pigs. The bacterium colonizes and invades the porcine lower respiratory tract and can survive phagocytosis by macrophages, eventually resulting in the death of the macrophage cell. Three RTS toxins, which result in cell damage, are produced by this organism.\n\nThis strain is being sequenced for comparative genome analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Actinobacillus pleuropneumoniae serovar 7 AP76</h1>
<div class="content">
<p>Actinobacillus pleuropneumoniae serovar 7 str. AP76 is a clinical isolated from a pig lung. This strain will be used for comparative genome analysis.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Actinobacillus succinogenes 130Z</h1>
<div class="content">
<p>Actinobacillus succinogenes. Originally isolated from bovine rumen, this species is one of the most promising succinate producers known. It can utilize a variety of sugars and produce a very high volume of succinate, which is used by the pharmaceutical industry. This organism has been improved over time by growth on fermentable carbon and sodium monofluoroacetate and selection of fluoroacetate-resistant mutants which have a higher succinate yield. The succinate yield has also been improved via a metabolic engineering approach, by cloning and overproducing the PEP-carboxykinase gene which catalyzes the addition of carbon dioxide to PEP to form oxalacetate and can work physiologically in both directions.\n\nThis strain is being used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Aeromonas hydrophila ATCC 7966</h1>
<div class="content">
<p>Aeromonas hydrophila is an enviromental bacterium that is often found in aquatic habitats, but can also be found contaminating food products. It causes a variety of diseases in both cold-blooded and warm-blooded organisms. The bacterium is becoming a problematic pathogen in humans, where it causes gastroenteritis and septicemia, mainly due to the development of antibiotic resistance by this organism. It mainly affect immunocompromised individuals, but also young children and the elderly. One of the major virulence factors is aerolysin, a toxin that is produced and secreted by the cell via a type II secretion apparatus. The toxin binds to high-affinity receptors and undergoes oligomerization to form a heptameric pore-forming complex which allows passage of small molecules in the plasma membrane, resulting in permeabilization of the cell, cell death, and eventually tissue destruction. Other virulence functions include a surface layer which inhibits complement-mediated killing, type IV pili for attachment, and a set of extracellular proteases which can cause tissue damage.\n\nThis strain is the type strain for the species and was isolated from canned milk from the United States.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Aeromonas salmonicida A449</h1>
<div class="content">
<p>Aeromonas salmonicida is the causal agent of furunculosis. It affects a large number of fresh and saltwater salmonid species. Although it was discovered nearly 100 years ago, it is not known how the diesase is spread. Furunculosis is temperature sensistive, with acute cases occurring when the water is above 20oC and chronic cases developing at temperatures below 13oC. The acute form of the disease causes the fish to turn a dark color and stop eating. The chronic form tends to have a lower mortality, but the infected fish act as carriers.\n\nAeromonas salmonicida A449 was isolated from a brown trout with furunculosis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Agrobacterium radiobacter K84</h1>
<div class="content">
<p>The Agrobacterium radiobacter species is used commercially to control crown gall, a tumorogenic plant disease caused by the ubiquitous soil-borne pathogen Agrobacterium tumefaciens, which affects susceptible woody plants worldwide. This strain has been developed for worldwide commercial use to control crown gall.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Agrobacterium vitis S4</h1>
<div class="content">
<p>Agrobacterium vitis is the causative agent of crown gall disease in grapes. Disease causing Agrobacterium spp. possess a tumor-inducing (Ti) plasmid, which carries genes for the formation of opines.\nAgrobacterium vitis S4 carries a Ti plasmid which produces helopine, vitopine, and rideopine.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Akkermansia muciniphila ATCC BAA 835</h1>
<div class="content">
<p>Akkermansia muciniphila was isolated from human feces and is able to degrade mucin. Mucin is the primary component of mucus which provides a protective barrier in the gastrointestinal tract of animals. Loss of this protective barrier may expose the gastrointestinal epithelial cells to pathogenic microorganisms or to the highly acidic gut environment. Akkermansia muciniphila is a member of the Verrucomicrobia which are also common in soil and water.\n\n Strain ATCC BAA-835 will be used for comparative genomics with other Verrucomicrobia.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Alcanivorax borkumensis SK2</h1>
<div class="content">
<p>Alcanivorax borkumensiswas isolated from a seawater sediment sample in the North Sea. The species is able to use Mihagol-S (C14,15-n-alkanes) as a principal carbon source. It uses a limited number of organic compounds, including aliphatic hydrocarbons, volatile fatty acids, and pyruvate.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain SK2 is the type strain for the species, and is used for comparative genomics.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Aliivibrio salmonicida LFI1238</h1>
<div class="content">
<p>Aliivibrio salmonicida, formerly Vibrio salmonicida, a psychrophile, is the causative agent of Hitra disease, also called cold water vibriosis, in Atlantic salmon and rainbow trout. This disease, a generalized infection of the blood, occurs primarily in late autumn, winter, or early spring when water temperatures are below 15oC.\n\n The genome sequence of strain LFI1238 will provide information on infectivity and pathogenicity of this organism.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Alkalilimnicola ehrlichii MLHE-1.</h1>
<div class="content">
<p>Alkalilimnicola ehrlichei is capable of growth with arsenite [As(III)] as the electron donor with nitrate as electron acceptor. It may be responsible for production of arsenate [As(V)] in anoxic lake bottom.\n\nAlkalilimnicola ehrlichii MLHE-1 is chemoautotrophic strain. It was isolated from Mono Lake in California, which contains arsenic and has high pH and salt concentrations.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Alkaliphilus metalliredigens QYMF</h1>
<div class="content">
<p>Alkaliphilus metalliredigens is an alkiphilic, moderately halophilic metal-reducing bacterium isolated from borax leachate ponds. It can reduce iron and other metals under alkaline conditions.\nAlkaliphilus metalliredigens strain QYMF is the type strain.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Alkaliphilus oremlandii OhILAs</h1>
<div class="content">
<p>The genome of this organism will provide information on the production and regulation of these and other genes involved in arsenate reduction.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Alteromonas macleodii Deep ecotype</h1>
<div class="content">
<p>Alteromonas macleodii is an obligate marine bacterium isolated from both surface and deep seawater and often associated with particulate matter. \nAlteromonas macleodii 'Deep ecotype' was isolated from seawater obtained at a depth of 3500 meters from the Urania Basin in the Mediterranean Sea by plating. The genome sequence of this isolate will be useful in understanding barophilic adaptation in bacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Anabaena variabilis ATCC 29413</h1>
<div class="content">
<p>Anabaena variabilis is a filamentous heterocyst-forming cyanobacterium that fixes nitrogen and CO2 using the energy of sunlight via oxygen-evolving plant-type photosynthesis. In addition, this organism has been studied extensively for the production of hydrogen using solar energy. The ATCC 29413 is a wild-type strain which differs from the closely related Anabaena sp. PCC7120 in the heterocyst formation system in that this strain does not have a programmed gene rearrangement at the hupL locus.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Anaerocellum thermophilum DSM 6725</h1>
<div class="content">
<p>Anaerocellum thermophilum was isolated from a hot spring on the Kamchatka peninsula in Russia. This organism produces thermostable enzymes which can degrade cellulose and starches and may be of use in industry. Anaerocellum thermophilum DSM 6725 is the type strain and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Anaeromyxobacter dehalogenans 2CP-C</h1>
<div class="content">
<p>Anaeromyxobacter dehalogenans were originally isolated by enrichment and isolation of single plate-grown colonies, and was the first pure culture of myxobacteria able to grow anaerobically. The unique physiological characteristics of this organism include the ability to use ortho-substituted mono- and dichlorinated phenols, nitrate, H2 and fumarate as terminal electron acceptors.\nAnaeromyxobacter dehalogenans Strain 2CP-C . \nThe 2CP-C strain differs from other isolates based on the combination of its ability to use 2,5-dichlorophenol and its inability to use nitrate as a strong electron acceptor. It can also dechlorinate 2,4-dichlorophenol and 2,4,6-trichlorophenol much faster than other isolates.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Anaeromyxobacter dehalogenans Strain 2CP-1</h1>
<div class="content">
<p>Anaeromyxobacter dehalogenans were originally isolated by enrichment and isolation of single plate-grown colonies, and was the first pure culture of myxobacteria able to grow anaerobically. The unique physiological characteristics of this organism include the ability to use ortho-substituted mono- and dichlorinated phenols, nitrate, H2 and fumarate as terminal electron acceptors.\n\nAnaeromyxobacter dehalogenans Strain 2CP-1 . This strain (2CP-1; ATCC BAA-258) is the type strain for the species and was isolated from stream sediment near Lansing, Mich.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Anaplasma marginale Florida</h1>
<div class="content">
<p>Anaplasma marginale is the causative agent of bovine anaplasmosis and is an obligate intracellular pathogen. Transmission occurs via an arthropod vector (tick) and is a major problem in tropical regions where it causes severe morbidity in cattle populations. The bacterium lives intracellularly within membrane-bound vesicles in the host erythrocyte, a cell type which it specifically targets, resulting in severe anemia for the host organism. These bacteria express a number of major surface proteins (MSPs) which are adhesins specific for either the erythrocyte or the tick.\nAnaplasma marginale str. Florida. was isolated from pooled blood samples from naturally infected cattle in Florida. The Florida strain is virulent and non-tick transmitted. This strain will be used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Anaplasma phagocytophilum HZ</h1>
<div class="content">
<p>Anaplasma phagocytophilum. This organism is a tick-borne (Ixodes spp.) obligate intracellular pathogen that infects humans and causes human granulocytic anaplasmosis as well as infecting several other types of animals. It specifically attaches to and is internalized by neutrophils (polymorphonuclear leukocytes). The bacterium persist within vacuoles intracellularly. This organism produces a number of pathogenic factors that aid virulence.\n\nAnaplasma phagocytophilum strain HZ was isolated from a patient in New York, USA, in 1995</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Anoxybacillus flavithermus WK1</h1>
<div class="content">
<p>Anoxybacillus flavithermus, formerly Bacillus flavothermus, was isolated from a hot spring in New Zealand. This organism can tolerate a pH of up to 9.0.\nAnoxybacillus flavithermus WK1 is the type strain and will be used for comparative analysis\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Arcobacter butzleri RM4018</h1>
<div class="content">
<p>Arcobacter butzleri is a member of the family Campylobacteraceae, related to the human pathogenic Campylobacter spp. and the environmental Sulfurospirillum spp. A. butzleri is mainly found in livestock and water/marine environments and can cause gastroenteritis and bacteremia in humans.\nArcobacter butzleri RM4018 was isolated from a human patient with gastroenteritis.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Aromatoleum aromaticum EbN1.</h1>
<div class="content">
<p>First isolated from sediment contaminated with ethylbenzene, this organism is capable of growing on crude oil, using polar aromatic compounds such as benzoate under both oxic and anoxic conditions. It is currently being evaluated for use in aromatic hydrocarbon bioremediation projects.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Arthrobacter FB24</h1>
<div class="content">
<p>Arthrobacter is a member of the actinomycetes, common in soils. These organisms are nutritionally diverse and some strains are able to degrade pesticides and aromatic compounds.\nArthrobacter sp. FB24. Arthrobacter sp. FB24 was isolated from soil contaminated with heavy metals and aromatic solvents. This strain has a high degree of tolerance to chromium and other metals, may be radiation resistant and is a hydrocarbon degrader. This strain is used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Arthrobacter aurescens TC1</h1>
<div class="content">
<p>Arthrobacter aurescens Strain TC1 was isolated from a South Dakota spill site soil that contained high concentrations (up to 29,000 microg/ml) of atrazine, as described by Strong et al. (Appl. Environ. Microbiol., 2002, 68:5973-5980). This strain is capable of using atrazine as the sole source of nitrogen, carbon, and energy. It is able to metabolize 23 diverse s-triazine ring compounds, which are commonly found in major sources of environmental contamination such as pesticides, plastics, and dyes.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Arthrobacter chlorophenolicus A6</h1>
<div class="content">
<p>Arthrobacter chlorophenolicus A6 (DSM 12829) was isolated from soil at Fort Collins, Colorado and is able to use 4-chlorophenol as a sole source of carbon and energy. This organism can degrade 4-chlorophenol in soil at temperatures ranging from 5 to 28 oC making it a good candidate for bioremediation. The genome sequence of this organism will help elucidate the genes involved in the bioremediation of polychlorinated compounds.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Aster yellows witches-broom phytoplasma AYWB</h1>
<div class="content">
<p>Aster yellows witches'-broom phytoplasma belongs to a group of mycoplasma-like organisms that infect plants and insects. Phytoplasmas are insect-transmitted plant pathogens that cause economically important diseases in many plant species, including vegetables, ornamentals, and fruit crops. Aster yellows is a sporadic disease of vegetable crops and is transmitted by the aster leafhopper. Symptoms include a yellowing of the leaves and reduced leaf size, which leads to reduced plant size and lower yields.\nAster yellows witches'-broom phytoplasma strain AYWB will be used for comparative analysis with other phyplasmas</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Azoarcus BH72</h1>
<div class="content">
<p>This organism is a nitrogen-fixing obligate endophyte first isolated from the roots of Kaller grass in the Punjab region of Pakistan. These organisms are never found in root-free soil and colonize root tissues without causing disease. This strain has been shown to express genes involved in nitrogen fixation in root tissue and is thought to provide significant amounts of nitrogen to its host plant. Azoarcus.sp. BH72 is being used to study the mechanisms and regulation of microbe-plant interactions and for comparative studies with other pathogenic and nonpathogenic plant bacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus amyloliquefaciens strain FZB42</h1>
<div class="content">
<p>Bacillus amyloliquefaciens strain FZB42 was isolated from soil, and is a member of a group of free-living soil bacteria known to promote plant growth and suppress plant pathogens. The genome sequence of this organism will allow analysis of the production</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus anthracis Ames 0581</h1>
<div class="content">
<p>This strain carries the anthrax toxin plasmid pXO1 but not the capsule plasmid pXO2 and is therefore avirulent but toxigenic. It is the counterpart to the Pasteur strain that carries pXO2 but not pXO1. This strain is often used for vaccine development.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus cereus AH187</h1>
<div class="content">
<p>This strain was isolated from the vomit of a person having eaten cooked rice by the Public Health Laboratory Service (PHLS), London, UK. The isolate was associated with an emetic outbreak in 1972. The strain is positive for cereulide (emetic toxin) formation, is negative for the HBL hemolytic enterotoxin and cytotoxin K, but carries the non-hemolytic NHE enterotoxin. The strain has been extensively characterized as part of a study recently published by Ehling-Schulz et al. (2005 Microbiology 151, 183-197), in which this isolate was characterized as sequence type 26 using the MLSTDB scheme that is commonly used to type the B. cereus group.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus cereus AH820</h1>
<div class="content">
<p>This strain was isolated in October 1995 in Akershus, Norway, from the periodontal pocket of a 76 year old female patient with marginal periodontitis and was further analyzed by Helgason et al. (2000 J Clin Microbiol. 38, 1615-1622). B. cereus AH820 resides very close to the B. anthracis cluster using MLST analyses, and is 100% identical in all 7 loci to 3 other periodontal B. cereus isolates, suggesting that these human pathogens form a distinct clade within the group.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus cereus ATCC 10987</h1>
<div class="content">
<p>Bacillus cereus. This organism is a soil-dwelling opportunistic pathogen that causes food poisoning in infected individuals. There are two forms of food poisoning that occur, one is rapid onset (emetic) and the other is late onset (diarrheal).\nThis strain was isolated in Canada during a study of cheese spoilage. It contains a plasmid, pBc10987, that is similar to Bacillus anthracis toxin plasmid pXO1, but lacks the anthrax toxin genes.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus cereus B4264</h1>
<div class="content">
<p>This strain was isolated in 1969 from a case of fatal pneumonia in a male patient. B. cereus B4264 was cultured from the blood and the pleural fluid. This isolate was negative for all Laboratory Response Network (LRN) PCR assays that target pXO1 and pXO2. The isolate belongs to MLST type 89 using the MLSTDB scheme. Interestingly, this sequence type appears to be unrelated to B. anthracis. The clinical presentation resembles that of B. cereus G9241 and B. cereus 03BB102, and access to the draft genome of this isolate will allow for the comparison of strains that causes similar disease but have different genetic backgrounds and potential virulence genes.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus cereus G9842</h1>
<div class="content">
<p>Bacillus cereus is a soil-dwelling opportunistic pathogen that causes food poisoning in infected individuals. There are two forms of food poisoning that occur, one is rapid onset (emetic) and the other is late onset (diarrheal).\nThis strain was isolated from stool samples from an outbreak that involved three individuals in Nebraska (1996). The isolate was characterized by MLST typing using the MLSTDB scheme and is sequence type 56. Interestingly, sequence type 56 is quite unrelated to the major clade of pathogenic B. cereus isolates and will provide a great insight into its pathogenic potential through comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus cereus Q1</h1>
<div class="content">
<p>This strain is non-pathogenic, and has potential for industrial use</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus cereus ZK</h1>
<div class="content">
<p>This strain was isolated from a swab of a dead zebra carcass in April, 1996</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus licheniformis DSM 13</h1>
<div class="content">
<p>Bacillus licheniformis is a soil-dwelling endospore-forming microbe similar to other Bacilli. This bacterium is used extensively in the industrial production of important enzymes such as proteases, penicllinases, and amylases as well as smaller compounds like the antibiotic bacitracin and various organic metabolites. This is the type strain of the species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacillus pumilus SAFR-032</h1>
<div class="content">
<p>Bacillus pumilus is a naturally occurring, uniquitous soil microorganism. Commonly found in a variety of food commodities, some strains have developed an increased tolerance to gamma irradiation. Bacillus pumilus SAFR-032 was isolated from the Spacecraft Assembly Facility at NASA Jet Propulsion Laboratory and produces spores highly resistant to UV radiation. This strain will be used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacteroides fragilis YCH46</h1>
<div class="content">
<p>This organism can become an opportunistic pathogen, infecting anywhere in the body and causing abcess formation. Enterotoxigenic Bacterioides fragilis (ETBF) is associated with diarrheal diseases. This strain was isolated from a patient with septicemia in Japan. Virulence functions include numerous capsular polysaccharide biosynthesis systems that are controlled by invertible promoters that are recombined by a site-specific recombinase, Mpi, that works at a set of inverted repeats found within the chromosome. This system allows a vast repertoire of variable antigens to be expressed, thereby aiding evasion of the immune system. This organism also carries a potentially conjugative plasmid, as well as a number of conjugative and mobilizable transposons.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bacteroides vulgatus ATCC 8482</h1>
<div class="content">
<p>Bartonella bacilliformis is an obligate intracellular parasite, which infects red blood cells. Bartonella bacilliformis is the causative agent of Carrion's disease, also called Oroya fever. This disease is found in the western Andes of South America, and is transmitted through an insect vector. This strain (ATCC 35685) is the neotype strain and will be used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bartonella tribocorum CIP 105476</h1>
<div class="content">
<p>Bartonella species are facultative intracellular pathogens infecting humans and other animals. Transmission of these organisms is often through an insect vector. Once in a host, this intracellular pathogen is internalized by an actin-dependent mechanism, and primarily targets endothelial cells, although other cells can be infected. Bartonella tribocorum CIP 105476 was isolated from the blood of wild rats. This is the type strain and will be used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The proliferation of the vascular endothelium (bacillary angiomatosis) is characterisitic of Bartonella infection and results in multiplication of the bacterium's host cells. Infected macrophages are stimulated to release vascular endothelial growth factor (VEGF) and interleukin 1 beta, both of which promote angiogenesis. Endothelial cells are also stimulated to grow and divide by direct contact with bacterial cells. In addition, programmed cell death (apoptosis) of endothelial cells is inhibited, combatting a common mechanism eukaryotic cells use to deal with bacterial infection.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Baumannia cicadellinicola Homalodisca coagulata</h1>
<div class="content">
<p>This newly discovered organism is an obligate endosymbiont of the leafhopper insect Homalodisca coagulata (Say), also known as the Glassy-Winged Sharpshooter, which feeds on the xylem of plants. The insect causes devastation to grape crops and may affect other grasses as it is a vector for the bacterial pathogen, Xylella fastidiosa, which can cause disease in grapevines. It is suspected that this genome will have undergone massive reduction, and pulsed-field gel electrophoresis suggests the genome size is only 671 kb. Presumably, the genes that have been retained will strongly reflect the nutritional co-dependence requirements of the bacterium and the host.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Beijerinckia indica ATCC 9039</h1>
<div class="content">
<p>Beijerinckia indica produces a water-soluble extracellular polysaccharide that forms a biopolymer useful for a wide range of applications to textile printing, oil and adhesive industries. This organism belongs to the Beijerinckiaceae family, which includes both methanotrophic (bacteria able to oxidize methane) and nonmethanotrophic species (like Beijerinckia indica subsp. indica). Sequencing of strains from this family presents an ideal opportunity for a comparative genomic study into the evolution and biochemistry of obligate methanotrophy. This is the type strain for the species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bifidobacterium longum infantis ATCC 15697</h1>
<div class="content">
<p>Bifidobacterium naturally colonize the human gastrointestinal tract (GIT) and are important for establishing and maintaining homeostasis of the intestinal ecosystem to allow for normal digestion. Their presence has been associated with beneficial health effects, such as prevention of diarrhea, amelioration of lactose intolerance, or immunomodulation. Bifidobacterium longum is found in adult humans and formula fed infants as a normal component of gut flora. This strain was isolated from human infant feces and will be used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bordetella avium 197N</h1>
<div class="content">
<p>Bordetella avium infects the respiratory tract of birds, and causes bordetellosis in commercially important animals such as turkeys, resulting in devastating losses every year due to secondary infections. This strain is a spontaneous nalidixic acid-resistant derivative of virulent strain 197.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bordetella petrii</h1>
<div class="content">
<p>Unlike other members of the genus, this organism is not known to be associated with humans or other warm-blooded animals. Bordetella petrii also differs from other Bordetella species in that it is a facultative anaerobe.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Borrelia duttonii Ly</h1>
<div class="content">
<p>Borrelia duttonii is the causative agent of tick-borne relapsing fever in east Africa. This disease is endemic in much of east Africa and is one of the top 10 diseases associated with deaths in children under the age of five. This disease is transmitted by the bite of the soft-bodied tick Ornithodoros moubata. Ticks become infected with Borrelia duttonii while feeding on an infected rodent. Borrelia then multiplies rapidly, causing a generalized infection throughout the tick. Borrelia duttonii Ly was isolated from a 2-year-old girl with tick-borne relapsing fever in Tanzania. This organism will be used for comparative analysis with other Borrelia species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Borrelia hermsii DAH</h1>
<div class="content">
<p>Borrelia hermsii is the causative agent of tick-borne relapsing fever in the western United States and Canada. Ticks become infected with Borrelia spp. while feeding on an infected mammal, usually a rodent or squirrel. This strain was isolated from a case of relapsing fever in western Washington. This organism will be used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Borrelia recurrentis A1</h1>
<div class="content">
<p>Borrelia recurrentis is the causative agent of louse-borne relapsing fever. Prior to World War II, this organism was responsible for large disease outbreaks with a mortality of up to 40% in much of the world. Currently this disease is limited to parts of Africa, China, and Peru. B. recurrentis is transmitted when infected human body lice (Pediculus humanus) are crushed and their fluids contaminate mucous membranes or breaks in the skin. This strain was isolated from adult patient with louse-borne relapsing fever in Ethiopia. This organism will be used for comparative analysis with other Borrelia species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Borrelia turicatae 91E135</h1>
<div class="content">
<p>Borrelia turicatae is the causative agent of tick-borne relapsing fever in the southwestern United States. Ticks become infected with Borrelia while feeding on an infected mammal, usually a rodent or squirrel. Borrelia then multiplies rapidly, causing a generalized infection throughout the tick. While feeding, the tick passes the spirochete into a mammalian host through its infectious saliva. Humans become exposed to the infected ticks through contact with rodents or rodent nests. This strain was isolated in the United States from the soft tick Ornithodoros turicatae and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Brachyspira hyodysenteriae WA1</h1>
<div class="content">
<p>Brachyspira hyodysenteriae is the causative agent of swine dysentery, which is a severe mucohemorrhagic diarrheal disease of pigs that has economic significance for pork-producing countries. Brachyspira hyodysenteriae WA1 (ATCC 49526) is the first representative of the genus Brachyspira to be sequenced. It has been subjected to comparative genomic analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Bradyrhizobium ORS278</h1>
<div class="content">
<p>This strain was isolated from an Aeschynomene stem nodule and is photosynthetic, which is a rare trait in Rhizobium bacteria. These strains exhibit a photoheterotrophic and strictly aerobic photosynthesis. In culture, most stem isolates show the same pink coloration, while a few strains produce orange pigmentation. Pigment analyses showed that bacteriochlorophyll and spirilloxanthin, two pigments of the light harvesting system, are common to all of these photosynthetic Bradyrhizobium strains, whereas orange strains produce an additional bicyclic carotenoid, canthaxanthin (4,4'-diketo-beta-carotene). Bradyrhizobium sp. strain ORS278 produces the highest quantity of canthaxanthin of all tested photosynthetic bacteria; canthaxanthin represents 85% of its total carotenoid content.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Brucella abortus 9-941</h1>
<div class="content">
<p>This is a spontaneously attenuated strain discovered by Dr. John Buck in 1923. However, the underlying molecular or physiological mechanisms causing the loss of virulence is not well understood. Since early 1930s, this strain has been used worldwide as an effective vaccine to prevent brucellosis in cattle until it was replaced by strain RB51 during the 1990s. The main objective of this project is to identify genes associated with the virulence or lack there of, through comparison of S19 genome with that of the virulent counterparts.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Brucella ovis</h1>
<div class="content">
<p>This organism specifically affects sheep, causing ovine brucellosis which results in significant economic losses. Infection by the organism can cause infertility in rams as the bacterium preferentially colonizes the genital tract, and causes inflammation of the epididymis which leads to necrotic lesions and a decrease in sperm numbers and motility. Infection in pregnant ewes can result in premature abortion as the bacterium affects both the unborn fetus and the placenta. This bacterium, like other Brucella, is an intracellular pathogen that infects phagocytic cells such as macrophages. Bacterial cells produces a lipopolysaccharide that lacks O-antigen side chains, which results in a rough colony phenotype, and aids in differentiation as compared to other Brucella.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Brucella suis ATCC 23445</h1>
<div class="content">
<p>This organism is a swine-specific Brucella. It causes infectious abortions in animals and a systemic, febrile (feverish) illness in humans. Human infection usually occurs by accidental contact with infected animals or by ingestion of contaminated food. Brucella suis ATCC 23445 will be the second strain to be sequenced and will provide interesting insights into the evolution of this species within the genus Brucella.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Buchnera aphidicola 5A Acyrthosiphon pisum</h1>
<div class="content">
<p>This strain is found in the pea aphid, Acyrthosiphon pisum. Buchnera of A. pisum are the prime model of an obligate symbiosis. This is a resequencing project for exploring variation in the symbionts of a single host species.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Buchnera aphidicola str. Cc Cinara cedri</h1>
<div class="content">
<p>This strain is found in the cedar aphid, Cinara cedri.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Buchnera aphidicola Tuc7 Acyrthosiphon pisum</h1>
<div class="content">
<p>This strain is found in the pea aphid, Acyrthosiphon pisum. Buchnera of A. pisum are the prime model of an obligate symbiosis. This is a resequencing project for exploring variation in the symbionts of a single host species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Buchnera_sp</h1>
<div class="content">
<p>This strain is found in the pea aphid, Acyrthosiphon pisum. Its genome consists of one circular chromosome and two small plasmids pLeu and pTrp.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia cenocepacia AU 1054</h1>
<div class="content">
<p>Burkholderia cenocepacia is part of the Burkholderia cepacia complex, which consists of 9 closely related species, also known as genomovars. These organisms are commonly found in soil and all are opportunistic pathogens espcially in cystic fibrosis patients. This strain was isolated from the blood of a patient with CF. This microbe has three chromosomes.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia cenocepacia HI2424</h1>
<div class="content">
<p>This organism (HI2424; BCC1) was isolated as part of a study to find this organism in natural environments. Samples taken from fields that were planted with onoins in New York in 1999 and 2000 were examined and it was discovered that isolates were the same as ones that can infect CF patients. This microbe has three chromosomes.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia cenocepacia J2315</h1>
<div class="content">
<p>Burkholderia cenocepacia is part of the Burkholderia cepacia complex, which consists of 9 closely related species, also known as genomovars. These organisms are commonly found in soil and all are opportunistic pathogens espcially in cystic fibrosis patients. This strain is the index strain for transmission among CF patients in the UK and is a genomovar III strain of the ET12 lineage.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia cenocepacia MC0 3</h1>
<div class="content">
<p>Burkholderia cenocepacia MC0-3 was isolated from soil associated with maize roots. This strain is a group B. cenocepacia III-B strain and will be used for comparative analysis with other Burkholderia species.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia cepacia AMMD</h1>
<div class="content">
<p>This strain (genomovar VII; AMMD; ATCC BAA-244; LMG 19182) is the type strain for the species and was isolated from the rhizosphere of healthy pea plants in Wisconsin, USA, in 1985. This strain is very effective in controlling phytopathogenic Pythium species (responsible for pre- and post-emergence damping-off in peas) and Aphanomyces euteiches (responsible for root-rot and the main factor limiting pea production in the American Midwest).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia mallei SAVP1</h1>
<div class="content">
<p>Burkholderia mallei is rarely associated with human infection, and is more commonly seen in domesticated animals such as horses, donkeys, and mules where it causes glanders, a disease first described by Aristotle. This organism is similar to B. pseudomallei and is differentiated by being nonmotile. The pathogen is host-adapted and is not found in the environment outside of its host. Glanders is often fatal if not treated with antibiotics, and transmission can occur through the air, or more commonly when in contact with infected animals. This strain is non-virulent and will be used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia phymatum STM815</h1>
<div class="content">
<p>The Burkholderia phymatum forms a symbiotic relationship with tropical legumes, where it is involved with nirogen fixation in the root nodules.\nThis strain was isolated from a root nodule of Machaerium lunatum in French Guiana. It is being sequenced for comparative genome analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia phytofirmans PsJN</h1>
<div class="content">
<p>Burkholderia phytofirmans, isolated from surface-sterilized onion roots, is a plant growth-promoting bacterium. Plants inoculated with this organism have larger root systems and are more resistant to pathogens. Burkholderia phytofirmans PsJN is the type strain and will be used to provide information on the production and regulation of compounds beneficial to plants.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia pseudomallei 1106a</h1>
<div class="content">
<p>Burkholderia thailandensis is a common soil saprophyte (lives on decaying organic matter in the soil). This bacterium is very similar to the human and animal pathogen Burkholderia pseudomallei but appears to be avirulent. Distinguishing the two organisms is very difficult and may depend on using monoclonal antibodies to detect differences in exopolysaccharide production. This strain was originally isolated from a rice field sample in Thailand. Burkholderia thailandensis ATCC 700388 (strain E264) is the type strain and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia vietnamiensis G4</h1>
<div class="content">
<p>This strain is a proficient degrader of trichloroethylene, benzene, and tolulene and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Burkholderia xenovorans LB400</h1>
<div class="content">
<p>Burkholderia xenovorans has been found on fungi, animals, and from human clinical isolates such as from cystic fibrosis (CF) patients. It may be tightly associated with white-rot fungus, as the degadation of lignin by the fungus results in aromatic compounds the bacterium can then degrade. This organism is exceptionally capable of degradation of polychlorinated biphenyls (PCBs), which are environmental pollutants, and thus it may play a role in bioremediation of polluted and toxic sites and is studied as a model bioremediator. Strain LB400  was found in contaminated soil in upstate New York, this organism is now classified in the genus Burkholderia. This is the type strain and is being sequenced for studies in comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Caldicellulosiruptor saccharolyticus DSM 8903</h1>
<div class="content">
<p>Caldicellulosiruptor saccharolyticus DSM 8903 was isolated from a hot spring in New Zealand. This is the type strain and genome of this organism should provide even further information on the production and regulation of cellulolytic enzymes.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Caldivirga maquilingensis.</h1>
<div class="content">
<p>Caldivirga maquilingensis was isolated from an acidic hot spring in the Philippines. It is a microaerophilic heterotroph and is able to use sulfur, thiosulfate, and sulfate as electron acceptors. This strain is the type strain and is being sequenced as part of a comparative genomic study of hyperthermophilic Archaea.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Campylobacter concisus 13826</h1>
<div class="content">
<p>Campylobacter concisus was first isolated from the human oral cavity in cases of gingivitis; however the role it plays in periodontal disease is unclear. Campylobacter concisus 13826 is a gastrointestinal clinical isolate and belongs to genomospecies 2. This organism will be used for comparative genomics with other Campylobacter species.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Campylobacter fetus 82-40</h1>
<div class="content">
<p>This strain (82-40) was isolated from the blood of a human patient who was having a renal transplant and is the best characterized isolate of this species.. The ratio of bloodstream infection to diarrheal illnesses for C. fetus is nearly 400-fold higher than for C. jejuni, indicating its marked propensity for invasive disease compared to C. jejuni.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Campylobacter hominis ATCC BAA-381</h1>
<div class="content">
<p>Campylobacter hominis is the only known species of Campylobacter that is not pathogenic to humans. Campylobacter hominis ATCC BAA-381 was isolated from the feces of a healthy human. This strain will be used for comparative analysis with pathogenic Campylobacter species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Campylobacter jejuni 81-176</h1>
<div class="content">
<p>Campylobacter jejuni is the leading cause of bacterial food poisoning (campylobacteriosis) in the world, and is more prevalent than Salmonella enteritis (salmonellosis). Severe health and economic problems are a result of widespread infections that affect up to 1% of the population. This strain was isolated during an outbreak of campylobacteriosis in Minnesota, USA, due to the consumption of contaminated milk.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Campylobacter jejuni 81116</h1>
<div class="content">
<p>Campylobacter jejuni subsp. jejuni 81116 is a human strain isolated from a waterborne outbreak in 1982 and now routinely used as a laboratory strain. This strain will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Campylobacter jejuni RM1221</h1>
<div class="content">
<p>This strain  was isolated from the skin of a retail chicken by the Produce Safety and Microbiology Research Unit (Western Regional Research Center, Albany, CA) and minimally passaged. This strain has unique features that distinguish it from  Campylobacter jejuni strain NCTC 11168 including unique lipooligosaccharide, capsule, and other loci, as well as the ability to colonize chicken skin and other differences.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Campylobacter lari RM2100</h1>
<div class="content">
<p>Campylobacter lari is closely related to C. jejuni and C. coli and is a foodborne pathogen that causes gastroenteritis and diarrhea. In addition this organism can cause bacteremia and septicemia in immunocompromised individuals. The bacterium is found in a variety of habitats including water environments, and seagulls are considered a reservoir for the organism. This strain is a clinical isolate and will be used for comparative analysis with other Campylobacter species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Amoebophilus asiaticus 5a2</h1>
<div class="content">
<p>Candidatus Amoebophilus asiaticus 5a2 is an obligate endosymbiont found in Acanthamoeba sp. TUMSJ-321. This bacterial strain could neither be maintained outside its original host nor transferred to other amoebae.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Azobacteroides pseudotrichonymphae genomovar CFP2</h1>
<div class="content">
<p>Candidatus Azobacteroides pseudotrichonymphae genomovar CFP2 was isolated from a single cell of the protist Pseudotrichonympha grassii which resides in the termite gut. The genome sequence of this organism will provide information on the proteins involved in endosymbiosis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Blochmannia pennsylvanicus BPEN</h1>
<div class="content">
<p>The genome shows a low GC content and accelerated rate of evolution and gene reduction.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Carsonella ruddii PV</h1>
<div class="content">
<p>Candidatus Carsonella ruddii PV is an obligate endosymbiont of the psyllid Pachypsylla venusta. The genome of Candidatus Carsonella ruddii PV, is 160 kbp and the smallest genome sequenced to date retaining genes primarily involved in amino acid synthesis and translation. This organism will be used for comparative analysis with other insect endosymbionts.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Desulfococcus oleovorans Hxd3</h1>
<div class="content">
<p>Candidatus  Desulfococcus oleovorans Hxd3 (DSM 6200) was isolated from an oil/water mixture from an oil production plant.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Desulforudis audaxviator MP104C</h1>
<div class="content">
<p>Candidatus Desulforudis audaxviator is a gram positive sulfate reducing bacteria identified in fracture water from a borehole at a depth of 2.8 km in a South African gold mine. Water from these boreholes is very old, suggesting that these bacteria have been isolated from the Earth's surface for as much as several million years. Sequencing of the genome of this strain will provide information on the development of bacteria under these conditions.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Methanoregula boonei 6A8</h1>
<div class="content">
<p>Candidatus Methanoregula boonei 6A8 a represents a group of microorganisms previously identified only by 16S rRNA analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Methanosphaerula palustris E1 9c</h1>
<div class="content">
<p>Candidatus Methanosphaerula palustris E1-9c (ATCC BAA1556, DSM 19958) is the type strain and will be used for comparative analysis with other methanogens.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Methanosphaerula palustris E1 9c</h1>
<div class="content">
<p>Candidatus Methanosphaerula palustris E1-9c  is the type strain and will be used for comparative analysis with other methanogens.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Pelagibacter ubique HTCC1062</h1>
<div class="content">
<p>This organism is used in comparative genomics. The chromosome encodes biosynthetic pathways for all twenty amino acids and many enzymatic cofactors. There appears to be no recently horizontally transferred DNA in this organism and it does not even contain transposable elements although it does encode DNA uptake genes. The bacterium can generate energy through a light-driven proteorhodopsin proton pump or via respiration. Many metabolic pathways, including the Entner-Duodoroff, TCA, and glyoxylate bypass are present. This organism also encodes genes for type II secretion and type IV pilus synthesis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Phytoplasma australiense</h1>
<div class="content">
<p>Phytoplasmas inhabit phloem (food-conducting vascular tissue) sieve elements of plants where they cause a variety of diseases. Transmission between plants is via insects that feed on the phloem. Candidatus Phytoplasma australiense was identified in diseased Chardonnay grapevines from South Australia. Symptoms include a yellowing of the leaves and reduced leaf size, which leads to reduced plant size and lower yields. This strain will be used for comparative genomics with other phytoplasmas to identify genes involved in virulence and disease transmission.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Phytoplasma mali</h1>
<div class="content">
<p>Candidatus Phytoplasma mali belongs to a group of mycoplasma-like organisms (Mollicutes) that infect plants and insects. This organism causes the economically significant apple proliferation disease in Europe. This disease is characterized by floral and phloem necrosis, wilting or general plant decline. Candidatus Phytoplasma mali can also infect cherry, apricot and plum trees.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Candidatus Sulcia muelleri GWSS</h1>
<div class="content">
<p>Candidatus Vesicomyosocius okutanii depends on sulfur-oxidizing symbiotic bacteria housed in its gill tissues for its sole nutritional support. The symbiont is transmitted vertically between generations via the clam's eggs. This anaerobic symbiosis oxidizes hydrogen sulfide as an energy source and fixes carbon dioxide into organic compounds. This strain was collected off Hatsushima island in Sagami Bay, Japan.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Carboxydothermus hydrogenoformans Z-2901</h1>
<div class="content">
<p>Carboxydothermus hydrogenoformans is extremely thermophilic and can utilize carbon monoxide as the sole carbon and energy source under anaerobic conditions with the production of carbon dioxide and hydrogen gas. This strain was isolated from a hot swamp from Kunashir Island, Russia.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Caulobacter K31</h1>
<div class="content">
<p>This organism inhabits aquatic environments and plays an important part in biogeochemical cycling of organic nutrients. This strain was isolated from a low-oxygen groundwater sample and is being sequenced for comparative genome analysis</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Caulobacter crescentus NA1000</h1>
<div class="content">
<p>Caulobacter crescentus NA1000 is a spontaneous mutant derivative of C. crescentus strain CB15. It is used by labs worldwide because it can easily be synchronized, allowing for studies of the bacterial cell cycle.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Cellvibrio japonicus Ueda107</h1>
<div class="content">
<p>Cellvibrio japonicus Ueda107 was isolated from field soil in Japan. This is the type strain and will be used for comparative analysis with other cellulolytic bacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlamydia trachomatis A HAR-13</h1>
<div class="content">
<p>This species causes infection that leads to blindness and sexually transmitted diseases in humans. There are 15 serovariants that preferentially cause disease in either the eye or the urogenital tract. The trachoma (infection of the mucous membrane of the eyelids) biovars are noninvasive and can cause blinding trachoma (variants A, B, Ba, and C), or sexually transmitted diseases (variants, D, E, F, G, H, I, J, and K). The lymphogranuloma venereum biovars (variants L1, L2, and L3) can cross the epithelial cells of mucous membranes and then travel through the lymphatic system where they multiply within mononuclear phagocytes found within the lymph nodes.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Chlamydia trachomatis A/HAR-13  is a serovar A strain.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlamydophila felis Fe C-56</h1>
<div class="content">
<p>Chlamydophila felis strain Fe/C-56 was isolated from a cat with conjunctivitis. The genome sequence of this organism will provide information on the genetic adaptations due to host specificity.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlamydophila pneumoniae CWL029</h1>
<div class="content">
<p>Isolated in the USA before 1987 and epidemiologically distinct from Chlamydophila pneumoniae strain AR39</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlamydophila pneumoniae CWL029</h1>
<div class="content">
<p>Isolated in the USA before 1987 and epidemiologically distinct from Chlamydophila pneumoniae strain AR39</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlamydophila pneumoniae J138</h1>
<div class="content">
<p>Acute infection with Chlamydophila pneumoniae can result in pneumonia, bronchitis and pharyngitis, while chronic infections can lead to lung cancer, stroke, coronary heart disease and chronic obstructive pulmonary disease in humans. Chlamydophila pneumoniae strain J138 was isolated in Japan in 1994 from the pharyngeal mucosa of a 5 year old male patient with acute bronchitis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlamydophila pneumoniae TW 183</h1>
<div class="content">
<p>This is the type strain for Chlamydophila pneumoniae. An ocular isolate first isolated in 1965 from a child's conjunctiva (the membranes lining the eyelid) during a trachoma vaccine trial in Taiwan. However, it naturally causes respiratory rather than ocular infections.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlorobium chlorochromatii CaD3</h1>
<div class="content">
<p>Chlorobium chlorochromatii is an green sulfur bacterium isolated from low temperature anaerobic fresh water habitats. This organism grows as part of a phototrophic consortium which consists of a colorless central bacterium surrounded by chlorosome-containing green sulfur epibionts. This organism is able to be grown in pure culture therefore is not an obligate symbiont. Thos organism is an obligate anaerobic phototroph able to fix nitrogen. Chlorobium chlorochromatii CaD3 was isolated from a phototrophic consortia obtained from a freshwater lake in Germany.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlorobium limicola DSM 245</h1>
<div class="content">
<p>Chlorobium limicola has been used to model a variety of enzyme and reaction center pathways, including ATP-citrate lyase, isocitrate dehydrogenase, and the reverse Krebs cycle used in photosynthesis. Now called the "reductive carbolic acid cycle", this was found to be the sole carbon dioxide assimilation pathway in other green sulfur bacteria since its discovery. Chlorobium limicola Strain DSM 245 is the type strain for the species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlorobium phaeobacteroides DSM 266</h1>
<div class="content">
<p>Chlorobium phaeobacteroides contains several carotenoids, predominated by isorenieratene and beta-isorenieratene. These are necessary for light harvesting and photoprotection, which are important functions in photosynthesis. This species demonstrates a novel type of carotenoid-mediated photoadaptation induced by the change of theta-end and beta-end carotenoid groups, under various light intensities.  Chlorobium phaeobacteroides strain DSM 266 was isolated from a lake in Norway. This is the type strain for the species and will be used for comparative genomics.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chloroflexus Y 400 fl</h1>
<div class="content">
<p>Chloroflexus live in hot springs of neutral to high pH and grow in mats, typically as the lowest layer in the mat with cyanobacteria above them, or as filamentous tendrils.\nChloroflexus sp Y-400-fl was isolated from an alkaline hot spring and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chloroherpeton thalassium ATCC 35110</h1>
<div class="content">
<p>Chloroherpeton thalassium ATCC 35110 is the type strain and will be used for comparative genomics with other green sulfur bacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chromohalobacter salexigens DSM 3043</h1>
<div class="content">
<p>Chromohalobacter salexigens DSM 3043 was first isolated from a solar salt facility on Bonaire Island, Netherlands Antilles. This is the type strain for the species and will be used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Citrobacter koseri ATCC BAA-895</h1>
<div class="content">
<p>Citrobacter koseri, previously known as Citrobacter diversus, Levinea diversus or Levinea malonatica resides in a wide range of environments, including soil, water and food products. It is an occasional inhabitant of human and animal intestines, but is mainly characterized as being a causative agent of neonatal meningitis with an extreme high rate of multiple brain abscess initiations and a concomitant high moratility rate. Citrobacter koseri ATCC BAA-895 is a clinical isolate from a human infant and will be used for comparative analysis with other Citrobacter species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clavibacter michiganensis sepedonicus</h1>
<div class="content">
<p>A wild type strain that harbors two plasmids, pCM1 and pCM2. The presence of both plasmids is not required to colonize tomato plants, but is necessary to cause disease.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium botulinum A3 Loch Maree</h1>
<div class="content">
<p>Clostridium botulinum A3 str. Loch Maree was isolated from duck liver paste during a botulism outbreak at a hotel in the Scottish highlands in 1922. This was the first and worst outbreak of botulism in the United Kingdom. Clostridium botulinum A3 str. Loch Maree will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium botulinum A ATCC 19397</h1>
<div class="content">
<p>Clostridium botulinum A ATCC 19397 is a stock, type A toxin-producing, laboratory strain of known toxicity and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium botulinum A Hall</h1>
<div class="content">
<p>Clostridium botulinum A strain Hall was received at Fort Detrick from Harvard University in 1947. The strain is presumably one from Dr. Ivan Hall's collection, but the exact strain number has been lost. This strain produces high amounts of type A toxin and will be used for comparative analysis.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium botulinum B1 Okra</h1>
<div class="content">
<p>Clostridium botulinum B1 str. Okra is a commonly used strain for studies involving toxin production and regulation. The growth requirements for this strain have also been extensively studied. Clostridium botulinum B1 str. Okra will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium botulinum B Eklund 17B</h1>
<div class="content">
<p>Clostridium botulinum B str. Eklund 17B is a nonproteolytic botulism neurotoxin B strain. This strain was isolated from marine sediments taken off the coast of Washington, USA and was not associated with botulism. Clostridium botulinum B str. Eklund 17B will be used for comparative analysis with other Clostridium botulinum strains.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium botulinum E3 Alaska E43</h1>
<div class="content">
<p>This strain was probably isolated from salmon eggs associated with a foodborne case of botulism in Alaska, however the exact details are not available. Clostridium botulinum E3 str. Alaska E43 will be used for comparative analysis with other Clostridium botulinum strains.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium botulinum F Langeland</h1>
<div class="content">
<p>Clostridium botulinum F strain Langeland was identified in 1958 from home-prepared liver paste involved in an outbreak of foodborne botulism on the island of Langeland, in Denmark. This strain will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium cellulolyticum H10</h1>
<div class="content">
<p>. The genome sequence of Clostridium cellulolyticum strain H10  will provide information on the production, export and regulation of cellulosome proteins and other proteins involved in degrading cellulose.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium difficile  630</h1>
<div class="content">
<p>This strain is the epidemic type X variant that has been extensively studied in research and clinical laboratories. It produces both toxin A, and B.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium novyi NT</h1>
<div class="content">
<p>This strain (NT) was created so that it lacks the lethal toxin. Injection of spores of this organism into mice harboring tumors resulted in an increased immune response and destruction of the tumor tissue in a method known as COBALT (combination bacteriolytic therapy) when the spores along with chemotherapeutic agents or radiation are co-administered. The anaerobic bacteria preferentially target hypoxic cancer tissue and stimulate immune responses to that tissue.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium perfringens ATCC 13124</h1>
<div class="content">
<p>This strain is a type A isolate and is also the type strain of the species. It is used as a quality control organism in clinical and laboratory diagnostics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium perfringens SM101</h1>
<div class="content">
<p>This is a enterotoxin-producing food poisoning strain and will be used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium phytofermentans ISDg</h1>
<div class="content">
<p>Type strain ISDg was originally isolated from forest soil near the Quabbin Reservoir in Massachusetts. This organism plays an important industrial and ecological role in the anaerobic fermentation of cellulose and produces economically significant levels of acetate and ethanol.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Clostridium thermocellum strain ATCC 27405</h1>
<div class="content">
<p>This strain is a well studied producer of endoglucanase and several restriction endonucleases.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Corynebacterium glutamicum strain ATCC 13032</h1>
<div class="content">
<p>This organism is a well-studied soil bacterium of considerable importance in biotechnology, in particular for the fermentative production of L-amino acids for food and fodder industry. The name was originaly given for this species for its ability to produce significant quantities (>100 g per liter) of glutamic acid (glutamate), an important food enhancer that has a meaty taste and flavor. C. glutamicum is currently used commercially to produce glutamate and other amino acids (L-lysine) and compounds. The first strain of the species was isolated in 1957 by S. Kinoshita and colleagues while searching for an efficient glutamate-producer. This strain (previously known as Micrococcus glutamicus) is the original strain isolated in the late 1950's.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Corynebacterium glutamicum R</h1>
<div class="content">
<p>This strain was isolated from meadow soil in Japan and will be used for comparative analysis</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Corynebacterium jeikeium K411</h1>
<div class="content">
<p>This strain was isolated from the axilla of a bone marrow transplant patient.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Corynebacterium urealyticum DSM 7109</h1>
<div class="content">
<p>Corynebacterium urealyticum DSM 7109 was isolated from a case of cystitis. This is the type strain and will provide information on the production and regulation of the proteins involved in virulence and disease.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Coxiella burnetii G Q212</h1>
<div class="content">
<p>Coxiella burnetii G Q212 was isolated from a case of endocarditis and does not carry a plasmid. This strain is being sequenced for comparative genome analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Coxiella burnetii K Q154</h1>
<div class="content">
<p>Coxiella burnetii K Q154 was isolated from a patient with endocarditis and contains the plasmid QpRS. This strain is being sequenced for comparative genome analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Coxiella burnetii Dugway 7E9-12</h1>
<div class="content">
<p>Coxiella burnetii Dugway 5J108-111(7E9-12) was isolated from rodents in Utah. This strain contains the plasmid QpDG. Coxiella burnetii Dugway 5J108-111 is part of a comparative genome project which is sequencing a strain from each of the six genomic groups of Coxiella burnetii.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Coxiella burnetii RSA 331</h1>
<div class="content">
<p>This strain is associated with acute Q fever and was isolated from the blood of an infected patient in northern Italy in 1945.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Cupriavidus taiwanensis</h1>
<div class="content">
<p>Cupriavidus taiwanensis, also called Ralstonia taiwanensis or Wautersia taiwanensis is similar to bacteria from the Agrobacterium and Rhizobium genera that form nitrogen-fixing nodules in the roots of various leguminous plants. However, Cupriavidus taiwanensis is only distantly related to these other root nodule-forming bacteria. In Cupriavidus taiwanensis the nodule formation and nitrogen fixation genes are contained a 0.5Mb plasmid.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Cyanobacteria bacterium Yellowstone A-Prime</h1>
<div class="content">
<p>Also known as Cyanobacteria Yellowstone A-Prime, this organism represents a group of Cyanobacteria that is commonly identified in surface and subsurface layers of hot spring microbial mats. Synechococcus sp. JA-3-3Ab will be used for comparative genomics with other Cyanobacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Cyanobacteria bacterium Yellowstone B-Prime</h1>
<div class="content">
<p>Also known as Cyanobacteria Yellowstone B-Prime, this organism represents a group of Cyanobacteria that have been identified along temperature and vertical gradients in hot spring microbial mats. Synechococcus sp. JA-2-3B'a(2-13) will be used for comparative genomics with other Cyanobacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Cyanothece ATCC 51142</h1>
<div class="content">
<p>Cyanothece sp. ATCC 51142 was isolated from intertidal sands from the Gulf of Mexico in Texas. This strain will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Cyanothece PCC 7424</h1>
<div class="content">
<p>This strain was isolated from a rice field in Senegal and will be used for comparative analysis.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Cyanothece PCC 7425</h1>
<div class="content">
<p>Cyanothece are unicellular nitrogen-fixing phototrophic bacteria. These bacteria have a distinct diurnal cycle where photosynthesis occurs during the day and nitrogen-fixation occurs at night.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Cyanothece sp. PCC 7425 was isolated from a rice field in Senegal and will be used for comparative analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Cyanothece PCC 8801</h1>
<div class="content">
<p>This strain was isolated from rice fields in Taiwan and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Cytophaga hutchinsonii ATCC 33406</h1>
<div class="content">
<p>This organism is abundantly found in soil, freshwater, and marine habitats and it is motile by a gliding motility system that does not utilize flagella. Degradation of various recalcitrant organic macromolecules such as chitin, pectin, and cellulose is a hallmark of this organism.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The genome sequence of this strain will provide information on the production and regulation of proteins involved in the degradation of chitin, cellulose and other compounds.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Dechloromonas aromatica RCB</h1>
<div class="content">
<p>Dechloromonas aromatica may be used for bioremediation as it can oxidize aromatic hydrocarbon compounds, including benzene, in the absence of oxygen. Benzene is an important pollutant, and is used in many manufacturing processes and is a component of diesel fuel.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Dechloromonas aromatica strain RCB was enriched as a hydrocarbon-oxidizing chlorate-reducer from the Potomac River, Maryland, USA. This organism is the first one to have the capability of benzene oxidation in pure anaerobic culture by coupling it to nitrate reduction which is of importance due to the anaerobic environments often found in bioremediation projects. It can reduce perchlorate and chlorate to chloride.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Dehalococcoides BAV1</h1>
<div class="content">
<p>Dehalococcoides species are able to dechlorinate organic compounds such as trichlorobenzene and trichloroethene which makes them potential candidates as bioremediation agents.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Dehalococcoides sp. BAV1 was isolated from a microcosm that could dechlorinate polychlorinated ethenes. It is capable of using vinyl chloride as an electron acceptor (and in the process dechorinating the carcinogenic compound) producing ethene and inorganic chloride and using acetate as the electron donor. This organism may be useful in bioremediation of contaminated sites.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Dehalococcoides CBDB1</h1>
<div class="content">
<p>Dehalococcoides species are able to dechlorinate organic compounds such as trichlorobenzene and trichloroethene which makes them potential candidates as bioremediation agents.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
was isolated from an enriched chlorobenzene-dechlorinatng mixed culture from anoxic river sediment. It is capable of reductive dechlorination, replacing chlorine with hydrogen, of environmental pollutants such as polychlorinated dibenzo-p-dioxins and dibenzofurans (PCDDs and PCDFs), by utilizing them as electron acceptors during anaerobic respiration. Genome analysis suggests this organism has 32 reductive-dehalogenase-homologous loci, arranged as two genes, rdhA and rdhB. RdhA is thought to be the active subunit while the hydrophobic protein RhdB is presumably a membrane-anchor for RdhA. The rdh loci are typically found in the origin-proximal portion of the chromosome, and are usually transcribed in the direction of DNA replication, suggesting that they are highly expressed. These genes are also associated with genes encoding two-component and other types of regulatory systems, suggesting that these loci are under regulatory control based on environmental stimuli.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Deinococcus geothermalis DSM 11300</h1>
<div class="content">
<p>Deinococcus geothermalis is a radiation resistant bacterium. A very efficient DNA repair system enables Deinococcus geothermalis to resist gamma radiation, UV radiation, and desiccation. This organism has been identified as a primary biofilm former on paper making machinery. Biofilms can clog filters, presses, and pipelines, and damage the quality of the paper. The sequence of this organism will provide information on the mechanism and regulation of attachment and biofilm formation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain DSM 11300 was isolated from a hot spring at Agnano, Naples, Italy and will be used for comparative analysis with other radiation resistant bacteria.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Delftia acidovorans SPH-1</h1>
<div class="content">
<p>This species was isolated from soil enriched with acetamide in Delft in the Netherlands. It was originally classified as Comamonas acidivorans in 1926, and redescribed and characterized in 1999. This organism is found in soil, sediment, activated sludge, crude oil, water, and various clinical samples. It has been shown to degrade a number of organic compounds such as 2-(4-sulfophenyl)butyrate (SPB) and may be useful degredation of linear alkylbenzenesulfonate (LAS) surfactant in wastewater treatment.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
strain SPH-1 is being sequenced for comparative genome analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Desulfatibacillum alkenivorans AK 01</h1>
<div class="content">
<p>Desulfatibacillum alkenivorans is an alkene-degrading, sulfate-reducing bacterium isolated from estuarine sediment. It activates alkanes via subterminal addition of the alkane to fumarate.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain AK-01 was isolated from the Arthur Kill, NJ/NY waterway. This site has a history of contamination from petrochemical industry. AK-01 is a delta-proteobacterium capable of utilizing C13-C18 alkanes as growth substrates (So et al., 1999). Analysis of labeled and fully deuterated metabolites shows that AK-01 activates alkanes via fumarate addition to the subterminal carbon.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Desulfitobacterium hafniense strain DCB-2</h1>
<div class="content">
<p>Desulfitobacterium hafniense can dehalogenate a variety of hydrocarbons and can utilize fumarate, sulfite, and thiosulfate (but not thiousulfate) as terminal electron acceptors. Some important pollutants such as polychlorinated biphenyls (PCBs) may be degraded by this organism.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
strain DCB-2 for this species and has been isolated from municipal sludge. It can dechlorinate chlorinated phenolic compounds such as PCBs and some pesticides.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Desulfitobacterium hafniense Y51</h1>
<div class="content">
<p>This organism can dehalogenate a variety of hydrocarbons and can utilize fumarate, sulfite, and thiosulfate (but not thiousulfate) as terminal electron acceptors. Some important pollutants such as polychlorinated biphenyls (PCBs) may be degraded by this organism.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from soil contaminated with tetrachloroethene (PCE) in Japan. It can efficiently dehalogenate PCEs via trichloroethene (TCE) to cis-1,2-dichloroethene (cis-1,2-DCE). It can also dehalogenase tetra-, penta-, and hexachloroethanes.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Desulfobacterium autotrophicum HRM2</h1>
<div class="content">
<p>This species plays an important role in the ecology of marine sediments. It is involved in the anaerobic mineralization of organic matter coming from the water column. Desulfobacterium autotrophicum is capable of growing litho-autotrophically with H2, CO2 and sulfate, but also by coupling sulfate reduction with the oxidation of fatty acids. CO2 fixation proceeds via the acetyl-CoA/CO-dehydrogenase pathway. This organism can transform tetra to dichlormethane.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain is the type strain of this species. It was isolated from a marine sediment in the Mediterranean sea near Venice, Italy.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Desulfotomaculum reducens  MI-1</h1>
<div class="content">
<p>Desulfotomaculum reducens is able to reduce uranium and chromium making it a potential agent for the bioremediation of heavy metals.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This organism was isolated from heavy metal-contaminated sediment collected at the Mare Island Naval Shipyard on the San Francisco Bay. The genome sequence of this organism will provide information on the mechanism and regulation of heavy metal reduction.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Desulfovibrio desulfuricans ATCC 27774</h1>
<div class="content">
<p>Desulfovibrio desulfuricans is a sulfate-reducing bacterium which reduce sulfate to sulfide found in soil, freshwater, saltwater and the intestinal tract of animals. Desulfovibrio desulfuricans grow anaerobically and utilize a wide variety of electron acceptors, including sulfate, sulfur, nitrate, and nitrite, as well as others. A number of toxic metals are reduced, including uranium (VI), chromium (VI) and iron (III), making these organisms of interest as bioremediators. Metal corrosion, a problem that is partly the result of the collective activity of these bacteria, results in billions of dollars in losses each year to the petroleum industry. These organisms are responsible for the production of poisonous hydrogen sulfide gas in marine sediments and in terrestrial environments such as drilling sites for petroleum products.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Desulfuricans str. ATCC 27774 was isolated from the rumen of a sheep and will be used for comparative analysis with other sulfate-reducing bacteria.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Desulfovibrio desulfuricans G20</h1>
<div class="content">
<p>This organism is a spontaneous nalidixic acid-resistant derivative of a wild-type strain (G100A) that was isolated from an oil well corrosion site. It has been cured of a 2.3 Kb cryptic plasmid as well. This organism is one of the few extensively studied sulfate-reducing bacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Diaphorobacter TPSY</h1>
<div class="content">
<p>Diaphorobacter sp. TPSY is able to anaerobically oxidize iron and uranium when coupled with the reduction of nitrate. This organism may be important for the remediation of uranium in contaminated environments.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Dictyoglomus thermophilum H-6-12</h1>
<div class="content">
<p>Dictyoglomus thermophilum was isolated from a slightly alkaline hot spring (Tsuetae Hot Spring) in Kumamoto Prefecture, Japan. It is an extremely thermophilic, chemoorganotrophic bacterium.\n\nDictyoglomus thermophilum H-6-12 (DSM 3960, ATCC 35947) is the type strain.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Dictyoglomus thermophilum H-6-12 is the type strain.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Dictyoglomus thermophilum H-6-12</h1>
<div class="content">
<p>Dictyoglomus thermophilum was isolated from a slightly alkaline hot spring (Tsuetae Hot Spring) in Kumamoto Prefecture, Japan. It is an extremely thermophilic, chemoorganotrophic bacterium.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Dictyoglomus thermophilum H-6-12 is the type strain.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>ehrlichia canis Jake</h1>
<div class="content">
<p>Ehrlichia canis causes canine monocytic ehrlichiosis, which is a tick-born disease that causes severe morbidity in domesticated and wild dogs. The brown dog tick, Rhipicephalus sanguineus, transfers the organism from its salivary glands to the animal when feeding. The bacterium infects monocytes and can spread throughout the lymphatic system including the spleen and liver, leading to hyperplasia (an abnormal increase in cell size and number). The bacterium are taken up by macrophages and exist in membrane-bound vacuoles after prevention of phagosome-lysozome fusions. Persistence in the vacuole leads to replication and cell division and eventual release from the cell which leads to further spreading of the bacterium throughout the host. Ehrlichia canis may utilize a type IV secretion system to transport effector proteins to the eukaryotic cell during these stages. The end result is an increase in platelet count and anemia.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain Jake  was isolated in 1989 in North Carolina from a 2-year-old male dog
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ehrlichia chaffeensis Arkansas</h1>
<div class="content">
<p>This organism is an obligate intracellular pathogen that exists within vacuoles in the cytoplasm of monocytes or granulocytes. Transferred during an insect (tick) bite, it can cause disease in humans (human monocytic ehrlichiosis) and can reside in several other animals and is a problem in immunocompromised patients. The bacterium inhibits phagosome-lysozome fusion as well as programmed cell death (apoptosis) of the host cell, similar to what is observed with Anaplasma phagocytophilum.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain is the first isolate of Ehrlichia chaffeensis, which was obtained from a patient on an army base in Arkansas in 1990.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Enterobacter  638</h1>
<div class="content">
<p>Enterobacter  638 was isolated in association with the poplar tree,  Populus trichocarpa x deltoids. This strain is an endophyte and grows within the plant tissue without harming the plant. Enterobacter sp. 638 will provide information on the host/endophyte relationship between plants and bacteria.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Erwinia tasmaniensis Et1/99.</h1>
<div class="content">
<p>Erwinia tasmaniensis is not pathogenic to plants and appears to be a common epiphyte, colonizing plant surfaces.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Erwinia tasmaniensis Et1/99 was isolated from apple flowers in Tasmania, Australia. This is the type strain and will be used for comparative analysis with other plant-associated bacteria.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Erythrobacter litoralis HTCC2594</h1>
<div class="content">
<p>Erythrobacter litoralis was isolated from a marine cyanobacterial mat. The presence of the carotenoids bacteriorubixanthinal and erythroxanthin sulfate give this organism a reddish color.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was collected from the Sargasso Sea at a depth of 10 meters, and was isolated by dilution to extinction methods followed by plating on solid media.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli  536</h1>
<div class="content">
<p>Escherichia coli was named for its discoverer, Theodore Escherich, and is one of the premier model organisms used in the study of bacterial genetics, physiology, and biochemistry. This enteric organism is typically present in the lower intestine of humans, where it is the dominant facultative anaerobe present, but it is only one minor constituent of the complete intestinal microflora. E. coli, is capable of causing various diseases in its host, especially when they acquire virulence traits. Strains of E. coli can cause urinary tract infections, neonatal meningitis, and many different intestinal diseases, usually by attaching to the host cell and introducing toxins that disrupt normal cellular processes. Virulence proteins may be encoded on extrachromosomal plasmids or within bacteriophages and distinct DNA segments termed pathogenicity islands (PAIs). PAIs are likely to have been transferred horizontally and may even have integrated into the chromosome through bacteriophage or plasmid integration or transposition.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain  is a uropathogenic Escherichia coli (UPEC) that was isolated from a patient with acute pyelonephritis (inflammation of the kidney and pelvis). It is a model organism of extraintestinal E. coli, which are among the most common causes of non-hospital-acquired urinary tract infections (UTIs) and UPEC strains are responsible for 40% of nosocomial UTIs. Comparative genomics with other E. coli, especially UPEC strains such as CFT073 will reveal important pathogenicity traits. Studies on strain 536 have revealed at least 5 pathogenicity islands. The project is a collaboration with the group of Prof. Hacker, Institute for Molecular Infection Biology, University of Wurzburg.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli 55989</h1>
<div class="content">
<p>Escherichia coli 55989 is a clinical enteroaggregative isolate. Enteroaggregative E. coli strains adhere to mucosal cells and are an emerging cause of gastroenteritis</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli APEC O1</h1>
<div class="content">
<p>This strain (serovar O1) is an avian pathogenic strain and causes respiratory, blood, and enteric infections primarily in poultry.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli ATCC 8739</h1>
<div class="content">
<p>This strain differs from the K-12 strain in having the IS1-13 insertion element associated with an upstream promotor and the first 114 bp of ompC. This deletion also eliminates micF, which negatively controls expression of ompF. The corresponding outer membrane proteins OmpC and OmpF, are the two major porins of K-12. Escherichia coli C str. ATCC 8739 expresses only OmpF.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli E24377A</h1>
<div class="content">
<p>This is an enterotoxigenic E. coli  strain E24377A and was obtained from Maj. Carl Brinkley at Walter Reed Army Institute of Research. ETEC is the leading cause of traveler's diarrhea, characterized by a large volume of watery diarrhea. ETEC primarily colonizes the small intestine by way of the pili. Colonization factor antigens (CFA) in this strain include two pili types (coli surface antigen one and three; CS1 and CS3). Additionally two toxins are thought to be responsible for the virulence, heat stable and heat labile enterotoxins. Other virulence factors include the O139:H28 serotype as well as genes carried on a number of uncharacterized plasmids. The stocks used for genomic DNA preparation were obtained from a master- and production cGMP cell bank available at WRAIR/NMRC. Pathogenicity has been confirmed in human challenge experiments.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli ED1a</h1>
<div class="content">
<p>Escherichia coli ED1a is an avirulent serogroup O81 strain and will be used for comparative analysis</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli HS</h1>
<div class="content">
<p>This strain is a human commensal that was originally isolated from a laboratory scientist at Walter Reed Army Institute of Research in 1978 (Levine, 1978). It was obtained from Drs. Kaper and Nataro at the School of Medicine at the University of Maryland. Strain HS colonizes the human gastrointestinal tract in challenge experiments, but no overt signs of disease occur. Thus, this strain represents a genomic baseline for human gastrointestinal tract colonization. It is competent and amenable to genetic manipulation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli IAI1</h1>
<div class="content">
<p>Escherichia coli IAI1 is a serogroup O8 strain and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli IAI1</h1>
<div class="content">
<p>Escherichia coli IAI1 is a serogroup O8 strain and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli IAI39</h1>
<div class="content">
<p>Escherichia coli IAI39 is a serotype O7:K1 strain from a urinary tract infection and will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli K 12 substr DH10B</h1>
<div class="content">
<p>This strain is a substrain of K-12. It was derived from DH10 (which was derived from MC1061 which in turn was derived from M182). It is a common laboratory strain.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli K12 substr. MG1655 K12</h1>
<div class="content">
<p>Non-pathogenic strain MG1655 approximates wild-type E. coli as it has been maintained with very little genetic manipulation except for the curing (removal) of bacteriophage lambda and the F plasmid. MG1655 was derived from strain W1485, which was derived by Joshua Lederberg from the original K-12 isolate obtained from a patient in 1922.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli K-12 substr. W3110</h1>
<div class="content">
<p>This strain is closely related to MG1655 and is descended from K-12 (W1485) which is a derivative of the original K-12 strain (EMG2). It is considered a more wild-type strain than MG1655, although it has also been cured of the lambda prophage and lacks the fertility factor (F+ plasmid) of the original E. coli K-12 (EMG2).\n\nComparison of the two genomes revealed that they are not identical with strain-specific differences reflecting their distinct lineages.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli O157H7</h1>
<div class="content">
<p>Escherichia coli O157:H7 is an enterohemorrhagic strain of the bacterium Escherichia coli and a cause of foodborne illness. Infection often leads to bloody diarrhea, and occasionally to kidney failure, especially in young children and elderly people. Most illness has been associated with eating undercooked, contaminated ground beef, drinking unpasteurized milk, swimming in or drinking contaminated water, and eating contaminated vegetables.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli O157H7 EDL933</h1>
<div class="content">
<p>This strain (substrain) is considered a reference strain of O157:H7, which was first isolated during an outbreak in 1982.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli SMS-3-5</h1>
<div class="content">
<p>Escherichia coli SMS-3-5 was isolated from a toxic-metal contaminated site, Shipyard Creek, Charleston, South Carolina. This strain is highly resistant to a number of antibiotics and will be used for comparative analysis with other E. coli strains.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia coli  UTI89</h1>
<div class="content">
<p>This strain (UTI89) is a uropathogenic strain isolated from a woman with uncomplicated cystitis (bladder inflammation) that has been demonstrated to cause cystitis in a murine urinary tract infection model.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The genome of UTI89 showed 4 pathogenicity islands. A plasmid was found that was similar to the F plasmid but which also contained virulence genes. Comparison of a set of orthologs from this organism with other E. coli strains showed positive selection on some genes, including those for iron acquisition. Comparison with additional sequences (found in Accession Numbers DQ389000:DQ389068 and DQ440980:DQ441250) of a subset of genes from a number of isolates from patients with UTI infections and isolates from healthy individuals showed positive selection of fepE and amiA genes and some weakly positive selection for ompC.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Escherichia fergusonii ATCC 35469</h1>
<div class="content">
<p>Escherichia fergusonii ATCC 35469 was isolated from human feces and will be used for comparative analysis with other Escherichia species</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Exiguobacterium sibiricum 255 15</h1>
<div class="content">
<p>This organism was isolated from a 2-3 million-year permafrost core in Siberia, Russia and can survive and grow rapidly at low temperatures. Analysis of long-term survival of psychrophilic organisms such as this one may aid understanding of the potential growth of organisms in astrobiology.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Fervidobacterium nodosum Rt17-B1</h1>
<div class="content">
<p>Fervidobacterium nodosum was isolated from a hot spring in New Zealand. The cellular morphology of this organism, a member of the Thermotogales, is characterized by a terminal spherical extension of the cell envelope. \n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Fervidobacterium nodosum Rt17-B1 will be used for comparative analysis with other thermophilic bacteria.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Finegoldia magna ATCC 29328</h1>
<div class="content">
<p>Finegoldia magna, formerly Peptostreptococcus magnus, is a commensal bacterium colonizing human skin and mucous membranes. It has been shown to cause valve endocarditic in humans. \n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Finegoldia magna ATCC 29328 is being sequenced for comparative genome analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Flavobacterium johnsoniae ATCC 17061</h1>
<div class="content">
<p>Flavobacterium johnsoniae, formerly Cytophaga johnsonae, is common in soil and freshwater. This organism belongs to a large group of related environmental microorganisms that are characterized by the ability to utilize a wide variety of naturally occurring complex compounds, such as chitin, cellulose, and lignin.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Flavobacterium johnsoniae ATCC 17061 was isolated from soil in England and is the type strain of this species. This organism will be used for comparative genomics.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Francisella philomiragia ATCC 25017</h1>
<div class="content">
<p>Francisella philomiragia, formerly Yersinia philomiragia, has been isolated from water, muskrats, fish and humans. Francisella philomiragia is made up of two supspecies, philomiragia and noatunensis. Francisella philomiragia supsp. philomiragia is able to cause an often fatal bacteremia in people with chronic granulomatous disease. This supspecies can also cause pneumonia in near-drowning victims.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Francisella philomiragia ATCC 25017 was isolated from water in the Bear River Refuge in Utah.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Francisella tularensis WY96-3418</h1>
<div class="content">
<p>Francisella tularensis was first identified by Edward Francis as the causative agent of a plague-like illness that affected squirrels in Tulare county in California in the early part of the 20th century. The organism now bears his name. The disease, which has been noted throughout recorded history, can be transmitted to humans by infected ticks or deerflies, infected meat, or by aerosol, and thus is a potential bioterrorism agent. This organism has a high infectivity rate, and can invade phagocytic and nonphagocytic cells, multiplying rapidly. Once within a macrophage, the organism can escape the phagosome and live in the cytosol. It is an aquatic organism, and can be found living inside protozoans, similar to what is observed with Legionella. Francisella tularensis has four subspecies, tularensis, holarctica, and mediaasiatica all of which can cause disease with subspecies tularensis being the most virulent.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Francisella tularensis WY96-3418 was isolated from a human finger wound and is being sequenced for comparative genome
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Francisella tularensis holarctica</h1>
<div class="content">
<p>This strain (live vaccine strain) was created in the 1960's in the USA and provides protection against tularemia in animal models as well as in humans.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Francisella tularensis holarctica OSU18</h1>
<div class="content">
<p>This is a virulent strain isolated from a dead beaver and will be used for comparative genomics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Francisella tularensis mediasiatica FSC147</h1>
<div class="content">
<p>Francisella tularensis mediasiatica strain FSC147 was collected from a gerbil in central Asia and will be used for comparative analysis</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Francisella novicida U112</h1>
<div class="content">
<p>Francisella novicida is closely related to Francisella tularensis, the causative agent for tularemia but is nonvirulent. This organism, however may cause disease in immunocompromised individuals.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Francisella novicida strain U112 is the type strain and will be used for comparative genomics.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Frankia CcI3</h1>
<div class="content">
<p>This genus is a member of the actinomycetales and forms a symbiotic nitrogen-fixing association with a number of plants. Vesicles, which are spherically differentiated structures, are formed inside of the plant cell and are the sites of bacterial nitrogen fixation. The genus is named after the Swiss microbiologist, A.B. Frank, and was originally thought to be a fungus due to the extremely hyphal morphology of the organism.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from root nodules of Casuarina cunninghamiana in 1983 at Harvard Forest.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Frankia EAN1pec</h1>
<div class="content">
<p>Frankia strain EAN1pec was isolated at the Kettering Research Laboratory in Ohio by M. Lalonde in 1978.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Frankia alni  ACN14a</h1>
<div class="content">
<p>Frankia alni strain ACN14a was isolated from a green alder (Alnus crispa) growing in Tadoussac, Canada. It is being sequenced to determine the nature of the symbiotic relationship with its hosts and the nature of the secondary metabolites produced.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Geobacillus thermodenitrificans NG80-2</h1>
<div class="content">
<p>Geobacillus thermodenitrificans NG80-2 was isolated from oil reservoir formation water taken at a depth of 2000 m and a temperature of 73oC. This strain can use crude oil as a sole carbon source and can degrade 16 to 36 carbon alkanes. Geobacillus thermodenitrificans NG80-2 produces an emulsifier which may be useful for high temperature biodegradation or other industrial purposes.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Geobacter bemidjiensis Bem</h1>
<div class="content">
<p>Geobacter is a genus of proteobacteria. Geobacter are an anaerobic respiration bacterial species which have capabilities that may make them useful in bioremediation. The geobacter was found to be the first organism with the ability to oxidize organic compounds and metals, including iron, radioactive metals and petroleum compounds into environmentally benign carbon dioxide while using iron oxide or other available metals as electron acceptor.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
eobacter bemidjiensis Bem was isolated from subsurface sediments collected in Bemidji, Minnesota. This organism is the type strain and will be used for comparative analysis with other Geobacter species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Geobacter bemidjiensis Bem</h1>
<div class="content">
<p>Geobacter is a genus of proteobacteria. Geobacter are an anaerobic respiration bacterial species which have capabilities that may make them useful in bioremediation. The geobacter was found to be the first organism with the ability to oxidize organic compounds and metals, including iron, radioactive metals and petroleum compounds into environmentally benign carbon dioxide while using iron oxide or other available metals as electron acceptor.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Geobacter bemidjiensis Bem was isolated from subsurface sediments collected in Bemidji, Minnesota. This organism is the type strain and will be used for comparative analysis with other Geobacter species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Gluconacetobacter diazotrophicus PAl 5</h1>
<div class="content">
<p>Gluconacetobacter diazotrophicus is endophytic and colonizes internal plant tissues, establishing a symbiotic relationship with its host. This bacterium has been found in sugarcane, coffee, rice, tea, and other plants. The nitrogen-fixation systems of the bacterium provide the plant with essential nitrogenous compounds while the plant provides a protected environment for the bacterium to grow in. Nitrogen-fixation is important for sugarcane production, and this organism can fix nitrogen even in the presence of nitrate.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Gluconacetobacter diazotrophicus strain PAL5 was isolated from sugarcane roots in Brazil and will be used for comparative analysis.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Haemophilus influenzae 86-028NP</h1>
<div class="content">
<p>Haemophilus influenzae was first isolated in the 1890s during an influenza pandemic by Pfeiffer, and was originally thought to be the source of influenza, although later it was shown to be a secondary pathogen and may be synergistic with the influenza virus. This bacterium is one of the leading causes of meningitis in young children, and it may also cause septicemia, otitis media (inflammation of the middle ear), sinusitis (inflammation of the sinus cavity) and chronic bronchitis. It is highly adapted to its human host and typically lives in the nasopharynx and is a major cause of lower respiratory infections in infants and small children in developing countries (type 1b strain), although vaccine use has resulted in the decline of infections. The presence of the type B polysaccharide is important for virulence, as it is the most pathogenic of the 6 major types. The encapsulated organism can penetrate the blood and avoid both phagocytosis and complement-mediated lysis. All known strains produce neuraminidase and an IgA protease as well as fimbrial adhesins for attachment.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This nontypeable strain is a clinical isolate from a pediatric patient with otitis media from Columbus Children's Hospital. It has been extensively characterized in chinchilla models of otitis media.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Haemophilus influenzae PittEE</h1>
<div class="content">
<p>Strain PittEE is a non-typeable Haemophilus influenzae (NTHi) isolate recovered by Dr. Robert Wadowsky from the middle-ear effusion of a child in Pittsburgh who was undergoing tympanostomy and ventilation tube placement for the treatment of chronic otitis media with effusion (COME). In the chinchilla model, PittEE infection induces COME but does not cause systemic disease.\n\nSequenced and assembled using technology developed by 454 Life Sciences, Inc. (www.454.com), supplemented with double-barreled Sanger dideoxy sequencing read data from traditional clone libraries. The genome sequence is currently comprised of 6 contigs, and has been sequenced to a depth of 35X.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Haemophilus influenzae PittGG</h1>
<div class="content">
<p>Strain PittGG is a non-typeable Haemophilus influenzae (NTHi) isolate recovered by Dr. Robert Wadowsky from the external ear discharge of a spontaneously perforated tympanic membrane of a child in Pittsburgh who had been diagnosed with otorrhea. In the chinchilla model, PittGG infection induces acute otitis media (AOM) and causes virulent, uniformly fatal, systemic disease.\n\nSequenced and assembled using technology developed by 454 Life Sciences, Inc. (www.454.com), supplemented with double-barreled Sanger dideoxy sequencing read data from traditional clone libraries. The genome sequence is currently comprised of 60 contigs, and has been sequenced to a depth of 35X.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Haemophilus parasuis SH0165</h1>
<div class="content">
<p>The genome sequence of this organism will provide information on the genes involved in pathogenicity and virulence.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Haemophilus somnus 129PT</h1>
<div class="content">
<p>This strain is an asymptomatic carrier strain that was serum-resistant.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Haemophilus somnus 2336</h1>
<div class="content">
<p>Haemophilus somnus is a major cause of disease in feedlot cattle. It causes constriction of blood vessels supplying major organs, restricting blood supply which leads to reduced oxygen delivery, and leads to tissue damage and death. The diseases that result are pneumonia, arthritis, myocarditis (inflammation of heart muscle), and reproductive problems resulting in the loss of millions of dollars in animals each year. This bacterium can live intracellularly, and avoids or inhibits the oxidative burst of phagocytes. Some of the major virulence determinants are the lipooligosaccharides and various surface antigens, a few of which appear to bind the Fc (conserved region) determinant on immunoglobulins</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from the lung of a calf which had pneumonia.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Halobacterium salinarum R1</h1>
<div class="content">
<p>Halobacterium salinarium is an obligately halophilic archeon and represents the type species for the genus Halobacterium. The first recorded isolation of this organism came from the spoilage of salted fish. It is also commonly found in neutral salt lakes, marine salterns, proteinaceous products heavily salted with crude solar salt and ancient evaporate deposits. Cells display pleomorphic forms in deficient media or elevated temperatures. Some strains contain gas vacuoles. Motility is via polar tufts of flagella. Growth occurs optimally at salt concentrations of 3.5-4.5 M NaCl, no growth occurs at concentrations below 3 M NaCl. Additionally cells require 0.05-0.1 M Mg2+.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Halobacterium salinarium strain R1 forms translucent red colonies as opposed to the opaque pink colonies formed by Halobacterium salinarium NRL. Strain R1 has no gas vacuoles and will be used for comparison with other Halobacterium species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Halobacterium sp</h1>
<div class="content">
<p>This microbe  is an obligately halophilic archeon that has adapted to growth under conditions of extremely high salinity. Motility is via tufts of polar flagella and intracellular gas vesicles are used for buoyancy. This organism grow aerobically and its ease of culturing combined with the availability of established methods of genetic manipulation in the laboratory make it an ideal model organism for study of the archaea</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Haloquadratum walsbyi</h1>
<div class="content">
<p>This hyperhalophile is of specific interest because of its unique square shape and extraordinarily large size (40m x 40m). Isolated in 1980, it initially could not be cultured in the laboratory despite its abundance in nature and easy recognition by microscopy. Its culture conditions were optimized in 2004.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Halorhodospira halophila SL1</h1>
<div class="content">
<p>Halorhodospira halophila  formerly Ectothiorhodospira halophila is alkaliphilic, phototrophic, halophile. This is one of the most halophilic eubacteria known, and produces organic solutes such as glycine betaine, ectoine, and trehalose to balance the osmotic pressure. This organism oxidizes sulfide to sulfur, which is deposited outside the cell and further oxidized to sulfate. This organism also produces a blue light sensor called photoactive yellow protein which is involved in signal transduction.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Halorhodospira halophila SL1 was isolated from salt lake mud and will be used for comparative analysis with other halophilic organisms.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Halorubrum lacusprofundi ATCC 49239</h1>
<div class="content">
<p>Halorubrum lacusprofundi is a red pigmented halophilic archaeon which has been found in hypersaline marine enviornments. Originally thought to be a psychrophile, further research determined that the optimum temperature for growth for Halorubrum lacusprofundi was from 31 - 37oC.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain formerly Halobacterium lacusprofundi, this organism is an extremely halophilic archaeon isolated from Deep Lake, Antarctica. This organism will be used for comparative genomics with other extreme halophiles.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Halothermothrix orenii H 168</h1>
<div class="content">
<p>Halothermothrix orenii is a member of a group of anaerobic halophilic bacteria related to the Clostridia. This organism is a thermophile and therefore is able to grow in hot anaerobic saline environments.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Halothermothrix orenii H 168 was isolated from the sediment of a salt lake in Tunisia. This organism will provide information on the adaptations necessary for survival in hot saline environments.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Helicobacter pylori HPAG1</h1>
<div class="content">
<p>Helicobacter pylori is a Gram-negative, microaerophilic bacterium that inhabits various areas of the stomach and duodenum. It causes a chronic low-level inflammation of the stomach lining and is strongly linked to the development of duodenal and gastric ulcers and stomach cancer. Over 80% of individuals infected with the bacterium are asymptomatic.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from a Swedish patient with chronic atrophic gastritis, the precursor to gastric adenocarcinoma. Since Helicobacter pylori is associated with gastric cancer, sequencing the genome of this clinical isolate, and comparing it to the two other sequenced type strains (J99 and 26695, both recovered from patients with acid peptic disease) should provide new insights about microbial factors that may contribute to tumorigenesis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Helicobacter pylori J99</h1>
<div class="content">
<p>This strain was isolated in 1994 in the USA from a patient with duodenal ulcer. It has undergone little subculturing prior to being sequenced.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Helicobacter pylori P12</h1>
<div class="content">
<p>Helicobacter pylori P12 is a clinical isolate obtained from a patient with duodenal ulcer. This organism will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Helicobacter pylori Shi470</h1>
<div class="content">
<p>Helicobacter pylori Shi470 is a clinical isolate from gastric antrum from Amerindian resident of remote Amazonian village of Shimaa, Peru. Helicobacter pylori chronically infects stomachs of billions of people worldwide, is implicated in peptic ulcer disease and gastric cancer, and is very diverse genetically. Different multilocus sequence types (MLST) (genotypes) predominate in different human populations. Shimaa village strains such as Shi470 is more closely related to strains from East Asia than other geographic regions, and is postulated to represent the strains of Native Americans before European conquest. To our knowledge, it is the first H. pylori strain completely sequenced that is not from an ethnic Europeans. Its genome sequence should be valuable for analyses of mechanisms of infection, virulence and genome evolution.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Herminiimonas arsenicoxydans</h1>
<div class="content">
<p>This heterotrophic bacterium is capable of reducing and oxidizing arsenic with the objective of detoxification. Arsenic is both a product from natural sources and of human activities, and is widely distributed in the environment, essentially in 3 different oxidation states: As (-III) (arsine), As (+III) (arsenite) and As (+V) (arseniate). The soluble and toxic forms, arsenite and arseniate are frequently reported as contaminants of water and arsenic has been identified as a major risk factor for human health in many places in the world. The ecology of this metalloid is strongly dependent on microbial transformations which affect the mobility and bioavailability as well as the toxicity of arsenic in the environment. To date, the best-studied mechanism of transformation is the reduction of arseniate (AsV) to arsenite (AsIII).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Herminiimonas arsenicoxydans was isolated from heavy metal contaminated sludge from an industrial water treatment plant. This organism has a number of mechanisms for metabolizing arsenic allowing it to effectively colonize arsenic-contaminated environments.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Herpetosiphon aurantiacus ATCC 23779</h1>
<div class="content">
<p>Herpetosiphon aurantiacus grows in long, flexible, filaments with a transparent section of empty cell wall at the end of the filament. Herpetosiphon spp. and Herpetosiphon-like organisms have been found in soil, freshwater, and sewage treatment plants. These organisms have been implicated in the bulking of sludge in sewage treatment plants, which is problematic in that the sludge retains more water, making subsequent disposal difficult.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from Birch Lake in Minnesota and will be used for comparative analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Hydrogenobaculum Y04AAS1</h1>
<div class="content">
<p>Hydrogenobaculum YO4AAS1 was isolated from Obsidian Pool, an acidic hot spring in Yellowstone National Park. This organism gets energy through the oxidation of hydrogen or reduced sulfur compounds. Hydrogenobaculum sp. YO4AAS1 will be used for comparative analysis with other thermophilic organisms.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Hyperthermus butylicus</h1>
<div class="content">
<p>This bacterium is an extreme hyperthermophilic, anaerobic, sulfur-reducing archaeon. H. butylicus grows optimally at 95-106°C, with a NaCl concentration of 17 g/l and a pH of 7.0.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Hyphomonas neptunium ATCC 15444</h1>
<div class="content">
<p>This organism is also known as Hyphomicrobium neptunium. It has a biphasic life style, which consists of a motile phase of flagellated swarmer cells, and a cessile phase in which a long prosthecate is produced at one end of the bacteria through which budding cells emerge. Newly budded cells in turn produce flagella and go through a motile phase and the cycle continues. These organisms can colonize the surfaces of marine environments which enables additional species to colonize at later stages. This organism may be of use in treatment of water as they attach to a solid surface and are capable of degradation of a number of pollutants including aromatic hydrocarbons, dimethyl sulfoxide and methyl chloride.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain ATCC 15444 is the type strain of the species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ignicoccus hospitalis Kin4</h1>
<div class="content">
<p>Ignicoccus hospitalis, an anaerobic, hyperthermophilic, chemoautolithotroph, uses hydrogen to reduce elemental sulfur and produce hydrogen sulfide. This organism is also moderately halophilic and acidophilic. Ignicoccus hospitalis is the host organism for Nanoarchaeum equitans.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Ignicoccus hospitalis Kin4 I was isolated from gravel obtained from the shallow marine hydrothermal system of the Kolbeinsey Ridge, north of Iceland and will be used for comparative analysis with other thermophiles.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Janthinobacterium Marseille</h1>
<div class="content">
<p>This organism, also called Minibacterium massiliensis, was isolated from a solution used in kidney dialysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Kineococcus radiotolerans SRS30216</h1>
<div class="content">
<p>K. radiotolerans exhibits a high degree of resistance to ionizing gamma-radiation. Cells are also highly resistant to dessication. Kineococcus-like 16S rRNA gene sequences have been reported from the Mojave desert and other arid environments where these bacteria seem to be ubiquitous. Because of its high resistance to ionizing radiation and desiccation, K. radiotolerans has potential use in applications involving in situ biodegradation of problematic organic contaminants from highly radioactive environments. Moreover, comparative functional genomic characterization of this species and other known radiotolerant bacteria such as Deinococcus radiodurans and Rubrobacter xylanophilus will shed light onto the strategies these bacteria use for survival in high radiation environments, as well as the evolutionary origins of radioresistance and their highly efficient DNA repair machinery.\n\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Kineococcus radiotolerans SRS30216 is a coccoid bacterium originally isolated from a high-level radioactive waste cell at the Savannah River Site in Aiken, South Carolina, USA, in 2002. This is the type strain of the species. Comparative functional genomic characterization of this species and other known radiotolerant bacteria such as Deinococcus radiodurans and Rubrobacter xylanophilus will shed light onto the strategies these bacteria use for survival in high radiation environments, as well as the evolutionary origins of radioresistance and their highly efficient DNA repair machinery.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Klebsiella pneumoniae 342</h1>
<div class="content">
<p>Klebsiella pneumoniae is the most medically important organism within the genus Klebsiella. It is an environmental organism found in water, soil, and on the surface of plants. Several strains have been isolated from plant tissues and are nitrogen-fixing endophytes that may be a source of nitrogen for the plant. Other strains can become opportunistic pathogens which infect humans, and typically causes hospital-acquired infections in immunocompromised patients. Major sites of infection include the lungs, where it causes a type of pneumonia, and urinary tract infections. Klebsiella can also enter the bloodstream (bacterimia) and cause sepsis. The pathogen can also infect animals and cause inflammation of the uterus in horses as well as more generalized infections in other mammals.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Klebsiella pneumoniae 342 was isolated from the stem tissue of Zea mays. This strain fixes atmospheric nitrogen and may be able to provide nitrogen, in the form of ammonia, to plant cells. This organism will be used for comparative analysis with endophytic and pathogenic Klebsiella pneumoniae strains.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Klebsiella pneumoniae MGH 78578</h1>
<div class="content">
<p>Klebsiella pneumoniae is the most medically important organism within the genus Klebsiella. It is an environmental organism found in water, soil, and on the surface of plants. Several strains have been isolated from plant tissues and are nitrogen-fixing endophytes that may be a source of nitrogen for the plant. Other strains can become opportunistic pathogens which infect humans, and typically causes hospital-acquired infections in immunocompromised patients. Major sites of infection include the lungs, where it causes a type of pneumonia, and urinary tract infections. Klebsiella can also enter the bloodstream (bacterimia) and cause sepsis. The pathogen can also infect animals and cause inflammation of the uterus in horses as well as more generalized infections in other mammals. \nThis organism expresses numerous pathogenicity factors, including multiple adhesins, capsular polysaccharide, siderophores, and lipopolysaccharide for the evasion of host defenses. Capsular polysaccharide (K antigen), for example, prevents phagocytosis. The multiple antibiotic resistance genes carried on the chromosome inhibit efforts to clear the organism from infected patients via antibiotic use. Many of the virulence factors remain to be elucidated. \n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Klebsiella pneumoniae strain MGH 78578 was isolated from a patient in 1994.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Kocuria rhizophila DC2201</h1>
<div class="content">
<p>Kocuria rhizophila was first isolated from the rhizosphere of the narrow-leaved cattail and is commonly found in water and soil. This organism has also been isolated from human skin and is considered to be part of the normal skin flora.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Kocuria rhizophila DC2201, formerly Micrococcus luteus, is used as a control strain for sterility and antibiotic susceptibility testing. This strain is also resistant to certain long-chain alcohols such as 1-octanol and will be used for comparative analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactobacillus brevis ATCC 367</h1>
<div class="content">
<p>Lactobacillus brevis is a species of lactic acid bacteria. It can be found in many different environments and in fermented foods such as sauerkraut and pickles. It is also one of the most common causes of beer spoilage. Ingestion has been shown to improve human immune function, and it has been patented several times. L. brevis is one of the major Lactobacillus species found in tibicos grains (aka water kefir grains), and has been identified as the species responsible for the production of the polysaccharide (dextran) that forms the grains. Major metabolites of L. brevis include lactic acid and ethanol. Strains of L. brevis and L. hilgardii have been found to produce the biogenic amines tyramine and phenylethylamine.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Lactobacillus brevis ATCC 367 will be used for comparative analysis with other lactic acid bacteria.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactobacillus casei ATCC 334</h1>
<div class="content">
<p>Lactobacillus casei is a species of genus Lactobacillus found in the human intestine and mouth. As a lactic acid producer, it has been found to assist in the propagation of desirable bacteria. This particular species of lactobacillus is documented to have a wide pH and temperature range, and complements the growth of L. acidophilus, a producer of the enzyme amylase (a carbohydrate-digesting enzyme). It is known to improve digestion and reduce lactose intolerance and constipation.\n\nThe most common application of L. casei is industrial, specifically for dairy production. However, a team of scientists from Simón Bolívar University in Caracas, Venezuela found that by using Lactobacillus casei bacteria in the natural fermentation of beans, the beans contained lower amounts of the compounds causing flatulence upon digestion.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Lactobacillus casei ATCC 334. This strain was isolated from Emmental cheese and will be used for comparative analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactobacillus delbrueckii  bulgaricus</h1>
<div class="content">
<p>Lactobacillus delbrueckii  bulgaricus contains three subspecies: subsp. delbrueckii, subsp. lactis, and subsp. bulgaricus. The subspecies bulgaricus is used as a starter culture for a number of fermented dairy products such as yogurt and Swiss and Italian-type cheeses, and is a thermophilic culture, where the optimum temperature is 42 C.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactobacillus delbrueckii bulgaricus ATCC BAA-365</h1>
<div class="content">
<p>Lactobacillus delbrueckii subsp.bulgaricus strain ATCC BAA-365 was derived from an existing strain (derived from a French starter culture; CHCC757).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactobacillus gasseri ATCC 33323</h1>
<div class="content">
<p>Lactobacillus gasseri appears to be the main species of lactobacilli that inhabits the human gastrointestinal tract. It is a rod-shaped, gram-positive anaerobe. Of the lactobacilli, it is the most amenable to DNA introduction and manipulation which should prove useful in the functional genomic analysis of this species. It has been shown to be an effective probiotic in suppressing Helicobacter pylori infections in humans.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Lactobacillus gasseri ATCC 33323 is a human isolate and is the type strain. This organism has been shown to have probiotic activities such as production of bacteriocins, macrophage stimulation and adherence to intestinal tissues.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactobacillus reuteri DSM 20016</h1>
<div class="content">
<p>Lactobacillus reuteri is a member of the normal microbial community of the gut in humans and animals. This organism produces antibiotic compounds, such as reutericin and reuterin, which have inhibitory effects on pathogenic microorganisms. In addition, Lactobacillus reuteri is commonly used as a probiotic to maintain the balance of the gut microbial flora and stimulate the intestinal immune system.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Lactobacillus reuteri strain DSM 20016 (JCM 1112, F275) is the type strain for the species. It is a human isolate, which is unable to colonize the intestinal tract of mice.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactobacillus sakei sakei 23K</h1>
<div class="content">
<p>Lactobacillus sakei is the predominant lactic acid bacteria found on fresh meat. This organism is used as a starter in the production of fermented meat products, and plays a major role in preserving meat products by inhibiting the growth of other bacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Lactobacillus sakei sakei 23K was originally isolated from a French sausage. It is a plasmid-free isolate that is a model strain for studies on this organism.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactobacillus salivarius UCC118</h1>
<div class="content">
<p>Lactobacillus salivarius was first isolated as part of a survey of human oral lactobacilli and is commonly found in the oral cavity and gastrointestinal tract of humans and other animals.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Lactobacillus salivarius UCC118 was isolated from the human gastrointestinal tract. This organism produces bacteriocins, antibiotic compounds which have inhibitory effects on pathogenic microorganisms. Lactobacillus salivarius subsp. salivarius strain UCC118 has been extensively studied for its effectiveness as a probiotic to maintain the balance of the gut microbial flora and stimulate the intestinal immune system.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lactococcus lactis cremoris SK11</h1>
<div class="content">
<p>Lactococcus lactis cremoris is used to make hard cheeses.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Lactococcus lactis cremoris SK11 is recognized for the beneficial flavors it produces during cheese production.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lawsonia intracellularis PHE MN1-00</h1>
<div class="content">
<p>Lawsonia intracellularis causes proliferative enteropathy (ileitis) in swine and other domesticated animals resulting in severe losses each year. This obligate intracellular pathogen infects the mucosa of the lower intestinal tract by initially infecting crypt cells, which are precursors that normally grow and divide in order to replace the epithelial cells. Once infection occurs, the crypt cells are stimulated to grow and divide abnormally, resulting in the proliferative phenotype. In severe cases of the disease the entire bowel can become affected and persist for up to 40 days, greatly affecting the host animal.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Lawsonia intracellularis PHE MN1-00 was isolated from intestinal mucosal lesions in pigs that had proliferative enteropathy (PE). When introduced into health pigs, this organism produced the clinical and histological signs of PE.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Legionella pneumophila Philadelphia 1</h1>
<div class="content">
<p>Legionella pneumophila is a thin, pleomorphic, flagellated Gram-negative bacterium of the genus Legionella. L. pneumophila is the primary human pathogen in this group and is the causative agent of legionellosis or Legionnaires' disease.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Legionella pneumophila Philadelphia 1 is the isolate from the initial event of Legionellosis in Philadelphia and is the type strain for this species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Leptospira biflexa  Patoc 1 (Ames)</h1>
<div class="content">
<p>Leptospira biflexa is a non pathogenic Leptospira commonly found in water and soil. A genetic manipulation system has been developed for this organism. \n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Leptospira biflexa strain 'Patoc 1 (Ames)' will be used for comparative genomics with other Leptospira species.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Leptospira biflexa Patoc 'Patoc 1 (Paris)'</h1>
<div class="content">
<p>Leptospira biflexa is a non pathogenic Leptospira commonly found in water and soil. A genetic manipulation system has been developed for this organism.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Leptospira biflexa Patoc strain Patoc 1 (Paris) is the type strain for the species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Leptospira borgpetersenii serovar Hardjo-bovis JB197</h1>
<div class="content">
<p>Leptospira borgpetersenii is a bacterial pathogen of cattle. The disease, leptospirosis, is of worldwide importance and responsible for considerable financial loss in the diary industry as a consequence of agalactia, abortion, stillbirth, birth of weak calves, and reduced fertility. This organism colonizes in the kidney and reproductive tracts of infected animals. There is also a risk of zoonotic infections in humans, especially those who work with cattle. Evaluations of antibiotics and vaccines for use in cattle are underway.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Leptospira borgpetersenii Hardjo-bovis strain JB197 is lethal in hamsters and will be used for comparative genomics with other Leptospira species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Leptospira borgpetersenii serovar Hardjo-bovis L550</h1>
<div class="content">
<p>Leptospira borgpetersenii Hardjo-bovis strain L550 was isolated from a human clinical sample and will be used for comparative genomics with other Leptospira species.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Leptothrix cholodnii SP 6</h1>
<div class="content">
<p>Leptothrix cholodnii, like other Leptothrix species is an iron- and manganese-oxidizing, sheathed bacterium common in organic rich aquatic environments. Iron oxides and manganese oxides often coat the sheathes of these organisms.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Leptothrix cholodnii SP-6 was isolated from water and flocculent from an artificial iron seep in Ithaca, NY. This organism is being sequenced for comparative analysis with other iron-oxidizing bacterium.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Leuconostoc mesenteroides ATCC 8293</h1>
<div class="content">
<p>Leuconostoc mesenteroides is a member of the lactic-acid bacteria, a group of organisms that produce lactic acid during growth, thus lowering the pH of the surrounding environment. This process of fermentation results in inhibition of competing food spoilage organisms from growing as they cannot tolerate the acidic environment. Leuconostoc mesenteroides is associated with fermentation during sauerkraut (pickled cabbage), kimchi (a traditional Korean condiment), and dill pickle production. This organism also produces exopolysaccharides such as dextran, and has been used for the commercial production of this polymer, which is used in cosmetics, some medical applications, as well as the creation of sephadex gels and beads which are used in industrial and laboratory settings.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Leuconostoc mesenteroides strain ATCC 8293 is resistant to the antibiotic vancomycin
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Listeria monocytogenes 4b F2365</h1>
<div class="content">
<p>Listeria monocytogenes, which causes listeriosis, is one of the leading causes of death from food-borne pathogens especially in pregnant women, newborns, the elderly, and immunocompromised individuals. It is found in environments such as decaying vegetable matter, sewage, water, and soil, and it can survive extremes of both temperatures (1-45C) and salt concentration marking it as an extremely dangerous food-born pathogen, especially on food that is not reheated. The bacterium can spread from the site of infection in the intestines to the central nervous system and the fetal-placental unit. Meningitis (inflammation of the membrane surrounding spinal cord and brain), gastroenteritis (inflammation of mucous membranes of stomach and intestine), and septicemia (systemic spread of bacteria and toxins in the blood) can result from infection. This organism is enteroinvasive, and utilizes an actin-based motility system by using a surface protein, ActA, that promotes actin polymerization, to spread intercellularly using the polymerized cytoskeletal protein as a "motor". There are 13 serovars associated with Listeria monocytogenes, and the serovar 4b strains are more commonly associated with invasive disease.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Listeria monocytogenes strain 4b F2365 was isolated in 1985 in California, USA, during an outbreak of listeriosis among patients with AIDS. The strain is of serotype 4b and was isolated from a cheese product that caused the outbreak.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Listeria monocytogenes HCC23</h1>
<div class="content">
<p>Listeria monocytogenes strain HCC23 was isolated from channel catfish. This strain is a serotype 4a strain and is naturally avirulent. Listeria monocytogenes HCC23 will be used for comparative analysis</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Listeria welshimeri serovar 6b SLCC5334</h1>
<div class="content">
<p>Listeria welshimeri is a nonpathogenic Listeria found in soil, water, food, and sewage</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Listeria welshimeri 6b SLCC5334 is a serovar 6b strain and will be used in comparative genomics with other Listeria species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Lysinibacillus sphaericus C3-41</h1>
<div class="content">
<p>Lysinibacillus sphaericus, formerly Bacillus sphaericus, is a common environmental organism which produces an insecticidal toxin similar to that produced by Bacillus thuringiensis. Lysinibacillus sphaericus produces two proteins during sporulation which accumulate in parasporal crystalline inclusions. These proteins bind in the gastric cavitiy and midgut of the larvae causing disruption of feeding by the infected host and death. B. sphaericus is highly toxic to the Culex genus of mosquitoes and, to a lesser extent, to the Anopheles and Aedes genera.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Lysinibacillus sphaericus C3-41 was isolated from a mosquito breeding site in China. This organism is highly toxic to Culex species of mosquito and will be used for comparative analysis with other insect pathogenic bacteria.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Macrococcus caseolyticus JCSC5402</h1>
<div class="content">
<p>This organism is a large gram positive coccus, related to the Staphylococcus, often associated meat and sheep, goat and cattle milk.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This methicillin-resistant strain was isolated from animal meat in a supermarket.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Magnetospirillum magneticum AMB-1</h1>
<div class="content">
<p>This strain was isolated from pond water in Tokyo, Japan. Magnetic particles from Magnetospirillum magneticum AMB-1 have been used in DNA and RNA isolation procedures. This organism will be used for comparative genomics with other magnetotactic bacterium.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Maricaulis maris MCS10</h1>
<div class="content">
<p>Maricaulis maris inhabits marine environments and plays an important part in biogeochemical cycling of organic nutrients. This bacterium undergoes an unusual developmental cycle in which a swarming motile cell becomes a stalked cell that is attached to a solid surface. The stalked cell then undergoes asymmetric cell division and produces one flagellated motile daughter cell and one stalked daughter cell. The stalked cell immediately undergoes replication and division, while in the swarmer cell these processes are inhibited</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Maricaulis maris MCS10 was isolated from Puget Sound in Washington and will be used for comparative analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Marinobacter aquaeolei VT8</h1>
<div class="content">
<p>Marinobacter hydrocarbonoclasticus is a moderately halophilic, hydrocarbon-degrading bacterium which has been isolated from a number of hydrocarbon polluted marine environments. Marinobacter hydrocarbonoclasticus is able to produce biofilms as well as survive in open seawater.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Marinobacter aquaeolei VT8, also known as Marinobacter hydrocarbonoclasticus  VT8, is a moderately halophilic, hydrocarbon-degrading bacterium isolated from an oil well off the coast of Vietnam. Marinobacter aquaeolei is also able to oxidize iron. The availability of iron is considered important to the cycling of carbon in the ocean and iron deficiency is believed to limit primary productivity. Marinobacter aquaeolei, as well as other members of this genus, may play a significant role in the iron and carbon cycles in the marine environment. This organism will be used for comparative genomics.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mesorhizobium BNC1</h1>
<div class="content">
<p>Mesorhizobium species are similar to the Rhizobium and Bradyrhizobium in that they can be plant root symbionts and provide fixed nitrogen to the plant.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This organism was isolated from a mixed-culture enriched from sewage using the chelating agent EDTA (binds divalent cations - ethylenediamine tetraacetic acid) as the sole carbon and nitrogen source. EDTA is often found at toxic waste sites where it is bound to metals and radionuclides, which affects bioremediation efforts. This bacterium is capable of EDTA degradation by itself.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Metallosphaera sedula DSM 5348</h1>
<div class="content">
<p>Metallosphaera sedula was isolated from a thermal pond in the Pisciarelli Solfatara in Italy. This organism is able to leach metals from sulfidic ores making it a potential bioleaching agent. The rate of oxidation of iron pyrite for this organism is increased under conditions of thermal and chemical stress.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Metallosphaera sedula DSM 5348 is the type strain. The genome sequence will provide information on the production and regulation of enzymes involved in the biooxidation of metals.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methanobrevibacter smithii ATCC 35061</h1>
<div class="content">
<p>Methanobrevibacter smithii can be isolated from anoxic environments such as anaerobic digesters, anaerobic soil and sediment, and the gastrointestinal tracts of humans, ruminants and other animals. This organism is the dominant archaeal symbiont found in the human gut.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Methanobrevibacter smithii ATCC 35061 was isolated from a sewage digester in Gainesville, Florida and will be used for comparative analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methanococcoides burtonii DSM 6242 \t</h1>
<div class="content">
<p>Methanococcoides burtonii is a strictly anaerobic, slightly halophilic archeon that was isolated from the anoxic hypolimnion (layer of cold noncirculating water below the thermocline) of Ace Lake, Antarctica. Motility is via a single monotrichous flagellum. Growth occurs at a temperature between 1.7-29.5oC with optimal growth at 23.4oC. In addition this organism requires 0.1-0.5 M NaCl and either magnesium sulfate or magnesium chloride at a concentration greater than 0.01 M for growth.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The genome sequence of this strain will be compared to other methanogen genomes in order to identify a subset of genes unique to the cold-adapted methanogen.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methanococcus maripaludis C5</h1>
<div class="content">
<p>Methanococcus maripaludis is an obligately anaerobic, mesophilic, methane-producing archaeon. Motility is by means of polar tufts of flagella. Growth occurs at 20-45oC with a pH of 6.5-8. Growth on hydrogen and carbon dioxide results in the production of methane as a waste product. M. maripaludis stands out among methanogenic archaea as an ideal model species because of fast reproducible growth and effective genetic tools.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Methanococcus maripaludis C5 is the natural host for plasmid pURB500 which has been sequenced and used to design a shuttle vector for M. maripaludis and E. coli. Methanococcus maripaludis C5 will be used for comparative analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methanocorpusculum labreanum Z</h1>
<div class="content">
<p>Methanocorpusculum labreanum fixes carbon dioxide, produces methane and is involved in biomethanation processes (wastewater treatment, sewage, landfills).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Methanocorpusculum labreanum Z was isolated from surface sediment from the LaBrea Tar Pits in Los Angeles, California. This organism will be used for comparative genomics with other methanogens.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methanosaeta thermophila PT</h1>
<div class="content">
<p>Methanosaeta thermophila is a thermophilic methanogen found widely distributed in the environment. This organism can metabolize acetate into methane and may be useful as a producer of biofuels.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This thermophilic strain was isolated from anaerobic digester sludge and is the type strain for the species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methanosphaera stadtmanae DSM 3091</h1>
<div class="content">
<p>This organism produces methane from methanol and hydrogen unlike most methanogens that use carbon dioxide and hydrogen.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from human feces and is the type strain for the species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methanospirillum hungatei JF-1</h1>
<div class="content">
<p>This organsim is a methanogenic archaeon that has been isolated from bovine rumens, anaerobic sludge digestors and mud. It is structurally unusual in that each cell has an inner cell wall and a sheathed filament enclodes numerous individual cells which are separated by spacer plugs.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain  is the type strain and was isolated from sewage sludge.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methylacidiphilum infernorum V4</h1>
<div class="content">
<p>Methylacidiphilum infernorum is one of the first reported methanotrophs from the Verrucomicrobia phylum. It is extremely acidophilic and is able to utilize methane as sole carbon source.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Methylacidiphilum infernorum V4 was first discovered in Hell's Gate geothermal area in New Zealand.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methylibium petroleiphilum PM1</h1>
<div class="content">
<p>Methylibium petroleiphilum is a methylotroph (able to utilize reduced one-carbon compounds) able to degrade methyl tertiary-butyl ether (MTBE) under aerobic conditions. MTBE is a gasoline additive used as an oxygenate and to raise the octane number. MTBE easily pollutes large quantities of groundwater when gasoline with MTBE is spilled or leaked at gas stations.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Methylibium petroleiphilum strain PM1 was isolated in 1998 from the biofilter of a treatment plant in an oil refinery in Los Angeles, California. Strain PM1 is capable of degrading aromatic compounds such as benzene, toluene and xylenes. The MTBE and aromatic hydrocarbon degradation mechanisms used by strain PM1 are poorly understood, for which the genome sequence provides scientist with more information to improve the bioremediation properties of this isolate.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methylobacillus flagellatus KT</h1>
<div class="content">
<p>Methylobacillus flagellatus is a methylotroph, able to grow on compounds such as methanol and methyamines. These organisms play a role in the environmental cycling of carbon and nitrogen.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain contains multiple formaldehyde degradation pathways including a ribulose monophosphate (RMP) cycle and a linear pathway utilizing tetrahydromethanopterin-dependent enzymes. Degradation of formaldehydge is part of the process of incorporating carbon into the cell, and as this organism is an obligate methylotroph and can only utilize methanol and methylamine as the sole carbon and energy sources, conversion of intermediates is both an important part of the growth of cell and an exploitable process for industrial purposes. This strain was isolated from activated sludge found at the wastewater treatment plant in Moscow, Russia and is the fastest growing methylotroph discovered. The fast growth and conversion rate make it useful in industrial production of biomolecules.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methylocella silvestris BL2</h1>
<div class="content">
<p>Methylocella silvestris was isolated from acidic forest soil near Marburg, Germany. This organism is a facultative methanotroph, able to grow on organic acids and alcohols as well as methane.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Methylocella silvestris BL2  is the type strain and will be used for comparative analysis with obligate methanotrophs.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Methylocella silvestris BL2</h1>
<div class="content">
<p>Methylocella silvestris was isolated from acidic forest soil near Marburg, Germany. This organism is a facultative methanotroph, able to grow on organic acids and alcohols as well as methane.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Methylocella silvestris BL2 is the type strain and will be used for comparative analysis with obligate methanotrophs.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Microcystis aeruginosa NIES-843</h1>
<div class="content">
<p>Microcystis aeruginosa is a common cyanobacterium which can form blooms from late summer to early autumn in nutrient-rich lakes, ponds, and reservoirs. This organism produces a hepatotoxin, microcystin, which can kill fish, birds, and farm animals. This toxin also causes skin irritation and gastrointestinal symptoms in humans.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Microcystis aeruginosa NIES-843 is the type strain and the genome sequence of this organism will provide information on the production and regulation of microcystin.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Moorella thermoacetica ATCC 39073</h1>
<div class="content">
<p>Moorella thermoacetica, formerly Clostridium thermoaceticum, can grow on hydrogen and fix carbon dioxide or carbon monoxide with the production of acetate. It can also use nitrite as a terminal electron acceptor. It is found at the bottom of stagnant ponds and forms extremely heat-resistant spores.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain will be used for comparative analysis with other acetogenic bacteria.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium JLS</h1>
<div class="content">
<p>This genus comprises a number of Gram-positive, acid-fast, rod-shaped aerobic bacteria and is the only member of the family Mycobacteriaceae within the order Actinomycetales. Like other closely related Actinomycetales, such as Nocardia and Corynebacterium, mycobacteria have unusually high genomic DNA GC content and produce mycolic acids as major components of their cell wall.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from creosote-contaminated soil from the Champion International Superfund site in Libby, Montana. This microbe, along with some others collected at this site, is able to rapidly mineralize C-labeled pyrene. Phylogenetic studies based on the rRNA 16S gene confirmed that strains collected at this site were distinct from one another and also from other known Mycobacterium isolates.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium KMS</h1>
<div class="content">
<p>This organism was isolated from creosote-contaminated soil from the Champion International Superfund site in Libby, Montana. This microbe, along with some others collected at this site, are able to rapidly mineralize C-labeled pyrene. Phylogenetic studies based on the rRNA 16S gene confirmed that strains collected at this site were distinct from one another and also from other known Mycobacterium isolates.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium abscessus ATCC 19977</h1>
<div class="content">
<p>Mycobacterium abscessus, formerly Mycobacterium chelonae subsp.  abscessus, is a rapid-growing Mycobacterium sp., commonly isolated from soil and water. This organism causes a chronic lung infection, similar to tuberculosis, in patients with cystic fibrosis. Mycobacterium abscessus also causes wound and ear infections, and in immunosuppressed patients causes a generalized skin infection. This organism is very resistant to many commonly used antibiotics, which makes treatment difficult.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Mycobacterium abscessus ATCC 19977 is the type strain and will be used for comparative genomics.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium avium 104</h1>
<div class="content">
<p>Mycobacterium avium comprises different species, serovars and morphologic forms that vary in virulence. Mycobacterium avium is ubiquitous in the environment, and can be found in stagnant waters and soils. This organism causes tuberculosis in birds and disseminated infections in immunocompromized humans (the elderly, children, and especially patients with AIDS). Infection results in a characteristic pulmonary disease which requires expensive drug therapy for successful treatment. Most prevalent colony morphotypes are smooth opaque, smooth transparent and rough, with the last two being the faster growers in vivo. All three morphotypes are interchangeable and can persist in the lung. Colony morphology correlates with the presence of specific glycopeptidolipids (GPL) responsible for serotyping differences.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain is a common representative of the species complex. It was derived from an AIDS patient and has been characterized for virulence in the murine model of low-dose aerosol infection in that it could colonize the lung, proliferate within the tissue and disseminate to other organs. This strain will be used for comparative analysis with other Mycobacterium species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium bovis BCG Pasteur 1173P2</h1>
<div class="content">
<p>Mycobacterium bovis is the causative agent of classic bovine tuberculosis, but it can also cause the disease in humans, especially if contaminated milk is consumed without prior pasteurization. The Mycobacterium bovis complex is a diverse group of species, serovars and morphotypes that cause tuberculosis-like diseases in animals and humans. Pasteurization of milk is a major preventitive factor in transmission of bovine tuberculosis to humans. However, spreading the disease through milk and dairy products is still a concern in underdeveloped countries where pasteurization is not practiced. The pathology in cows is similar to the pathology of Mycobacterium tuberculosis in humans, with pulmonary TB leading to chronic debilitation, coughing, and further systemic spread to other organs. In addition, 1 to 2% of infected cows develop mycobacterial mastitis that results in shedding of the bacteria into the milk.\n\nMycobacterium bovis strain BCG was used to produce BCG (Bacille de Calmette et Guan) vaccine, a well-known tuberculosis vaccine, originally developed by Calmette and Guan in the 1920's by multiple subculturings that resulted in attenuation (weakened virulence) of the strain.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was used to produce the antigen which was utilized for vaccine development.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium gilvum PYR-GCK</h1>
<div class="content">
<p>Mycobacterium gilvum is a soil bacterium able to degrade pyrene and other aromatic hydrocarbons.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Mycobacterium gilvum PYR-GCK was isolated from river sediment and will be used for comparative genomics.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium leprae Br4923</h1>
<div class="content">
<p>Mycobacterium leprae  is a close relative of M. tuberculosis. However, compared to the latter, the genome of M. leprae is smaller due to reductive genome evolution, with many important metabolic activities including siderophore production, part of the oxidative chain, most of the microaerophilic and anaerobic respiratory chains, and numerous catabolic systems and their regulatory circuits eliminated due to extensive recombination events between dispersed repetitive sequences. Twenty seven percent of the M. leprae genome consists of pseudogenes that have functional counterparts in M. tuberculosis. One of the possible reasons for elimination of these genes is the loss of the DnaQ-mediated proofreading activities of DNA polymerase III. It is evident that this species has undergone massive genome reduction over time as a result of its parasitic nature, discarding more than half its genes and rendering it the most striking example of genome reduction in a microbial pathogen.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Mycobacterium leprae strain Br4923 was isolated from a human skin biopsy in Brazil, and passaged in nude mice and armadillos.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium marinum M</h1>
<div class="content">
<p>Mycobacterium marinum is commonly found in fresh and salt water in North America. This bacterium, a close relative of M. tuberculosis, causes a tuberculosis-like disease in frogs, fish and other cold-blooded animals, and a peripheral granulomatous disease in humans. Human infections can be persistent and may lead to systemic infection and arthritis. M. marinum is photochromogenic, i.e. it produces bright yellow pigmentation (mainly beta-carotene) upon exposure to visible light. Infection is followed by a 2-8 week incubation period resulting in suppuration (pus) and organized ulcerating granulomas (tumorous lesions), predominantly localized in lymphatic and tendon sheaths, and cooler body tissues in the extremities.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Mycobacterium marinum strain M is a well-studied human isolate that is susceptible to a relatively wider range of antibiotics than other M. marinum isolates and thus has a potential to be easier to use in genetic and molecular biological studies.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium smegmatis MC2 155</h1>
<div class="content">
<p>Mycobacterium smegmatis was initially isolated from human smegma. It is associated with soft tissue lesions following trauma or surgery. It is also reported as a possible factor in penile carcinogenesis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain (MC2 155) is a mutant of M. smegmatis. It was isolated in 1990. Unlike other strains of this species, MC2 can be 10 to 100 thousand times more efficiently transformed with plasmid vectors using electroporation than the parental strain and thus is invaluable in analysis of mycobacterial gene function, expression and replication.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium tuberculosis F11</h1>
<div class="content">
<p>Mycobacterium tuberculosis (MTB) is a pathogenic bacterial species in the genus Mycobacterium and the causative agent of most cases of tuberculosis. First discovered in 1882 by Robert Koch, M. tuberculosis has an unusual, waxy coating on the cell surface (primarily mycolic acid), which makes the cells impervious to Gram staining; acid-fast techniques are used instead. The physiology of M. tuberculosis is highly aerobic and requires high levels of oxygen. Primarily a pathogen of the mammalian respiratory system, MTB infects the lungs, causing tuberculosis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain (genotype F11) represents the largest portion of isolates recovered from tuberculosis patients during a TB epidemic in the Western Cape of South Africa. It is also found in other parts of the world.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium tuberculosis H37Ra</h1>
<div class="content">
<p>This attenuated strain is a standard isolate from the CDCC. It is derived from the original human-lung H37 isolate in 1934, and has been used extensively worldwide in mycobacterial virulence research.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium tuberculosis H37Rv</h1>
<div class="content">
<p>This strain has been derived from the original human-lung H37 isolate in 1934, and has been used extensively worldwide in biomedical research. Unlike some clinical isolates, it retains full virulence in animal models of tuberculosis and is susceptible to drugs and receptive to genetic manipulation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium ulcerans Agy99</h1>
<div class="content">
<p>Infection by Mycobacterium ulcerans causes the formation of skin ulcers which, if left untreated, can lead to extensive scarring and/or amputation. This disease, called Buruli or Bairnsdale ulcer, is prevalent throughout much of Africa and Asia. Mycobacterium ulcerans produces a cytotoxin, mycolactone, which causes the destruction of skin tissue seen in this disease. This organism is resistant to antibiotics and treatment involves the surgical removal of infected tissues. Mycobacterium ulcerans has been identified in stagnant water and in aquatic insects. It is thought that spread of this disease is through contact with water or through the bite of these aquatic insects.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain causes Buruli ulcer, the third most common mycobacterial pathogen after Mycobacterium tuberculosis and Mycobacterium leprae. The disease has increased dramatically through central and West Africa since the late 1980s. A whole genome shotgun approach was used for sequencing this strain. Three libraries were constructed (2-3kbp, 3-5kb and 5-10kbp) using the vector pCDNA2.1. The assembly was scaffolded using a large insert library (60-120kbp) with pBeloBAC11.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycobacterium vanbaalenii PYR-1</h1>
<div class="content">
<p>Mycobacterium vanbaalenii is a fast-growing soil mycobacterium which can degrade polycyclic aromatic hydrocarbons such as fluoranthene, pyrene, phenanthrene. This organism may be useful for bioremediation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain is the type strain and was isolated from contaminated sites exposed to petrogenic chemicals in the watershed of Redfish Bay, Texas, in 1986.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycoplasma agalactiae  PG2</h1>
<div class="content">
<p>Mycoplasma agalactiae causes mastitis (infection of the mammary glands), arthritis, and pneumonia in sheep and goats. Asymptomatic animals can carry and spread the disease, makng detection and erradication difficult. This disease is widespread in the Mediterranean and is of considerable economic importance, causing decreased yields of milk, and mortality or slowed growth of kids and lambs.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Mycoplasma agalactiae strain PG2 is the type strain and will be used for comparative genomics.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycoplasma arthritidis 158L3-1</h1>
<div class="content">
<p>This organism causes arthritis in both rats and mice. Strains of Mycoplasma arthritidis can be virulent or avirulent depending, in part, on the presence of the MAV1 bacteriophage.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This is a virulent strain which carries the lysogenic bacteriophage, MAV1, integrated into the chromosome.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycoplasma capricolum capricolum ATCC 27343</h1>
<div class="content">
<p>Mycoplasma capricolum is primarily a pathogen of goats, causing caprine mycoplasma, though it has also been found in sheep and cows. In goats, Mycoplasma capricolum is highly destructive, causing important mortality and morbidity. The primary clinical manifestation is severe arthritis (polyarthritis). The infection progresses in the form of septicemia, with severe lesions in the joints leading to permanent lameness. In young goats it also produces fever.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain is the type strain for the species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycoplasma hyopneumoniae 7448</h1>
<div class="content">
<p>Mycoplasma hyopneumoniae causes enzootic pneumonia in swine. The disease is considered the most prevalent infection in swine and results in annual economical losses of more than US $200 million. Its main symptoms are coughing and reduction in feeding and activity. The bacteria colonize the ciliated epithelial cells of the swine respiratory tract, compromising the host immune system and allowing secondary, primarily bacterial, infections. Transmission is primarily through direct contact.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This pathogenic strain was isolated from an infected pig in Lindoia do Sul, Santa Catarina, in Brazil.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycoplasma hyopneumoniae J</h1>
<div class="content">
<p>This strain was isolated from an agricultural sample. It is nonpathogenic and exhibits decreased adhesion capacity to porcine cilia.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Mycoplasma synoviae 53</h1>
<div class="content">
<p>This bacterium, like the closely related Mycoplasma gallisepticum, causes chronic respiratory disease in chickens. This infection results in considerable economic losses in the poultry industry.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from a broiler breeder in the state of Parana in Brazil
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Myxococcus xanthus DK 1622</h1>
<div class="content">
<p>Myxococcus xanthus undergoes a complex development and differentiation pathway. Single cells can be motile by "adventurous motility". When cell density increases, the organism switches to "social motility" where aggregates of cells can gather together into masses termed fruiting bodies that may consist of up to 100 000 cells. Upon nutrient depletion, the cells in these fruiting bodies sporulate. The decision to switch to social motility requires intercellular communication that is partly dependent on cellular density. The motility system is not dependent on flagella like most bacteria, but instead relies on twitching pili: short extracellular appendages that may function analogously to oars in a rowboat. This organism is also utilized industrially to produce the anticancer drug epithilone.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Myxococcus xanthus strain DK 1622 has a large genome (9.14Mb) containing more than 1500 gene duplication events. These gene duplications more commonly involve genes used in cell to cell signaling, molecule sensing and transcription control.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Natranaerobius thermophilus JW/NM-WN-LF</h1>
<div class="content">
<p>Natranaerobius thermophilus is a halophilic, alkalithermophilic bacterium isolated from saline lakes in the Wadi El Natrun depression, Egypt.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Natranaerobius thermophilus JW/NM-WN-LF  is the type strain and will be used for comparative analysis with other halophilic bacteria.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Natronomonas pharaonis</h1>
<div class="content">
<p>Natronomonas pharaonis is able to survive at high salt and pH conditions which results in limited nitrogen availability through ammonium. In order to compensate for this, Natronomonas pharaonis has developed three systems to promote nitrogen assimilation: direct uptake of ammonia, uptake of nitrate, and uptake of urea. Another problem with high pH environments is the use of a proton gradient for the generation of ATP, which other alkaliphiles have adapted to by substitution of sodium ions for protons. However, this organism utilizes protons for ATP generation as determined by experimental data.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nautilia profundicola AmH</h1>
<div class="content">
<p>Nautilia profundicola is a thermophilic, hydrogen-oxidizing, sulfur-reducing chemoautotroph that uses the reductive TCA cycle for carbon fixation. It is a member of the deepest branch of the Epsilonproteobacteria and is a model organism of the anaerobic autotrophic Epsilonproteobacteria at hydrothermal vents.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Nautilia sp.strain AM-H, also known as Nautilia profundicola, was the first published Epsilonproteobacterium to be isolated from hydrothermal vents. This strain was isolated from the epidermis of Alvinella pompejana.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Neisseria gonorrhoeae NCCP11945</h1>
<div class="content">
<p>Neisseria gonorrhoeae is one of two pathogenic Neisseria, this species causes the sexually transmitted disease (STD) gonorrhea, which is the leading reportable STD in adults in the USA. This human-specific organism colonizes and invades the mucosal surface of the urogenital epithelium, crosses the epithelial barrier, and ends up multiplying on the basement membrane. The result is severe painful inflammation and purulent discharge.\n\nPathogenic functions include the antigenically variable type IV pilus and the opacity (Opa) proteins. The Opa proteins are responsible for the opaque colony phenotype due to the tight junctions between adjacent Neisseria, and are also responsible for tight adherence to host cells. This organism, like Neisseria meningitidis, is also naturally competent for DNA uptake.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Neisseria gonorrhoeae NCCP11945 was isolated in 2002 from a vaginal smear of a Korean patient. This is a multidrug-resistant strain and will be used to understand the prevalence, antibiotic resistance, and importance of horizontal gene transfer within this important, naturally competent organism.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Neisseria meningitidis FAM18</h1>
<div class="content">
<p>Neisseria meningitidis is a heterotrophic gram-negative diplococcal bacterium best known for its role in meningitis and other forms of meningococcal disease such as meningococcemia. N. meningitidis is a major cause of morbidity and mortality in childhood in industrialized countries and is responsible for epidemics in Africa and in Asia.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain FAM18 is a member of the serogroup C ET-37 complex and was isolated from a patient with meningococcal septicemia.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Neisseria meningitidis Z2491</h1>
<div class="content">
<p>Neisseria meningitidisis the second of two pathogenic Neisseria, this organism causes septicemia and is the leading cause of life-threatening meningitis (inflammation of the meninges, the membrane surrounding the brain and spinal cord) in children. This organism typically residies in the nasopharynx cavity but can invade the respiratory epthelial barrier, cross into the bloodstream and the blood brain barrier, and cause inflammation of the meninges. The antiphagocytic capsular polysaccharide (CPS) aids in virulence. There are 5 serogroups based on CPS typing (A, B, C, Y, and W135). Serogroup B results in sporadic outbreaks in Europe and North America, and there is no vaccine currently. The use of a vaccine is hindered by the fact that the serogroup B capsular polysaccharide vaccine recognizes a common human carbohydrate, and vaccines against the surface proteins suffer due to the antigenic variability of those proteins.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This is a serogroup A strain isolated in Gambia in 1983.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Neorickettsia sennetsu Miyayama</h1>
<div class="content">
<p>Like other Ehrlichia, this organism is an obligate intracellular pathogen. It can be transmitted from fish flukes (trematodes) to humans, and was one of the first Ehrlichia that was found to cause a human-specific illness. People eating uncooked fish are particularly at risk of acquiring the disease, hence the predominance in Japan and Malaysia. The symptoms include fever, headaches, anorexia, and lymphadenopathy (enlargement of the lymph nodes) among others. The bacterium primarily infects monocytes and macrophages where it exists in vacuoles.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This organism was first described in 1953 in Japan as causing Sennetsu fever.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nitratiruptor SB155-2</h1>
<div class="content">
<p>This strain was isolated from a deep-sea hydrothermal vent in the Iheya North field in the Mid-Okinawa Trough as part of a larger diversity study.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nitrobacter hamburgensis X14</h1>
<div class="content">
<p>Nitrobacter hamburgensis is commonly isolated from freshwater, soil, and sewage sludge. This organism has been used in biofilms to remove nitrogen from wastewater.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Nitrobacter hamburgensis X14 was isolated from soil and will be used for comparative genomics with other Nitrobacter spp.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nitrobacter winogradskyi Nb-255</h1>
<div class="content">
<p>Nitrobacter winogradskyi is commonly isolated from soil, fresh and sea water, sewage, and compost. This organism can grow anaerobically using nitrate as the electron acceptor, forming nitrite, nitric oxide, and nitrous oxide.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Nitrobacter winogradskyi Nb-255 will be used for comparative genomics with other Nitrobacter spp.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nitrosomonas eutropha C91</h1>
<div class="content">
<p>This species consists of ammonia-oxidizing bacteria that catalyze the oxidation of ammonia to nitrite and contribute to the global cycling of nitrogen (a process called nitrification). This obligate chemolithoautotroph is studied for that reason</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Nitrosomonas eutropha C91 was isolated from a sewage disposal plant in Chicago. The genome sequence will provide information on the proteins involved in ammonia oxidization.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nitrosopumilus maritimus SCM1</h1>
<div class="content">
<p>Nitrosopumilus maritimus is an autotrophic ammonia-oxidizing archaeon.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Nitrosopumilus maritimus SCM1 was isolated from a salt-water aquarium. This organism is able to fix carbon and oxidize ammonia and will be used for comparative analysis.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nitrosospira multiformis ATCC 25196</h1>
<div class="content">
<p>Nitrosospira multiformis is a lithotrophic, ammonia-oxidizer, deriving its energy from the oxidiation of ammonia to nitrite. In a two step process, ammonia is oxidized to hydroxylamine by ammonia monooxygenase; hydroxylamine is then oxidized to nitrite by hydroxylamine oxidoreductase. This organism is commonly isolated from neutral soils and plays an important role in the terrestrial nitrogen cycle.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Nitrosospira multiformis ATCC 25196 was isolated from soil in Suriname and will be used for comparative genomics with other ammonia-oxidizing bacteria.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nocardioides JS614</h1>
<div class="content">
<p>Nocardioides species are common soil bacteria. Many of these organisms are able to degrade herbicides like phenanthrene and may be useful for bioremediation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Nocardioides JS614 grows quickly and efficiently on media containing vinyl chloride and ethene. If starved of vinyl chloride for one day, it will not recover for more than 40 days. This strain was placed in the Nocardioides genus on the basis of 16S rRNA analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nostoc punctiforme PCC 73102</h1>
<div class="content">
<p>Nostoc punctiforme can form nitrogen-fixing symbiotic relationships with plants and fungi such as the bryophyte Anthoceros punctatus. The relationship is relatively simple as compared to the Rhizobial symbiotic relationship. In the presence of the plant, hormogonia (short motile filaments) infect the plant, and then form long heterocyst-containing (nitrogen-fixing differentiated bacterial cells) filaments. The bacterial cell receives carbon sources in exchange for fixed nitrogen.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The strain was isolated from a symbiotic association with the gymnosperm cycad Macrozamia sp. (Rippka and Herdman, 1992). It typically grows in freshwater habitats.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Nostoc sp</h1>
<div class="content">
<p>This genera of cyanobacteria are typically terrestrially-associated and are especially found in limestone or nutrient-poor soils. They are very similar to Anabaena spp. and historically they have been distinguished on the basis of morphological and life cycle characteristics. Nostoc spp. can grow heterotrophically or photoheterotrophically, and form heterocysts for nitrogen fixation.\nfixation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Novosphingobium aromaticivorans DSM 12444</h1>
<div class="content">
<p>This organism is unusual in that it has glycosphingolipid in the cell envelope instead of the lipopolysaccharide found in most other gram negative organisms. It is typically isolated from a wide range of environmental sites. This bacterium is most noteworthy for its ability to degrade a wide variety of aromatic hydrocarbons.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain, also known as F199, was isolated from a sample obtained at a depth of 410 m from a borehole sample that was drilled at the Savannah River Site in South Carolina, USA, by the Department of Energy.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Oenococcus oeni PSU-1</h1>
<div class="content">
<p>Oenococcus oeni is another member of the lactic acid bacteria and it occurs naturally in marshes and similar environments. It carries out malolactic conversion during secondary fermentation in wine production which is the conversion of malic acid to lactic acid with a concomitant rise in pH, making the wine microbiologically stable and enhancing the sensory properties of the wine (aroma, flavor, and texture). This step occurs after the yeast have converted the sugars in the wine to ethanol and carbon dioxide. The organism's high tolerance to sulfite and ethanol mean that it will be the predominant organism in the wine at the end of fermentation where it cleans up the remaining sugars and converts the bitter-tasting malic acid.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated at Penn State University and is used commercially for malolactic fermentation in wines.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Oligotropha carboxidovorans OM5</h1>
<div class="content">
<p>Oligotropha carboxidovorans is able to use carbon monoxide as an energy source. The gene for carbon monoxide dehydrogenase, a key enzyme in the utilization of carbon monoxide, is carried on a low-copy-number megaplasmid along with genes required for the utilization of hydrogen and carbon dioxide.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Oligotropha carboxidovorans OM5 is the type strain and will be used for comparative analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Onion yellows phytoplasma OY-M \t</h1>
<div class="content">
<p>Onion yellows Phytoplasmas inhabit phloem (food-conducting vascular tissue) sieve elements of plants where they cause a variety of diseases. Transmission between plants is via insects that feed on the phloem. Symptoms include a yellowing of the leaves and reduced leaf size which leads to reduced plant size and lower yields. There is great interest in sequencing these organisms since they are currently unculturable and examination of the genome may lead to methods to deal with the diseases they cause including the development of antimicrobial agents.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain (OY-M) is derived from a wild-type disease-causing strain (OY-W; onions yellow disease) which was isolated in Saga Prefecture, Japan, in 1982, and shows mild symptoms and does not cause stunting nor phloem hyperplasia (excessive increase in number of cells). The genome consists of a circular chromosome and two smaller extrachromosomal elements. This bacterium lacks a phosphotransferase system, pentose phosphate pathway, and surprisingly, any ATP synthase genes, suggesting it imports ATP from the cytoplasm of the host cell in which it resides.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Opitutus terrae PB90-1</h1>
<div class="content">
<p>Opitutus terrae is a member of the Verrucomicrobia, a diverse phylum of organisms common in many terrestrial and aquatic habitats. This organism was isolated from rice paddy soil and has been detected in peat bog soil and other anoxic terrestrial environments.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Opitutus terrae PB90-1 will be used for comparative genomics with other Verrucomicrobia.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Orientia tsutsugamushi Boryong</h1>
<div class="content">
<p>Orientia tsutsugamushi is the causative agent of scrub typhus, and like other members of the Rickettsiales is an obligate intracellular organism. Scrub typhus, which occurs in rural and agricultural areas, is one of the most common infectious diseases in southeast Asia where an estimated 1 million cases occur each year. Many cases are mild, but if left untreated, a variety of serious illnesses can arise and fatality rates can range from 1-35%. Transmission occurs via an insect vector through the bite of the larval trombiculid mite (chiggers). The bacterium is transmitted transovarially in mites (from females to their offspring).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from a human in 1995 in Boryong, Korea.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Parabacteroides distasonis ATCC 8503</h1>
<div class="content">
<p>Parabacteroides distasonis is one of the most common fecal isolates, however, this organism is rarely isolated from clinical specimens.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This organism will be used for comparative genomics with other Bacteroides species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Paracoccus denitrificans PD1222</h1>
<div class="content">
<p>Paracoccus denitrificans is commonly found in soil and sewage sludge. This organism is able to survive in both aerobic and anaerobic environments due to its ability to use both oxygen and nitrogenous oxides making it a popular model organism for the study of electron transfer and and energy conservation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain (Pd 1222) is a genetically modified version of strain DSM 413 (type strain). The genome consists of 3 molecules (megaplasmids).
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Parvibaculum lavamentivorns DS-1</h1>
<div class="content">
<p>Parvibaculum lavamentivorns is able to degrade linear alkylbenzenesulfonate and other surfactants making this organism useful in industrial and bioremediation applications.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Parvibaculum lavamentivorns DS-1 is the type strain of the species. The genome sequence will provide information on the production and regulation of enzymes involved in the degradation of surfactants.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pediococcus pentosaceus ATCC 25745</h1>
<div class="content">
<p>Pediococcus pentosaceus can be isolated from a variety of plant materials and bacterial-ripened cheeses. This organism is used as an acid producing starter culture in the fermentation of some sausages, cucumbers, green beans, soy milk, and silage. They are also a typical component of the adventitious microflora of most cheese varieties during ripening. Some strains have been reported to contain several (3-5) resident plasmids that render the bacterium capable of fermenting some sugars (raffinose, melibiose, and sucrose), as well as producing bacteriocins.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This is the wild-type strain.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pelobacter carbinolicus</h1>
<div class="content">
<p>Pelobacter carbinolicus is commonly isolated from marine and freshwater sediments, and sewage sludge. This organism can make up a significant portion of the anaerobic microbial community in these environments. Pelobacter carbinolicus can ferment ethanol in the presence of hydrogen-utilizing bacteria via interspecies hydrogen transfer. The utilization of hydrogen reduces the hydrogen partial pressure making ethanol fermentation by Pelobacter carbinolicus energetically favorable. Pelobacter carbinolicus is also able to grow using iron and sulfur as terminal electron acceptors. This organism is closely related to the sulfur-reducing Desulfuromonas spp. and iron-reducing Geobacter spp..</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pelobacter propionicus DSM 2379.</h1>
<div class="content">
<p>Pelobacter propionicus is ubiquitous in both marine and fresh water, and in anaerobic sedmiments. It is able to convert the unsaturated hydrocarbon acetylene to to acetate and ethanol via acetylaldehyde as an intermediate. These microorganisms may survive in some sediments as an Fe(III) or elemental sulfur reducer as well as growing fermentatively as an ethanol-oxidizing acetogen. Organisms that can completely oxidize organic acids with Fe(III) serving as the sole electron acceptor are responsible for most of the oxidation of organic matter in anaerobic sediments.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain is used for comparative genome analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pelodictyon luteolum DSM 273</h1>
<div class="content">
<p>Pelodictyon luteolum (also called Chlorobium luteolum) is a phototropic green sulfur bacteria that inhabits hydrogen sulfide containing water and freshwater mud, brackish waters and marine environments.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Pelodictyon luteolum DSM 273 is the type strain
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pelodictyon phaeoclathratiforme BU-1</h1>
<div class="content">
<p>Pelodictyon phaeoclathratiforme is a phototrophic green sulfur bacteria that inhabits hydrogen sulfide-containing water and freshwater mud, brackish waters and marine environments. This organism forms net-like microcolonies and contains gas vescicles. It is believed that the gas vesicles provide the buoyancy necessary to keep the cell at the optimal depth for photosynthesis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Pelodictyon phaeoclathratiforme BU-1 was isolated from lake water in 1989.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pelotomaculum thermopropionicum SI</h1>
<div class="content">
<p>In syntrophic association with hydrogenotrophic methanogens, Pelotomaculum thermopropionicum can utilize propionate, ethanol, lactate, ethylene glycol, 1-butanol, 1-propanol, 1-pentanol and 1,3-propanediol. In pure culture, Pelotomaculum thermopropionicum can ferment pyruvate and fumarate. Fumarate can also be used as an electron acceptor in the presence of propionate, ethanol or lactate as an electron donor. This is the first mesophilic, syntropic propionate-oxidizing species known which is not a member of the delta-proteobacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This type strain was isolated from granular sludge from a thermophilic upflow anaerobic sludge blanket (UASB) reactor. It was isolated both in pure culture and in co-culture with methanogens.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Petrotoga mobilis SJ95</h1>
<div class="content">
<p>Petrotoga mobilis, a member of the Thermotogales, has the characteristic morphology of one or more cells contained in a sheath-like envelope which extends beyond the cell wall. Petrotoga species appear to be common members of the oil well microbial community. Petroleum reservoirs are a unique subsurface environment characterized by high temperatures, moderate to high salt concentrations, and abundant organic matter.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Petrotoga mobilis SJ95 is an anaerobic thermophile, isolated from the production waters of a North Sea oil reservoir. This is the type strain for this species and will be used for comparative genomics with other thermophiles.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Polaromonas JS666</h1>
<div class="content">
<p>The type species of this genus was originally isolated from marine waters in the Antarctic. It was classified as a novel group in the betaproteobacteria based on 16S ribosomal DNA analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
strain JS666 was isolated from sediment contaminated with cis-dichloroethane (cDCE), a common pollutant resulting from widespread manufacture and use of industrial solvents. This bacterium is the only known organism capable of using cDCE as a sole carbon and energy source. The ability of this strain to convert ethene to epoxyethane suggests that the first step in the cDCE biodegradation pathway is the oxidation of cDCE to an epoxide compound. Bacteria that are able to grow on cDCE are rare, and have only been found in very few highly selective artificial environments. The discovery of this bacteria may provide a low cost, self-sustaining bioremediation method in areas where cDCE is a problem contaminant.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Polaromonas naphthalenivorans  CJ2</h1>
<div class="content">
<p>Polaromonas naphthalenivorans was isolated from naphthalene-contaminated, freshwater sediment. This organism is capable of aerobic degradation of naphthalene at temperatures of less than 20oC. Naphthalene is a polycyclic aromatic hydrocarbon (PAH). PAHs can be toxic and/or carcinogenic, therefore PAH contamination is of considerable concern.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The ability of Polaromonas naphthalenivorans strain CJ2 to degrade PAHs at low temperatures makes it a potential bioremediation agent. The genome sequence will provide information on the mechanisms and regulation of naphthalene degradation.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Polynucleobacter necessarius</h1>
<div class="content">
<p>This group of organisms is common in freshwater, comprising up to 60% of the bacteria in freshwater habitats. Polynucleobacter species are also known to be endosymbionts of ciliates. These strains live within the cytoplasm and are essential for their host organisms.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain STIR1 is an obligate intracellular endosymbiont of the cilliate Euplotes aediculatus and will be used for comparative genomics.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Porphyromonas gingivalis</h1>
<div class="content">
<p>Porphyromonas gingivalis belongs to the Bacteroides genus and is a non-motile, gram-negative, rod-shaped, anaerobic pathogenic bacterium. It forms black colonies on blood agar. It is found in the oral cavity, where it is implicated in certain forms of periodontal disease, as well as the upper gastrointestinal tract, respiratory tract, and in the colon. Collagen degradation that is observed in Chronic Periodontal Disease results in part from the collagenase enzymes of this species. Immunoglobulin G (IgG) is the specific antibody for porphyromonas gingivalis. And in patients with this bacteria you will kind high levels of IgG in the serum</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Prochlorococcus marinus AS9601</h1>
<div class="content">
<p>Prochlorococcus marinus is a free-living marine organism that is one of the most abundant, as well as the smallest, on earth, and contributes heavily to carbon cycling in the marine environment. This cyanobacterium grows in areas of nitrogen and phosphorus limitation and is unique in that it utilizes divinyl chlorophyll a/b proteins as light-harvesting systems instead of phycobiliproteins. These pigments allow harvesting of light energy from blue wavelengths at low light intensity. There are two ecotypes, one that is low light adapted and one that is high light adapted.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Prochlorococcus marinus strain AS9601. This cyanobacterium was collected in the Arabian Sea at a depth of 50 meters and isolated by filter fractionation. It belongs to Clade HL-II, the most abundant group in the North Atlantic and North Pacific Oceans, often constituting over 90% of the total population. Strain AS9601 outgrows MIT9312 in temperatures exceeding 28oC, and can grow at significantly higher temperatures.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Prochlorococcus marinus</h1>
<div class="content">
<p>This non-motile bacterium is a free-living marine organism that is one of the most abundant, as well as the smallest, on earth, and contributes heavily to carbon cycling in the marine environment. This cyanobacterium grows in areas of nitrogen and phosphorus limitation and is unique in that it utilizes divinyl chlorophyll a/b proteins as light-harvesting systems instead of phycobiliproteins. These pigments allow harvesting of light energy from blue wavelengths at low light intensity. There are two ecotypes, one that is low light adapted and one that is high light adapted.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Prochlorococcus marinus MIT 9211</h1>
<div class="content">
<p>This strain was isolated from the equatorial Pacific at a depth of 83 meters by filter fractionation. MIT9211 is the type strain of the high B/A III (LLII) clade, and makes up a small but significant fraction of the population. It is unique among Prochlorococcus in its photophysiology and pigment composition.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Prochlorococcus marinus MIT 9301</h1>
<div class="content">
<p>This strain was isolated from the Sargasso Sea at a depth of 90 m (34o 45.5'N; 66o 11.1'W). This strain belongs to the 'high light-adapted' ecotype, clade II, and has a low Chl b/a2 ratio.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Prochlorococcus marinus MIT 9303</h1>
<div class="content">
<p>This strain was collected from the Sargasso Sea at a depth of 100 m and was isolated by filter fractionation. It can grow only in a narrow range of light intensities. This strain belongs to the 'low light-adapted' ecotype, clade IV, and has a high Chl b/a2 ratio.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Prochlorococcus marinus NATL1A</h1>
<div class="content">
<p>This strain was collected from the north Atlantic Ocean at a depth of 30 m and was isolated by filter fractionation. This strain belongs to the 'low light-adapted' ecotype, clade I, and has a high Chl b/a2 ratio.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Prosthecochloris aestuarii</h1>
<div class="content">
<p>This species is a green sulfur bacterium which forms sedimentary biofilm layers. It has been shown to be associated with coral killed by Black-Band Disease (BBD) a microbial infection of larger coral species. This is a concern for reef conservationists as the larger species are responsible for coral scaffolds, and their reduction by disease would have considerable impact on the reef structure. While there is currently no cause-and-effect link between Prosthecochloris aestuarii and BBD, the species was found on coral which was killed by the disease and was not found on healthy coral or in the surrounding seawater.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Prosthecochloris aestuarii Strain DSM 271 is the type strain for the species and will be used for comparative genome analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Chlorobium phaeovibrioides.</h1>
<div class="content">
<p>This organism is unique in its ability to form extracellular sulfur globules from thiosulfate. It is one of the few strains which can use both thiosulfate and tetrathionate as electron donors.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Proteus mirabilis.</h1>
<div class="content">
<p>This bacterium is part of the normal gut flora of a healthy individual, but is also an opportunistic pathogen. Flagellar movement allows this organism to travel through the urinary tract into the bladder and kidney where it may cause infection or stones. Fimbriae allow attachment to mucosal and epithelial surfaces. Cell death is caused by hemolysin, a pore-forming toxin which disrupts osmotic balance across the host cell membrane. This species is inherently resistant to nitrofuran and tetracycline, and some strains have recently become resistant to ampicillin and trimethoprin.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
P. mirabilis can be diagnosed in the lab due to characteristic swarming motility, and inability to metabolize lactose (on a MacConkey agar plate, for example.) Also P. mirabilis produces a very distinct odour.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas aeruginosa</h1>
<div class="content">
<p>Pseudomonas aeruginosa is an opportunistic human pathogen. While it rarely infects healthy individuals, immunocompromised patients, like burn victims, AIDS-, cancer- or cystic fibrosis-patients are at increased risk for infection with this environmentally versatile bacteria. It is an important soil bacterium with a complex metabolism capable of degrading polycyclic aromatic hydrocarbons, and producing interesting, biologically active secondary metabolites including quinolones, rhamnolipids, lectins, hydrogen cyanide, and phenazines. Production of these products is likely controlled by complex regulatory networks making Pseudomonas aeruginosa adaptable both to free-living and pathogenic lifestyles. The bacterium is naturally resistant to many antibiotics and disinfectants, which makes it a difficult pathogen to treat.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Pseudomonas aeruginosa LESB58 is a member of the Liverpool epidemic strains (LES) first isolated at the Liverpool Cystic Fibrosis (CF) clinic center. These isolates are highly virulent and readily transfered between CF patients and to non-CF individuals. This organism will be used for comparative genomics with other Pseudomonas aeruginosa strains.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas aeruginosa PA7</h1>
<div class="content">
<p>Pseudomonas aeruginosa is a common bacterium which can cause disease in animals and humans. It is found in soil, water, skin flora and most man-made environments throughout the world. It thrives not only in normal atmospheres, but also with little oxygen, and has thus colonised many natural and artificial environments. It uses a wide range of organic material for food; in animals, the versatility enables the organism to infect damaged tissues or people with reduced immunity. The symptoms of such infections are generalised inflammation and sepsis. If such colonisations occur in critical body organs such as the lungs, the urinary tract, and kidneys, the results can be fatal.[1] Because it thrives on moist surfaces, this bacterium is also found on and in medical equipment including catheters, causing cross infections in hospitals and clinics. It is implicated in hot-tub rash.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
strain PA7 is a clinical isolate and will be used for comparative genomics.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas entomophila</h1>
<div class="content">
<p>Pseudomonas entomophila is highly pathogenic for a variety of insects in both larvae and adults. It was isolated from fruit flies and decaying fruits taken from a sample set obtained from the Island of Guadeloupe and tested for induction of a systemic immune response in Drosophila. Destruction of the insect gut tissue occurs during infection.\n\nAnalysis of the proteins encoded by the genome indicated a number of potential virulence factors, although a type III secretion system was not found. One important pathogenicity factor was a secreted metalloprotease, AprA, that provides defense against the insect's immune response.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The determination of the complete sequence of the genome of strain L48 will help in the genetic analysis of this bacterium, and will also lead to the development of genomic tools for the study of the phylogenetic relationships between this strain with other Pseudomonaceae.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas fluorescens Pf0-1</h1>
<div class="content">
<p>Pseudomonas fluorescens is a nonpathogenic saprophyte which inhabits soil, water and plant surface environments. If iron is in low supply, it produces a soluble, greenish fluorescent pigment, which is how it was named. As these environmentally versatile bacteria possess the ability to degrade (at least partially) multiple different pollutants, they are studied in their use as bioremediants. Furthermore a number of strains also posses the ability to suppress agricultural pathogens like fungal infections, hence their role as biocontrol (biological disease control) agents is under examination.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
strain Pf0-1 was isolated from agricultural loam (sand, clay, and organic matter) soil in 1988 by Compeau et al. and is well adapted to soil environments.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas mendocina</h1>
<div class="content">
<p>Pseudomonas mendocina is a Gram-negative environmental bacterium that can cause opportunistic nosocomial infections, such as infective endocarditis and spondylodiscitis, although cases are very rare. It has potential use in bioremediation as it is able to degrade toluene. Based on 16S rRNA analysis, P. mendocina has been placed in the P. aeruginosa group</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas putida F1</h1>
<div class="content">
<p>Pseudomonas putida share a high degree of homology to Pseudomonas aeruginosa strains. However, key virulence factors like certain exotoxin genes and type III secretion systems are absent, which makes these bacteria non-pathogenic in nature. As they are metabolically versatile, and well characterized, it makes them great candidates for biocatalysis, bioremediation and other agricultural applications. Certain strains have been used in the production of bioplastics.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from a polluted creek in Urbana, Illinois by enrichment culture with ethylbenzyne as a sole source of carbon and energy. Its ability to degrade several different compounds including benzene, toluene, and ethylbenzene makes this species useful in the bioremediation of sites contaminated with multiple aromatic hydrocarbons. Underground gasoline tanks which have developed leaks can contaminate soil and water with a variety of these compounds. A more detailed knowledge of the metabolic pathways which allow Pseudomonas putida F1 to degrade aromatic hydrocarbons will be invaluable for the bioremediation industry.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas putida GB-1</h1>
<div class="content">
<p>Pseudomonas putida is a gram-negative rod-shaped saprotrophic soil bacterium. Based on 16S rRNA analysis, P. putida has been placed in the P. putida group, to which it lends its name. It is the first patented organism in the world. Due to the fact it is a living organism the patent was disputed and brought before the United States Supreme Court in the historic court case Diamond v. Chakrabarty which the inventer, Ananda M. Chakrabarty, won. It demonstrates a very diverse metabolism, including the ability to degrade organic solvents such as toluene[2]. This ability has been put to use in bioremediation, or the use of microorganisms to biodegrade oil. Use of P. putida is preferable to some other Pseudomonas species capable of such degradation as it is a safe strain of bacteria, unlike P. aeruginosa for example, which is an opportunistic human pathogen.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
strain GB-1 is able to oxidize soluble manganese(II) to the insoluble manganese(IV) which is deposited outside the cell. Pseudomonas putida GB-1 is being sequenced for comparative genome analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas putida W619</h1>
<div class="content">
<p>Pseudomonas putida is a common endophytic and rhizosphere bacterium.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Pseudomonas putida W619 was isolated from the Black Cottonwood tree and is closely related to other endophytic and rhizosphere strains of Pseudomonas putida. This strain is being sequenced for comparative genome analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas stutzeri</h1>
<div class="content">
<p>Pseudomonas stutzeri is a Gram-negative, rod-shaped, motile, single polar-flagellated, soil bacterium first isolated from human spinal fluid[1][2]. Unlike many other members of the genus Pseudomonas, it is not fluorescent. It is a denitrifying bacterium[3], and strain KC of P. stutzeri may be used for bioremediation as it is able to degrade carbon tetrachloride[4]. It is also an opportunistic pathogen in clinical settings, although infections are rare[5]. Based on 16S rRNA analysis, P. stutzeri has been placed in the P. stutzeri group, to which it lends its nam</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain A1501 was isolated from rice roots that had been inoculated with strain A15 in a rice paddy in China. It is considered wild-type and has been studied for the nitrogen-fixation process and is used as an inoculant during rice cultivation.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas syringae pv. phaseolicola 1448A</h1>
<div class="content">
<p>Pseudomonas syringae includes many plant pathogens of important crops, which makes it a model organism in plant pathology. Its natural environment is on the surface of plant leaves and it can withstand various stressful conditions, like rain, wind, UV radiation and drought. It can colonize plants in a non-pathogenic state and can rapidly take advantage of changing environmental conditions to induce disease in susceptible plants by shifting gene expression patterns.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain 1448A is characterized as causing halo blight on beans (Phaseolus vulgaris). This bacterium is closely related to Pseudomonas syringae pv. glycinea and Pseudomonas syringae pv. savastanoi and produces phaseolotoxin, which is a phytotoxin. Strain 1448A is a natural occurring rifampicin-resistant mutant strain and contains a non-functional homolog of avrPphE.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pseudomonas syringae pv. syringae B728a</h1>
<div class="content">
<p>This strain is the causal agent of brown spot disease on beans. This strain exhibits an epiphytic phase on plants that subsequently invades and causes disease. It was isolated from a snap bean leaflet in Wisconsin. It can also nucleate ice at temperatures above the freezing point.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Psychrobacter PRwf-1</h1>
<div class="content">
<p>These bacteria are commonly isolated from low temperature environments, Psychrobacter spp. are cold-adapted organisms that are often isolated from extreme environments such as permafrost or the Antarctic ice.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This organism grows slower than Psychrobacter arcticus 273-4 at 4oC, but is capable of growth at 37oC, while strain 273-4 is not.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Psychrobacter cryohalolentis</h1>
<div class="content">
<p>Psychrobacter cryohalolentis, formerly Psychrobacter cryopegella is an psychrotolerant bacterium. This organism can grow at -100C with rapid growth at these temperatures.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain K5 was isolated from saline liquid (12-14%) found 11-24 m below the surface within a forty thousand-year-old Siberian permafrost at the Kolyma-Indigirka lowland in Siberia. This strain will provide insight into growth at extremely low temperatures.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Psychromonas ingrahamii</h1>
<div class="content">
<p>Psychromonas ingrahamii is a marine psycrophile. This organism requires salt for growth and can grow at temperatures that are below freezing.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Psychromonas ingrahamii 37 was isolated from sea ice, off Point Barrow in northern Alaska. This organism is able to grow at -12oC, the lowest growth temperature for any bacterium to date. This organism will provide information on adaptation to extreme cold.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Pyrobaculum arsenaticum</h1>
<div class="content">
<p>Pyrobaculum arsenaticum is able to grow both chemolithoautotrophically, using hydrogen as the electron donor, to reduce arsenate, thiosulfate, or elemental sulfur, and organotrophically, using arsenate and thiosulfate as electron acceptors. Pyrobaculum arsenaticum may play an important role in the cycling of arsenic and sulfur in hyperthermal environments.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Pyrobaculum arsenaticum strain DSM 13514 was isolated from a hot spring at Pisciarelli Solfatara in Italy, and is the type strain for this species.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ralstonia eutropha H16</h1>
<div class="content">
<p>Cupriavidus necator also known as Ralstonia eutropha is a soil bacterium with diverse metabolic abilities. Strains of this organism are resistant to high levels of copper or are able to degrade chloroaromatic compounds such as halobenzoates and nitrophenols making them useful for bioremediation. Other strains have been studied for their ability to produce polyhydroxybutyrates which have industrial application. Another strain is able to attack other bacteria and fungi when nutrients in the soil are low.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain H16, formerly Alcaligenes eutrophus was originally isolated from sludge. This organism can utilize hydrogen, carbon dioxide, as well as organic compounds for growth and is a model organism for hydrogen oxidation as it can grow on hydrogen as the sole energy source.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ralstonia eutropha JMP134</h1>
<div class="content">
<p>This organism is found in both soil and water and has great potential for use in bioremediation as it is capable of degrading a large list of pollutants including chlorinated aromatic compounds. The bacterium can utilize hydrogen, carbon dioxide, as well as organic compounds for growth and is a model organism for hydrogen oxidation as it can grow on hydrogen as the sole energy source. It was originally isolated due to its ability to degrade the herbicide 2,4-dichlorophenoxyacetic acid, which is due to the degradative functions being encoded on a plasmid (pJP4).\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ralstonia metallidurans CH34</h1>
<div class="content">
<p>Ralstonia metallidurans is capable of growth in the presence of a large number of heavy metals including silver, cadmium, chromium, cobalt, copper, mercury, nickel, lead, thallium, and zinc.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain CH34 was first identified in the heavy metal-contaminated sludge of a settling tank in Belgium in the late 1970s. Metal resistance functions are predominantly encoded on two plasmids, pMOL28 and pMOL30, which produce metal exporters that pump metal ions out of the cell, thus protecting intracellular macromolecules from the toxic effects of high concentrations of metal.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Ralstonia pickettii</h1>
<div class="content">
<p>Ralstonia pickettii can be isolated from a wide range of environmental and clinical samples. This species is a nocosomial pathogen. It is associated with infections caused by contaminated solutions, such as distilled water and intravenous solutions. Although not common, the infections can be serious when they do occur, and include bacteremia and septicemia. Ralstonia pickettii strains resistant to heavy metals have also been isolated from heavy metal contaminated environments.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Ralstonia pickettii 12J was isolated from copper-contaminated sediment from a lake in Michigan. In addition to being highly resistant to copper, strain 12J is also resistant to zinc and cadmium. This organism will be used for comparative analysis with other heavy metal resistant microorganisms.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Renibacterium salmoninarum</h1>
<div class="content">
<p>This bacterial pathogen is the causal agent of bacterial kidney disease (BKD) of salmonid fish. This disease is a significant threat to the healthy and sustainable production of salmon for consumption and species conservation efforts in the United States. Control of BKD is exceptionally difficult as there are neither specific vaccines nor approved therapeutics. The development of treatments are further complicated as the organism is vertically transmitted from infected females to their eggs, is difficult to culture in vitro, and is a facultative intracellular pathogen.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
The completion of the genome sequence of Renibacterium salmoninarum will be a benefit to the scientific community at many levels and will be of immediate use to researchers and fish health practitioners dealing with BKD in farmed fish as well as captive broodstocks of endangered species-listed populations. For example, genome data will serve as a foundation for identification of virulence factors, DNA vaccine gene candidates, and improved diagnostic tests.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhizobium etli CFN 42 CFN42</h1>
<div class="content">
<p>Rhizobium etli belongs to the Rhizobiacea family of the alpha proteobacteria subdivision. It is a nitrogen-fixing symbiotic bacteria that interacts with the root of the common bean plant Phaseolus vulgaris. The molecular biology and evolution of R. etli and its relation to other nitrogen/fixing symbionts has been well studied. The genomic sequence will provide information on the process of symbiosis, on the genetic systems that allow the survival and adaptations of this bacteria to the soil, and on the evolutionary relationship and the symbiosis origin of this organism.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain is a model organism for the study of Rhizobium etli and is being sequenced in order to gain a better understanding of symbiosis and adaptation to soil and root environments.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhizobium etli CFN 42 CFN42</h1>
<div class="content">
<p>This strain is a model organism for the study of Rhizobium etli and is being sequenced in order to gain a better understanding of symbiosis and adaptation to soil and root environments.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhizobium leguminosarum</h1>
<div class="content">
<p>Rhizobium leguminosarum, like other Rhizobia, establishes a symbiotic relationship with a legume plant, providing nitrogen in exchange for a protected environment. The legume roots secrete flavonoids and isoflavonoids which the bacteria recognize and use to turn on genes involved in root nodulation. Many of the root nodulation genes are involved in synthesis and secretion of a nodule inducing signal, a lipochito-oligosaccharide molecule, which the plant recognizes, triggering nodule formation. The bacterium is endocytosed and exists inside a membrane bound organelle, the symbiosome, and fixes nitrogen for the plant cell while the host cell provides carbon compounds for the bacterium to grow on. The nitrogen fixation is important as it obviates the need for expensive and environmentally damaging fertilizer use.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Rhizobium leguminosarum biovar  trifolii WSM2304. This biovar is a symbiont of clover plants and is important commercially as it is used in the agricultural industry. Strain WSM1325 is compatible with many perennial clovers of American and African origin used in farming, and is therefore one of the most important clover inoculants but is incompatible with Mediterranean perennial clovers, such as those nodulated by the dissimilar strain WSM1325.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhizobium leguminosarum bv viciae 3841</h1>
<div class="content">
<p>This biovar nodulates legumes in the Tribe Viciae (Vicia, Pisum, Lathyrus, Lens). This strain is a spontaneous streptomycin-resistant mutant of strain 300. The host-specificity and nitrogen fixation genes are carried on large plasmids (of which it has 6).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhodobacter sphaeroides</h1>
<div class="content">
<p>The most extensively studied bacteria with regards to its photosynthetic capabilities which includes the structure, function and regulation of its photosynthetic membranes, its mechanisms of CO2 and nitrogen fixation, cytochrome diversity and its electron transport systems. It can grow aerobically and anaerobically in the light and anaerobically in the dark. It produces an intracytoplasmic membrane system consisting of membrane invaginations where the light harvesting complexes (LH1 and LH2) and the reaction center are synthesized. Furthermore, it has the ability to detoxify metal oxides and oxyanions and hence has a role in bioremediation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhodococcus jostii</h1>
<div class="content">
<p>Rhodococcus jostii is capable of degrading a wide variety of polychlorinated biphenyls (PCBs), which are toxic compounds produced by industry, and is studied extensively for this reason. This microbe is being examined thoroughly under different environmental conditions in order to better understand metabolic pathways and gene functions.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Rhodococcus jostii RHA1 was isolated from soil contaminated with gamma-hexachlorocyclohexane (lindane: a toxic insecticide) in Japan. This organism may be useful in the bioremediation of contaminated environmental sites.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhodoferax ferrireducens</h1>
<div class="content">
<p>Rhodoferax ferrireducens is distinct from other members of this genus in that it is not photosynthetic, and can couple the oxidation of organic molecules to the reduction of iron by using insoluble Fe(III) as an electron acceptor. In addition, this organism is able to generate electricity in microbial fuel cells.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Rhodoferax ferrireducens T118 was isolated from aquifer sediment collected at a depth of 18 feet. The genome sequence will provide information on the mechanism and control of electricity generation and iron reduction by Rhodoferax ferrireducens.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhodopseudomonas palustris</h1>
<div class="content">
<p>Rhodopseudomonas palustris has a diverse metabolism and is capable of growth using light, inorganic, or organic compounds as energy sources and carbon dioxide or organic compounds as carbon sources. Commonly found in soil and water environments this bacterium is also capable of degrading a wide range of toxic organic compounds, and may be of use in bioremediation of polluted sites. The bacterium undergoes differentiation to produce a stalked nonmotile cell and a motile flagellated cell. In the presence of light, this bacterium produces a number of intracellular membranous vesicles to house the photosynthetic reaction centers.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Rhodopseudomonas palustris strain BisA53. Four different strains were isolated from 2 sites, one pristine and one polluted. The 16S rRNA sequences of these strains differ from that of strain CGA009 by about 2%. Comparison with the sequenced strain will reveal differences in metabolic diversity as well as phenotypic differences.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhodospirillum centenum</h1>
<div class="content">
<p>Rhodospirillum centenum, also called Rhodocista centenaria, is a nitrogen-fixing photoheterotroph with a complex life cycle. R. centenum is one of the few known thermotolerant purple bacteria species with optimal growth temperature of 44oC and a maximal growth temperature of 48oC. In liquid media this organism is motile by a single polar flagellum. Upon contact with a surface, R. centenum produces lateral flagella to become a swarming cell. Colonies of swarm cells exhibit phototactic behavior. Under low nutrient conditions R. centenum forms a desiccation- and UV-resistant cyst. Cysts can also survive temperatures as high as 65oC. Consequently, R. centenum can often be cultivated from hot springs such as those found at Yellowstone National Park. R. centenum is emerging as a model organism for genetic and molecular genetic analysis of cyst formation.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rhodospirillum rubrum</h1>
<div class="content">
<p>This organism is an anoxygenic phototrophic bacterium that does not produce oxygen, but instead produces extracellular elemental sulfur when harvesting light energy. The bacterium lacks the light harvesting complex 2 (LHC2) normally found in photosynthetic bacteria meaning it contains one of the simplest photosynthetic systems studied.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Rhodospirillum rubrum ATCC 11170 can grow using carbon monoxide as the sole carbon and energy source and the cells contain a well characterized nitrogenase system that is post-translationally modified by ADP-ribosylation.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rickettsia akari</h1>
<div class="content">
<p>This organism causes a mild disease, Rickettsialpox, which is an acute fever-inducing illness transmitted by a hematophagous mite that infects the common house mouse and bites humans. A black eschar may develop, along with a fever, headache, rash, and systemic symptoms. Infection by this organism may be confused with anthrax due to the black eschar. This bacterium is a member of the spotted fever group of Rickettsiales and is endemic to New York, USA, but is also found in other cities in the USA, Russia, South Korea, and South Africa.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rickettsia bellii</h1>
<div class="content">
<p>This organism represents the third group of Rickettsial species and is separate from the spotted fever group and the typhus group. This organism is transovarially transmitted in American ticks and shows the largest range of host organisms including Dermacentor and Amblyomma.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain (RML 369-C) was isolated in embryonated chicken eggs from a triturated (powdered) pool of unfed adult Dermacentor variabilis ticks collected from vegetation near Fayetteville, Arkansas, USA, in 1966. The genome sequence does not exhibit colinearity with other Rickettsial genomes. A set of conjugative genes were found in the chromosome.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rickettsia canadensis</h1>
<div class="content">
<p>This organism was originally thought to be a member of the typhus group of Rickettsiales, however, it is now thought to represent a distict group with the rickettsia. Serology has indicated that R. canadensis may occasionally infect humans.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Rickettsia canadensis str. McKiel was originally isolated from ticks in a field study on tick-transmitted diseases of small mammals in Canada. Rickettsia canadensis str. McKiel will be used for comparative analysis with other Rickettsia species.\n
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rickettsia felis URRWXCa12</h1>
<div class="content">
<p>The genome of Rickettsia felis strain URRWXCa12 (strain California 2) reveals some unusual features. There are two plasmids (pRF and pRFdelta), the first such identified in the Rickettsia genus, and the pRF plasmid has genes that encode proteins similar to those found on conjugative plasmids such as TraA from Agrobacterium tumefaciens, F plasmid TraD and TraG, and the chromosome encodes proteins similar to the vir genes of A. tumefaciens. The chromosome encodes a protein that is similar to the IncP plasmid primase protein TraC. All of these functions may be involved in pathogenesis and the movement of proteins and DNA and electron microscopy confirmed the evidence of a conjugative pilus in this organism as well as other fimbriae.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rickettsia massiliae</h1>
<div class="content">
<p>This bacterium is a member of the spotted fever group of the Rickettsiales and has been isolated from ticks in Europe and Africa. Rickettsia massiliae does not appear to cause disease in humans.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Rickettsia rickettsii.</h1>
<div class="content">
<p>This organism was first identified by Dr. Howard Rickets as the causative agent of Rocky Mountain Spotted Fever, which was originally named for its geographic distribution at the time, it is now known to be widespread throughout the North American continent. This bacterium is an obligate intracellular pathogen that infects primarily the vascular endothelium, and occasionally smooth muscle tissue. It is passed to the human host from a tick bite, and the tick acts as both a natural reservoir and a vector for disease transmission. Once the organism is endocytosed by the host cell, it quickly escapes the phagozome, and replicates intracellularly, causing cell death and tissue damage. The disease is characterized by a spotted rash and has a high mortality rate if left untreated.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Roseiflexus RS-1</h1>
<div class="content">
<p>Roseiflexus sp. RS-1 was isolated from a hot spring microbial mat. This organism is a member of a genus of anoxygenic phototrophic bacteria, characterized by the absence of chlorosomes (pigment containing vesicles) and bacteriochlorophyll-c. Bacteriochlorophyll-a is the major photosynthetic pigment for Roseiflexus spp. Roseiflexus sp. RS-1 will be used for comparative genomics with other photosynthetic bacteria.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Roseiflexus castenholzii</h1>
<div class="content">
<p>Roseiflexus castenholzii is related to the Chloroflexaceae, a family of anoxygenic phototrophs. However, it differs from other members of that family in that it lacks chlorosomes and bacteriochlorophyll-c. Bacteriochlorophyll-a is the major photosynthetic pigment in Roseiflexus castenholzii.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Roseobacter denitrificans OCh 114</h1>
<div class="content">
<p>A single colony of Roseobacter denitrificans strain OCh 114 was grown heterotrophically and total DNA was isolated using proteinase K treatment followed by phenol extraction. The DNA was fragmented by kinetic shearing, and three shotgun libraries were generated: small and medium insert libraries in pOTWI3 (using size fractions of 2-3 kb and 6-8 kb, respectively), and a large insert fosmid library in pEpiFOS-5 (insert sizes ranging from 28-47 kb), which was used as a scaffold. The relative amount of sequence coverage obtained from the small, medium, and large insert libraries was 8X, 1X, and 1X, respectively. The whole genome sequence was established from 55,081 end sequences (giving 9.6X coverage) derived from these libraries using dye terminator chemistry on ABI 3730xl automated sequencers. The aerobic phototrophic bacteria are ubiquitous as plant symbionts, free-living in lakes and ocean surface waters, soils and even near deep sea hydrothermal vents. In fact, ~10% of all microorganisms in marine surface water samples appear to be aerobic phototrophic bacteria, suggesting a significant contribution to the global carbon cycle. Although the numbers of aerobic phototrophic bacteria vary with locale (Goericke, 2002), it is clear that these organisms are a previously unrecognized major component of the bacterial biomass on Earth. The genome sequence of this representative aerobic phototrophic bacteria species will enable discovery in the following three areas: 1) the evolutionary genesis of photosynthesis genes; 2) pathways of carbon dioxide fixation and production; 3) light and oxygen signal transduction in gene expression.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Saccharopolyspora erythraea</h1>
<div class="content">
<p>Saccharopolyspora erythraea is the soil bacterium that produces the industrially important antibiotic erythromycin A. Erythromycin is a clinically important and potent macrolide antibiotic. It is used to treat infections caused by several prokaryotic pathogens such as Streptococcus, Staphylococcus, Mycoplasma, Ureaplasma, Chlamydia and Legionella. Production of this antibiotic is lower than others in the same class, such as penicillin or cephalosporin, which has led to the development of a genetic system to attempt to enhance the production of erythromycin.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salinibacter ruber</h1>
<div class="content">
<p>This organism is an extremely halophilic aerobe originally isolated from saltern crystallizer ponds in Spain. This red-pigmented bacterium belongs to the Cytophaga-Flexibacter-Bacteroides group. These bacteria can coexist in significant colonies with halophilic archaea under saline conditions. In contrast to other bacteria they do not regulate their intracellular salt conditions through proton pumps, but instead their protein makeup has adapted to be functional under high ionic conditions.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica arizonae serovar 62:z4,z23:RSK2980</h1>
<div class="content">
<p>This subspecies (IIIa) is usually found associated with reptiles, although contact with infected animals can result in the spread of the organism to humans or animals such as turkeys. This strain was originally isolated from a cornsnake in 1986 in Oregon.\n</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica enterica serovar Agona</h1>
<div class="content">
<p>Salmonella enterica subsp. enterica serovar Agona causes gastroenteritis in humans and is also pathogenic to swine and other food animals. This serovar is able to contain the Salmonella genomic island 1 multidrug resistance gene cluster. This strain will be used for comparative analysis with other Salmonella serovars.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica enterica serovar Dublin CT_02021853</h1>
<div class="content">
<p>This is a bovine-adapted serovar that is genetically related to S. Enteritidis that lives in the bovine intestinal tract and can causes disease in humans. While rare in incidence, S. dublin infection classically produces a syndrome of sustained bacteremia with fever, resulting in high morbidity and mortality.\n\nSequence was obtained from two libraries - a fosmid library (33-39 kb; 21,645 reads) and a small insert library (4-6 kb; 29,869 reads).</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica enterica serovar Gallinarum  287/91</h1>
<div class="content">
<p>Salmonella gallinarum is the causative agent of Fowl typhoid, a severe systemic disease of poultry. Salmonella enterica subsp. enterica serovar Gallinarum str. 287/91 will be used for comparative analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica enterica serovar Heidelberg</h1>
<div class="content">
<p>Salmonella enterica subsp. enterica serovar Heidelberg is one of the more common serovars causing disease in the United States.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
Strain SL476 this is a multidrug resistant strain and will be used for comparative analysis.
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica enterica serovar Paratyphi A AKU 12601</h1>
<div class="content">
<p>This strain is a clinical isolate collected from a child with paratyphoid fever in Karachi, Pakistan, 2004 and has been sequenced for comparative genome analysis.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica enterica serovar Paratyphi C  RKS4594</h1>
<div class="content">
<p>The genome of this strain harbors a large pathogenicity island, SPI7, with genes coding for the Vi (virulence) antigen and other genes potentially associated with virulence. The genome of strain RKS4594, when compared to other S. enterica subsp. enterica serovar Paratyphi C strains, features several insertions, deletions and rearrangements.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica enterica serovar Paratyphi B SPB7</h1>
<div class="content">
<p>This strain was isolated from a stool sample of an infected woman in Penang, Malaysia, May 16, 2002. This strain is susceptible to antibiotics, and was classified as serovar Paratyphi B because it was unable to metabolize D-tartrate.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica enterica serovar Schwarzengrund CVM19633</h1>
<div class="content">
<p>This serovar is the predominant cause of Salmonellosis in Southeast Asia, a major source of imported food products to the United States. It was also the cause of the first recognized outbreak of fluoroquinolone-resistant salmonellosis in the U.S.. Recent reports suggest that high-level fluoroquinolone resistance is emerging in S. Schwarzengrund in different parts of the world. This serovar is remarkably disposed to nosocomial spread, and presents a unique opportunity to identify factors that facilitate this important type of transmission. MDR strain CVM19633 and susceptible strain DBS_GA_F25499 belong to the S. Schwarzengrund serovar.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Salmonella enterica  enterica serovar Typhi Ty2.</h1>
<div class="content">
<p>This pathogenic strain of Salmonella typhi was isolated in the early 1970s. It contains no multidrug resistance plasmids and has been used for vaccine development.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
NULL
</div>
</div>
</div>
</div>
</div>

<div class="node">
<h1>Shewanella  ANA-3</h1>
<div class="content">
<p>This genus includes species that inhabit a wide range of environments and are capable of utilizing a wide variety of electron acceptors during anaerobic respiration including some insoluble metal oxides while using very few carbon sources such as lactate or acetate. This group of organisms have been studied extensively for their electron transport systems.</p>
<div class="field field-type-text field-field-notes">
<div class="field-label">notes:&nbsp;</div>
<div class="field-items">
<div class="field-item odd">
This strain was isolated from an arsenate treated wood pier that was in a brackish estuary (Eel Pond) in Woods Hole, Massachusetts. It is capable of converting As(V) to arsenite and may be used in bioremediation of contaminated sites.
</div>
</div>
</div>
</div>
</div>

</div><!-- mainpage -->
{% endblock content %}
